// @caffeine/template-frontend/src/backend.ts
/* eslint-disable */

// @ts-nocheck

// This file was automatically generated by @caffeinelabs/bindgen@0.1.0.
// You should NOT make any changes in this file as it will be overwritten.
// Additionally, you should also exclude this file from your linter and/or formatter to prevent it from being checked or modified.

import { Actor, HttpAgent, type HttpAgentOptions, type ActorConfig, type Agent, type ActorSubclass } from "@icp-sdk/core/agent";
import type { Principal } from "@icp-sdk/core/principal";
import { idlFactory, type _SERVICE } from "./declarations/backend.did";
export interface Some<T> {
    __kind__: "Some";
    value: T;
}
export interface None {
    __kind__: "None";
}
export type Option<T> = Some<T> | None;
function some<T>(value: T): Some<T> {
    return {
        __kind__: "Some",
        value: value
    };
}
function none(): None {
    return {
        __kind__: "None"
    };
}
function isNone<T>(option: Option<T>): option is None {
    return option.__kind__ === "None";
}
function isSome<T>(option: Option<T>): option is Some<T> {
    return option.__kind__ === "Some";
}
function unwrap<T>(option: Option<T>): T {
    if (isNone(option)) {
        throw new Error("unwrap: none");
    }
    return option.value;
}
function candid_some<T>(value: T): [T] {
    return [
        value
    ];
}
function candid_none<T>(): [] {
    return [];
}
function record_opt_to_undefined<T>(arg: T | null): T | undefined {
    return arg == null ? undefined : arg;
}
export class ExternalBlob {
    _blob?: Uint8Array<ArrayBuffer> | null;
    directURL: string;
    onProgress?: (percentage: number) => void = undefined;
    private constructor(directURL: string, blob: Uint8Array<ArrayBuffer> | null){
        if (blob) {
            this._blob = blob;
        }
        this.directURL = directURL;
    }
    static fromURL(url: string): ExternalBlob {
        return new ExternalBlob(url, null);
    }
    static fromBytes(blob: Uint8Array<ArrayBuffer>): ExternalBlob {
        const url = URL.createObjectURL(new Blob([
            new Uint8Array(blob)
        ], {
            type: 'application/octet-stream'
        }));
        return new ExternalBlob(url, blob);
    }
    public async getBytes(): Promise<Uint8Array<ArrayBuffer>> {
        if (this._blob) {
            return this._blob;
        }
        const response = await fetch(this.directURL);
        const blob = await response.blob();
        this._blob = new Uint8Array(await blob.arrayBuffer());
        return this._blob;
    }
    public getDirectURL(): string {
        return this.directURL;
    }
    public withUploadProgress(onProgress: (percentage: number) => void): ExternalBlob {
        this.onProgress = onProgress;
        return this;
    }
}
export interface Statistik {
    totalArtikel: bigint;
    totalDokumen: bigint;
    perKategori: Array<[KategoriId, string, bigint]>;
    totalUnduhan: bigint;
}
export type Timestamp = bigint;
export interface FilterDokumen {
    kategoriId?: KategoriId;
    status?: StatusDokumen;
    tahun?: bigint;
    katakunci?: string;
    workflowStatus?: WorkflowStatus;
}
export interface _ImmutableObjectStorageRefillResult {
    success?: boolean;
    topped_up_amount?: bigint;
}
export interface TentangPageInput {
    judul: string;
    konten: TentangContent;
}
export interface TentangPage {
    id: bigint;
    slug: TentangSlug;
    judul: string;
    updatedAt: Timestamp;
    updatedBy: string;
    konten: TentangContent;
}
export interface GaleriItem {
    id: GaleriItemId;
    album: string;
    createdAt: Timestamp;
    createdBy: string;
    gambar?: ExternalBlob;
    judul: string;
    deskripsi: string;
}
export interface Artikel {
    id: ArtikelId;
    status: string;
    tanggal: Timestamp;
    tags: Array<string>;
    ringkasan: string;
    gambar?: ExternalBlob;
    judul: string;
    author: string;
    tanggalPublikasi: bigint;
    konten: string;
}
export interface TentangContent {
    blocks: Array<TentangBlock>;
}
export interface Kategori {
    id: KategoriId;
    nama: string;
}
export type GaleriItemId = bigint;
export type WorkflowResult = {
    __kind__: "ok";
    ok: string;
} | {
    __kind__: "err";
    err: string;
};
export type DokumenId = bigint;
export interface AdminUser {
    principal: string;
    grantedMenus: Array<MenuPermission>;
    nama?: string;
    isActive: boolean;
    addedAt: Timestamp;
    addedBy: string;
}
export interface DokumenInput {
    tag: Array<string>;
    kategoriId: KategoriId;
    status: StatusDokumen;
    tahun: bigint;
    judul: string;
    relasiHukum?: string;
    abstrak: string;
    tanggalPenetapan: string;
    nomor: string;
    tanggalPengundangan?: bigint;
    workflowStatus?: WorkflowStatus;
    filePdf: ExternalBlob;
}
export interface MyPermissions {
    grantedMenus: Array<MenuPermission>;
    isAdmin: boolean;
}
export interface ActivityLog {
    id: bigint;
    action: string;
    performedAt: Timestamp;
    performedBy: string;
    targetTitle: string;
    targetType: string;
    targetId: string;
}
export interface _ImmutableObjectStorageRefillInformation {
    proposed_top_up_amount?: bigint;
}
export type ArtikelId = bigint;
export interface _ImmutableObjectStorageCreateCertificateResult {
    method: string;
    blob_hash: string;
}
export interface GaleriItemInput {
    album: string;
    gambar?: ExternalBlob;
    judul: string;
    deskripsi: string;
}
export type TentangBlock = {
    __kind__: "daftarItem";
    daftarItem: Array<string>;
} | {
    __kind__: "paragraf";
    paragraf: string;
};
export type KategoriId = bigint;
export interface ArtikelInput {
    status: string;
    tags: Array<string>;
    ringkasan: string;
    gambar?: ExternalBlob;
    judul: string;
    author: string;
    tanggalPublikasi?: bigint;
    konten: string;
}
export interface Dokumen {
    id: DokumenId;
    tag: Array<string>;
    kategoriId: KategoriId;
    status: StatusDokumen;
    tahun: bigint;
    judul: string;
    relasiHukum?: string;
    reviewedAt?: Timestamp;
    reviewedBy?: Principal;
    catatanKoreksi: string;
    abstrak: string;
    tanggalPenetapan: string;
    nomor: string;
    downloadCount: bigint;
    tanggalPengundangan?: bigint;
    workflowStatus: WorkflowStatus;
    filePdf: ExternalBlob;
}
export enum MenuPermission {
    galeri = "galeri",
    userManagement = "userManagement",
    kategori = "kategori",
    dokumen = "dokumen",
    statistik = "statistik",
    berita = "berita",
    tentang = "tentang"
}
export enum StatusDokumen {
    Berlaku = "Berlaku",
    Dicabut = "Dicabut",
    TidakBerlaku = "TidakBerlaku"
}
export enum TentangSlug {
    visiMisi = "visiMisi",
    sejarah = "sejarah",
    fungsi = "fungsi",
    struktur = "struktur",
    dasarHukum = "dasarHukum"
}
export enum UserRole {
    admin = "admin",
    user = "user",
    guest = "guest"
}
export enum WorkflowStatus {
    PendingReview = "PendingReview",
    Draft = "Draft",
    Published = "Published"
}
export interface backendInterface {
    _immutableObjectStorageBlobsAreLive(hashes: Array<Uint8Array>): Promise<Array<boolean>>;
    _immutableObjectStorageBlobsToDelete(): Promise<Array<Uint8Array>>;
    _immutableObjectStorageConfirmBlobDeletion(blobs: Array<Uint8Array>): Promise<void>;
    _immutableObjectStorageCreateCertificate(blobHash: string): Promise<_ImmutableObjectStorageCreateCertificateResult>;
    _immutableObjectStorageRefillCashier(refillInformation: _ImmutableObjectStorageRefillInformation | null): Promise<_ImmutableObjectStorageRefillResult>;
    _immutableObjectStorageUpdateGatewayPrincipals(): Promise<void>;
    _initializeAccessControl(): Promise<void>;
    addAdminUser(principal: string, nama: string | null, grantedMenus: Array<MenuPermission>): Promise<{
        __kind__: "ok";
        ok: AdminUser;
    } | {
        __kind__: "err";
        err: string;
    }>;
    addBerita(input: ArtikelInput): Promise<Artikel>;
    addDokumen(input: DokumenInput): Promise<Dokumen>;
    addGaleriItem(input: GaleriItemInput): Promise<GaleriItem>;
    addKategori(nama: string): Promise<Kategori>;
    assignCallerUserRole(user: Principal, role: UserRole): Promise<void>;
    deleteBerita(id: ArtikelId): Promise<boolean>;
    deleteDokumen(id: DokumenId): Promise<boolean>;
    deleteGaleriItem(id: GaleriItemId): Promise<boolean>;
    deleteKategori(id: KategoriId): Promise<boolean>;
    downloadDokumen(id: DokumenId): Promise<boolean>;
    getActivityLog(): Promise<Array<ActivityLog>>;
    getBeritaDetail(id: ArtikelId): Promise<Artikel | null>;
    getCallerUserRole(): Promise<UserRole>;
    getDokumenDetail(id: DokumenId): Promise<Dokumen | null>;
    getGaleriItem(id: GaleriItemId): Promise<GaleriItem | null>;
    getMyPermissions(): Promise<MyPermissions>;
    getPendingReviewCount(): Promise<bigint>;
    getStatistik(): Promise<Statistik>;
    getTentangPage(slug: TentangSlug): Promise<TentangPage | null>;
    isCallerAdmin(): Promise<boolean>;
    listAdminUsers(): Promise<Array<AdminUser>>;
    listBerita(): Promise<Array<Artikel>>;
    listDokumen(filter: FilterDokumen): Promise<Array<Dokumen>>;
    listDokumenAdmin(filter: FilterDokumen): Promise<Array<Dokumen>>;
    listGaleri(): Promise<Array<GaleriItem>>;
    listKategori(): Promise<Array<Kategori>>;
    listTentangPages(): Promise<Array<TentangPage>>;
    publishDokumen(id: DokumenId): Promise<WorkflowResult>;
    removeAdminUser(principal: string): Promise<{
        __kind__: "ok";
        ok: null;
    } | {
        __kind__: "err";
        err: string;
    }>;
    returnToDraft(id: DokumenId, catatan: string): Promise<WorkflowResult>;
    submitForReview(id: DokumenId): Promise<WorkflowResult>;
    updateAdminUser(principal: string, nama: string | null, grantedMenus: Array<MenuPermission>, isActive: boolean): Promise<{
        __kind__: "ok";
        ok: AdminUser;
    } | {
        __kind__: "err";
        err: string;
    }>;
    updateBerita(id: ArtikelId, input: ArtikelInput): Promise<Artikel | null>;
    updateDokumen(id: DokumenId, input: DokumenInput): Promise<Dokumen | null>;
    updateGaleriItem(id: GaleriItemId, input: GaleriItemInput): Promise<GaleriItem | null>;
    updateKategori(id: KategoriId, nama: string): Promise<Kategori | null>;
    updateTentangPage(slug: TentangSlug, input: TentangPageInput): Promise<TentangPage | null>;
}
import type { AdminUser as _AdminUser, Artikel as _Artikel, ArtikelId as _ArtikelId, ArtikelInput as _ArtikelInput, Dokumen as _Dokumen, DokumenId as _DokumenId, DokumenInput as _DokumenInput, ExternalBlob as _ExternalBlob, FilterDokumen as _FilterDokumen, GaleriItem as _GaleriItem, GaleriItemId as _GaleriItemId, GaleriItemInput as _GaleriItemInput, Kategori as _Kategori, KategoriId as _KategoriId, MenuPermission as _MenuPermission, MyPermissions as _MyPermissions, StatusDokumen as _StatusDokumen, TentangBlock as _TentangBlock, TentangContent as _TentangContent, TentangPage as _TentangPage, TentangPageInput as _TentangPageInput, TentangSlug as _TentangSlug, Timestamp as _Timestamp, UserRole as _UserRole, WorkflowResult as _WorkflowResult, WorkflowStatus as _WorkflowStatus, _ImmutableObjectStorageRefillInformation as __ImmutableObjectStorageRefillInformation, _ImmutableObjectStorageRefillResult as __ImmutableObjectStorageRefillResult } from "./declarations/backend.did.d.ts";
export class Backend implements backendInterface {
    constructor(private actor: ActorSubclass<_SERVICE>, private _uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, private _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, private processError?: (error: unknown) => never){}
    async _immutableObjectStorageBlobsAreLive(arg0: Array<Uint8Array>): Promise<Array<boolean>> {
        if (this.processError) {
            try {
                const result = await this.actor._immutableObjectStorageBlobsAreLive(arg0);
                return result;
            } catch (e) {
                this.processError(e);
                throw new Error("unreachable");
            }
        } else {
            const result = await this.actor._immutableObjectStorageBlobsAreLive(arg0);
            return result;
        }
    }
    async _immutableObjectStorageBlobsToDelete(): Promise<Array<Uint8Array>> {
        if (this.processError) {
            try {
                const result = await this.actor._immutableObjectStorageBlobsToDelete();
                return result;
            } catch (e) {
                this.processError(e);
                throw new Error("unreachable");
            }
        } else {
            const result = await this.actor._immutableObjectStorageBlobsToDelete();
            return result;
        }
    }
    async _immutableObjectStorageConfirmBlobDeletion(arg0: Array<Uint8Array>): Promise<void> {
        if (this.processError) {
            try {
                const result = await this.actor._immutableObjectStorageConfirmBlobDeletion(arg0);
                return result;
            } catch (e) {
                this.processError(e);
                throw new Error("unreachable");
            }
        } else {
            const result = await this.actor._immutableObjectStorageConfirmBlobDeletion(arg0);
            return result;
        }
    }
    async _immutableObjectStorageCreateCertificate(arg0: string): Promise<_ImmutableObjectStorageCreateCertificateResult> {
        if (this.processError) {
            try {
                const result = await this.actor._immutableObjectStorageCreateCertificate(arg0);
                return result;
            } catch (e) {
                this.processError(e);
                throw new Error("unreachable");
            }
        } else {
            const result = await this.actor._immutableObjectStorageCreateCertificate(arg0);
            return result;
        }
    }
    async _immutableObjectStorageRefillCashier(arg0: _ImmutableObjectStorageRefillInformation | null): Promise<_ImmutableObjectStorageRefillResult> {
        if (this.processError) {
            try {
                const result = await this.actor._immutableObjectStorageRefillCashier(to_candid_opt_n1(this._uploadFile, this._downloadFile, arg0));
                return from_candid__ImmutableObjectStorageRefillResult_n4(this._uploadFile, this._downloadFile, result);
            } catch (e) {
                this.processError(e);
                throw new Error("unreachable");
            }
        } else {
            const result = await this.actor._immutableObjectStorageRefillCashier(to_candid_opt_n1(this._uploadFile, this._downloadFile, arg0));
            return from_candid__ImmutableObjectStorageRefillResult_n4(this._uploadFile, this._downloadFile, result);
        }
    }
    async _immutableObjectStorageUpdateGatewayPrincipals(): Promise<void> {
        if (this.processError) {
            try {
                const result = await this.actor._immutableObjectStorageUpdateGatewayPrincipals();
                return result;
            } catch (e) {
                this.processError(e);
                throw new Error("unreachable");
            }
        } else {
            const result = await this.actor._immutableObjectStorageUpdateGatewayPrincipals();
            return result;
        }
    }
    async _initializeAccessControl(): Promise<void> {
        if (this.processError) {
            try {
                const result = await this.actor._initializeAccessControl();
                return result;
            } catch (e) {
                this.processError(e);
                throw new Error("unreachable");
            }
        } else {
            const result = await this.actor._initializeAccessControl();
            return result;
        }
    }
    async addAdminUser(arg0: string, arg1: string | null, arg2: Array<MenuPermission>): Promise<{
        __kind__: "ok";
        ok: AdminUser;
    } | {
        __kind__: "err";
        err: string;
    }> {
        if (this.processError) {
            try {
                const result = await this.actor.addAdminUser(arg0, to_candid_opt_n8(this._uploadFile, this._downloadFile, arg1), to_candid_vec_n9(this._uploadFile, this._downloadFile, arg2));
                return from_candid_variant_n12(this._uploadFile, this._downloadFile, result);
            } catch (e) {
                this.processError(e);
                throw new Error("unreachable");
            }
        } else {
            const result = await this.actor.addAdminUser(arg0, to_candid_opt_n8(this._uploadFile, this._downloadFile, arg1), to_candid_vec_n9(this._uploadFile, this._downloadFile, arg2));
            return from_candid_variant_n12(this._uploadFile, this._downloadFile, result);
        }
    }
    async addBerita(arg0: ArtikelInput): Promise<Artikel> {
        if (this.processError) {
            try {
                const result = await this.actor.addBerita(await to_candid_ArtikelInput_n19(this._uploadFile, this._downloadFile, arg0));
                return from_candid_Artikel_n22(this._uploadFile, this._downloadFile, result);
            } catch (e) {
                this.processError(e);
                throw new Error("unreachable");
            }
        } else {
            const result = await this.actor.addBerita(await to_candid_ArtikelInput_n19(this._uploadFile, this._downloadFile, arg0));
            return from_candid_Artikel_n22(this._uploadFile, this._downloadFile, result);
        }
    }
    async addDokumen(arg0: DokumenInput): Promise<Dokumen> {
        if (this.processError) {
            try {
                const result = await this.actor.addDokumen(await to_candid_DokumenInput_n26(this._uploadFile, this._downloadFile, arg0));
                return from_candid_Dokumen_n32(this._uploadFile, this._downloadFile, result);
            } catch (e) {
                this.processError(e);
                throw new Error("unreachable");
            }
        } else {
            const result = await this.actor.addDokumen(await to_candid_DokumenInput_n26(this._uploadFile, this._downloadFile, arg0));
            return from_candid_Dokumen_n32(this._uploadFile, this._downloadFile, result);
        }
    }
    async addGaleriItem(arg0: GaleriItemInput): Promise<GaleriItem> {
        if (this.processError) {
            try {
                const result = await this.actor.addGaleriItem(await to_candid_GaleriItemInput_n41(this._uploadFile, this._downloadFile, arg0));
                return from_candid_GaleriItem_n43(this._uploadFile, this._downloadFile, result);
            } catch (e) {
                this.processError(e);
                throw new Error("unreachable");
            }
        } else {
            const result = await this.actor.addGaleriItem(await to_candid_GaleriItemInput_n41(this._uploadFile, this._downloadFile, arg0));
            return from_candid_GaleriItem_n43(this._uploadFile, this._downloadFile, result);
        }
    }
    async addKategori(arg0: string): Promise<Kategori> {
        if (this.processError) {
            try {
                const result = await this.actor.addKategori(arg0);
                return result;
            } catch (e) {
                this.processError(e);
                throw new Error("unreachable");
            }
        } else {
            const result = await this.actor.addKategori(arg0);
            return result;
        }
    }
    async assignCallerUserRole(arg0: Principal, arg1: UserRole): Promise<void> {
        if (this.processError) {
            try {
                const result = await this.actor.assignCallerUserRole(arg0, to_candid_UserRole_n45(this._uploadFile, this._downloadFile, arg1));
                return result;
            } catch (e) {
                this.processError(e);
                throw new Error("unreachable");
            }
        } else {
            const result = await this.actor.assignCallerUserRole(arg0, to_candid_UserRole_n45(this._uploadFile, this._downloadFile, arg1));
            return result;
        }
    }
    async deleteBerita(arg0: ArtikelId): Promise<boolean> {
        if (this.processError) {
            try {
                const result = await this.actor.deleteBerita(arg0);
                return result;
            } catch (e) {
                this.processError(e);
                throw new Error("unreachable");
            }
        } else {
            const result = await this.actor.deleteBerita(arg0);
            return result;
        }
    }
    async deleteDokumen(arg0: DokumenId): Promise<boolean> {
        if (this.processError) {
            try {
                const result = await this.actor.deleteDokumen(arg0);
                return result;
            } catch (e) {
                this.processError(e);
                throw new Error("unreachable");
            }
        } else {
            const result = await this.actor.deleteDokumen(arg0);
            return result;
        }
    }
    async deleteGaleriItem(arg0: GaleriItemId): Promise<boolean> {
        if (this.processError) {
            try {
                const result = await this.actor.deleteGaleriItem(arg0);
                return result;
            } catch (e) {
                this.processError(e);
                throw new Error("unreachable");
            }
        } else {
            const result = await this.actor.deleteGaleriItem(arg0);
            return result;
        }
    }
    async deleteKategori(arg0: KategoriId): Promise<boolean> {
        if (this.processError) {
            try {
                const result = await this.actor.deleteKategori(arg0);
                return result;
            } catch (e) {
                this.processError(e);
                throw new Error("unreachable");
            }
        } else {
            const result = await this.actor.deleteKategori(arg0);
            return result;
        }
    }
    async downloadDokumen(arg0: DokumenId): Promise<boolean> {
        if (this.processError) {
            try {
                const result = await this.actor.downloadDokumen(arg0);
                return result;
            } catch (e) {
                this.processError(e);
                throw new Error("unreachable");
            }
        } else {
            const result = await this.actor.downloadDokumen(arg0);
            return result;
        }
    }
    async getActivityLog(): Promise<Array<ActivityLog>> {
        if (this.processError) {
            try {
                const result = await this.actor.getActivityLog();
                return result;
            } catch (e) {
                this.processError(e);
                throw new Error("unreachable");
            }
        } else {
            const result = await this.actor.getActivityLog();
            return result;
        }
    }
    async getBeritaDetail(arg0: ArtikelId): Promise<Artikel | null> {
        if (this.processError) {
            try {
                const result = await this.actor.getBeritaDetail(arg0);
                return from_candid_opt_n47(this._uploadFile, this._downloadFile, result);
            } catch (e) {
                this.processError(e);
                throw new Error("unreachable");
            }
        } else {
            const result = await this.actor.getBeritaDetail(arg0);
            return from_candid_opt_n47(this._uploadFile, this._downloadFile, result);
        }
    }
    async getCallerUserRole(): Promise<UserRole> {
        if (this.processError) {
            try {
                const result = await this.actor.getCallerUserRole();
                return from_candid_UserRole_n48(this._uploadFile, this._downloadFile, result);
            } catch (e) {
                this.processError(e);
                throw new Error("unreachable");
            }
        } else {
            const result = await this.actor.getCallerUserRole();
            return from_candid_UserRole_n48(this._uploadFile, this._downloadFile, result);
        }
    }
    async getDokumenDetail(arg0: DokumenId): Promise<Dokumen | null> {
        if (this.processError) {
            try {
                const result = await this.actor.getDokumenDetail(arg0);
                return from_candid_opt_n50(this._uploadFile, this._downloadFile, result);
            } catch (e) {
                this.processError(e);
                throw new Error("unreachable");
            }
        } else {
            const result = await this.actor.getDokumenDetail(arg0);
            return from_candid_opt_n50(this._uploadFile, this._downloadFile, result);
        }
    }
    async getGaleriItem(arg0: GaleriItemId): Promise<GaleriItem | null> {
        if (this.processError) {
            try {
                const result = await this.actor.getGaleriItem(arg0);
                return from_candid_opt_n51(this._uploadFile, this._downloadFile, result);
            } catch (e) {
                this.processError(e);
                throw new Error("unreachable");
            }
        } else {
            const result = await this.actor.getGaleriItem(arg0);
            return from_candid_opt_n51(this._uploadFile, this._downloadFile, result);
        }
    }
    async getMyPermissions(): Promise<MyPermissions> {
        if (this.processError) {
            try {
                const result = await this.actor.getMyPermissions();
                return from_candid_MyPermissions_n52(this._uploadFile, this._downloadFile, result);
            } catch (e) {
                this.processError(e);
                throw new Error("unreachable");
            }
        } else {
            const result = await this.actor.getMyPermissions();
            return from_candid_MyPermissions_n52(this._uploadFile, this._downloadFile, result);
        }
    }
    async getPendingReviewCount(): Promise<bigint> {
        if (this.processError) {
            try {
                const result = await this.actor.getPendingReviewCount();
                return result;
            } catch (e) {
                this.processError(e);
                throw new Error("unreachable");
            }
        } else {
            const result = await this.actor.getPendingReviewCount();
            return result;
        }
    }
    async getStatistik(): Promise<Statistik> {
        if (this.processError) {
            try {
                const result = await this.actor.getStatistik();
                return result;
            } catch (e) {
                this.processError(e);
                throw new Error("unreachable");
            }
        } else {
            const result = await this.actor.getStatistik();
            return result;
        }
    }
    async getTentangPage(arg0: TentangSlug): Promise<TentangPage | null> {
        if (this.processError) {
            try {
                const result = await this.actor.getTentangPage(to_candid_TentangSlug_n54(this._uploadFile, this._downloadFile, arg0));
                return from_candid_opt_n56(this._uploadFile, this._downloadFile, result);
            } catch (e) {
                this.processError(e);
                throw new Error("unreachable");
            }
        } else {
            const result = await this.actor.getTentangPage(to_candid_TentangSlug_n54(this._uploadFile, this._downloadFile, arg0));
            return from_candid_opt_n56(this._uploadFile, this._downloadFile, result);
        }
    }
    async isCallerAdmin(): Promise<boolean> {
        if (this.processError) {
            try {
                const result = await this.actor.isCallerAdmin();
                return result;
            } catch (e) {
                this.processError(e);
                throw new Error("unreachable");
            }
        } else {
            const result = await this.actor.isCallerAdmin();
            return result;
        }
    }
    async listAdminUsers(): Promise<Array<AdminUser>> {
        if (this.processError) {
            try {
                const result = await this.actor.listAdminUsers();
                return from_candid_vec_n66(this._uploadFile, this._downloadFile, result);
            } catch (e) {
                this.processError(e);
                throw new Error("unreachable");
            }
        } else {
            const result = await this.actor.listAdminUsers();
            return from_candid_vec_n66(this._uploadFile, this._downloadFile, result);
        }
    }
    async listBerita(): Promise<Array<Artikel>> {
        if (this.processError) {
            try {
                const result = await this.actor.listBerita();
                return from_candid_vec_n67(this._uploadFile, this._downloadFile, result);
            } catch (e) {
                this.processError(e);
                throw new Error("unreachable");
            }
        } else {
            const result = await this.actor.listBerita();
            return from_candid_vec_n67(this._uploadFile, this._downloadFile, result);
        }
    }
    async listDokumen(arg0: FilterDokumen): Promise<Array<Dokumen>> {
        if (this.processError) {
            try {
                const result = await this.actor.listDokumen(to_candid_FilterDokumen_n68(this._uploadFile, this._downloadFile, arg0));
                return from_candid_vec_n70(this._uploadFile, this._downloadFile, result);
            } catch (e) {
                this.processError(e);
                throw new Error("unreachable");
            }
        } else {
            const result = await this.actor.listDokumen(to_candid_FilterDokumen_n68(this._uploadFile, this._downloadFile, arg0));
            return from_candid_vec_n70(this._uploadFile, this._downloadFile, result);
        }
    }
    async listDokumenAdmin(arg0: FilterDokumen): Promise<Array<Dokumen>> {
        if (this.processError) {
            try {
                const result = await this.actor.listDokumenAdmin(to_candid_FilterDokumen_n68(this._uploadFile, this._downloadFile, arg0));
                return from_candid_vec_n70(this._uploadFile, this._downloadFile, result);
            } catch (e) {
                this.processError(e);
                throw new Error("unreachable");
            }
        } else {
            const result = await this.actor.listDokumenAdmin(to_candid_FilterDokumen_n68(this._uploadFile, this._downloadFile, arg0));
            return from_candid_vec_n70(this._uploadFile, this._downloadFile, result);
        }
    }
    async listGaleri(): Promise<Array<GaleriItem>> {
        if (this.processError) {
            try {
                const result = await this.actor.listGaleri();
                return from_candid_vec_n71(this._uploadFile, this._downloadFile, result);
            } catch (e) {
                this.processError(e);
                throw new Error("unreachable");
            }
        } else {
            const result = await this.actor.listGaleri();
            return from_candid_vec_n71(this._uploadFile, this._downloadFile, result);
        }
    }
    async listKategori(): Promise<Array<Kategori>> {
        if (this.processError) {
            try {
                const result = await this.actor.listKategori();
                return result;
            } catch (e) {
                this.processError(e);
                throw new Error("unreachable");
            }
        } else {
            const result = await this.actor.listKategori();
            return result;
        }
    }
    async listTentangPages(): Promise<Array<TentangPage>> {
        if (this.processError) {
            try {
                const result = await this.actor.listTentangPages();
                return from_candid_vec_n72(this._uploadFile, this._downloadFile, result);
            } catch (e) {
                this.processError(e);
                throw new Error("unreachable");
            }
        } else {
            const result = await this.actor.listTentangPages();
            return from_candid_vec_n72(this._uploadFile, this._downloadFile, result);
        }
    }
    async publishDokumen(arg0: DokumenId): Promise<WorkflowResult> {
        if (this.processError) {
            try {
                const result = await this.actor.publishDokumen(arg0);
                return from_candid_WorkflowResult_n73(this._uploadFile, this._downloadFile, result);
            } catch (e) {
                this.processError(e);
                throw new Error("unreachable");
            }
        } else {
            const result = await this.actor.publishDokumen(arg0);
            return from_candid_WorkflowResult_n73(this._uploadFile, this._downloadFile, result);
        }
    }
    async removeAdminUser(arg0: string): Promise<{
        __kind__: "ok";
        ok: null;
    } | {
        __kind__: "err";
        err: string;
    }> {
        if (this.processError) {
            try {
                const result = await this.actor.removeAdminUser(arg0);
                return from_candid_variant_n75(this._uploadFile, this._downloadFile, result);
            } catch (e) {
                this.processError(e);
                throw new Error("unreachable");
            }
        } else {
            const result = await this.actor.removeAdminUser(arg0);
            return from_candid_variant_n75(this._uploadFile, this._downloadFile, result);
        }
    }
    async returnToDraft(arg0: DokumenId, arg1: string): Promise<WorkflowResult> {
        if (this.processError) {
            try {
                const result = await this.actor.returnToDraft(arg0, arg1);
                return from_candid_WorkflowResult_n73(this._uploadFile, this._downloadFile, result);
            } catch (e) {
                this.processError(e);
                throw new Error("unreachable");
            }
        } else {
            const result = await this.actor.returnToDraft(arg0, arg1);
            return from_candid_WorkflowResult_n73(this._uploadFile, this._downloadFile, result);
        }
    }
    async submitForReview(arg0: DokumenId): Promise<WorkflowResult> {
        if (this.processError) {
            try {
                const result = await this.actor.submitForReview(arg0);
                return from_candid_WorkflowResult_n73(this._uploadFile, this._downloadFile, result);
            } catch (e) {
                this.processError(e);
                throw new Error("unreachable");
            }
        } else {
            const result = await this.actor.submitForReview(arg0);
            return from_candid_WorkflowResult_n73(this._uploadFile, this._downloadFile, result);
        }
    }
    async updateAdminUser(arg0: string, arg1: string | null, arg2: Array<MenuPermission>, arg3: boolean): Promise<{
        __kind__: "ok";
        ok: AdminUser;
    } | {
        __kind__: "err";
        err: string;
    }> {
        if (this.processError) {
            try {
                const result = await this.actor.updateAdminUser(arg0, to_candid_opt_n8(this._uploadFile, this._downloadFile, arg1), to_candid_vec_n9(this._uploadFile, this._downloadFile, arg2), arg3);
                return from_candid_variant_n12(this._uploadFile, this._downloadFile, result);
            } catch (e) {
                this.processError(e);
                throw new Error("unreachable");
            }
        } else {
            const result = await this.actor.updateAdminUser(arg0, to_candid_opt_n8(this._uploadFile, this._downloadFile, arg1), to_candid_vec_n9(this._uploadFile, this._downloadFile, arg2), arg3);
            return from_candid_variant_n12(this._uploadFile, this._downloadFile, result);
        }
    }
    async updateBerita(arg0: ArtikelId, arg1: ArtikelInput): Promise<Artikel | null> {
        if (this.processError) {
            try {
                const result = await this.actor.updateBerita(arg0, await to_candid_ArtikelInput_n19(this._uploadFile, this._downloadFile, arg1));
                return from_candid_opt_n47(this._uploadFile, this._downloadFile, result);
            } catch (e) {
                this.processError(e);
                throw new Error("unreachable");
            }
        } else {
            const result = await this.actor.updateBerita(arg0, await to_candid_ArtikelInput_n19(this._uploadFile, this._downloadFile, arg1));
            return from_candid_opt_n47(this._uploadFile, this._downloadFile, result);
        }
    }
    async updateDokumen(arg0: DokumenId, arg1: DokumenInput): Promise<Dokumen | null> {
        if (this.processError) {
            try {
                const result = await this.actor.updateDokumen(arg0, await to_candid_DokumenInput_n26(this._uploadFile, this._downloadFile, arg1));
                return from_candid_opt_n50(this._uploadFile, this._downloadFile, result);
            } catch (e) {
                this.processError(e);
                throw new Error("unreachable");
            }
        } else {
            const result = await this.actor.updateDokumen(arg0, await to_candid_DokumenInput_n26(this._uploadFile, this._downloadFile, arg1));
            return from_candid_opt_n50(this._uploadFile, this._downloadFile, result);
        }
    }
    async updateGaleriItem(arg0: GaleriItemId, arg1: GaleriItemInput): Promise<GaleriItem | null> {
        if (this.processError) {
            try {
                const result = await this.actor.updateGaleriItem(arg0, await to_candid_GaleriItemInput_n41(this._uploadFile, this._downloadFile, arg1));
                return from_candid_opt_n51(this._uploadFile, this._downloadFile, result);
            } catch (e) {
                this.processError(e);
                throw new Error("unreachable");
            }
        } else {
            const result = await this.actor.updateGaleriItem(arg0, await to_candid_GaleriItemInput_n41(this._uploadFile, this._downloadFile, arg1));
            return from_candid_opt_n51(this._uploadFile, this._downloadFile, result);
        }
    }
    async updateKategori(arg0: KategoriId, arg1: string): Promise<Kategori | null> {
        if (this.processError) {
            try {
                const result = await this.actor.updateKategori(arg0, arg1);
                return from_candid_opt_n76(this._uploadFile, this._downloadFile, result);
            } catch (e) {
                this.processError(e);
                throw new Error("unreachable");
            }
        } else {
            const result = await this.actor.updateKategori(arg0, arg1);
            return from_candid_opt_n76(this._uploadFile, this._downloadFile, result);
        }
    }
    async updateTentangPage(arg0: TentangSlug, arg1: TentangPageInput): Promise<TentangPage | null> {
        if (this.processError) {
            try {
                const result = await this.actor.updateTentangPage(to_candid_TentangSlug_n54(this._uploadFile, this._downloadFile, arg0), to_candid_TentangPageInput_n77(this._uploadFile, this._downloadFile, arg1));
                return from_candid_opt_n56(this._uploadFile, this._downloadFile, result);
            } catch (e) {
                this.processError(e);
                throw new Error("unreachable");
            }
        } else {
            const result = await this.actor.updateTentangPage(to_candid_TentangSlug_n54(this._uploadFile, this._downloadFile, arg0), to_candid_TentangPageInput_n77(this._uploadFile, this._downloadFile, arg1));
            return from_candid_opt_n56(this._uploadFile, this._downloadFile, result);
        }
    }
}
function from_candid_AdminUser_n13(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: _AdminUser): AdminUser {
    return from_candid_record_n14(_uploadFile, _downloadFile, value);
}
async function from_candid_Artikel_n22(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: _Artikel): Promise<Artikel> {
    return await from_candid_record_n23(_uploadFile, _downloadFile, value);
}
async function from_candid_Dokumen_n32(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: _Dokumen): Promise<Dokumen> {
    return await from_candid_record_n33(_uploadFile, _downloadFile, value);
}
async function from_candid_ExternalBlob_n25(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: _ExternalBlob): Promise<ExternalBlob> {
    return await _downloadFile(value);
}
async function from_candid_GaleriItem_n43(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: _GaleriItem): Promise<GaleriItem> {
    return await from_candid_record_n44(_uploadFile, _downloadFile, value);
}
function from_candid_MenuPermission_n16(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: _MenuPermission): MenuPermission {
    return from_candid_variant_n17(_uploadFile, _downloadFile, value);
}
function from_candid_MyPermissions_n52(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: _MyPermissions): MyPermissions {
    return from_candid_record_n53(_uploadFile, _downloadFile, value);
}
function from_candid_StatusDokumen_n34(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: _StatusDokumen): StatusDokumen {
    return from_candid_variant_n35(_uploadFile, _downloadFile, value);
}
function from_candid_TentangBlock_n64(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: _TentangBlock): TentangBlock {
    return from_candid_variant_n65(_uploadFile, _downloadFile, value);
}
function from_candid_TentangContent_n61(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: _TentangContent): TentangContent {
    return from_candid_record_n62(_uploadFile, _downloadFile, value);
}
function from_candid_TentangPage_n57(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: _TentangPage): TentangPage {
    return from_candid_record_n58(_uploadFile, _downloadFile, value);
}
function from_candid_TentangSlug_n59(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: _TentangSlug): TentangSlug {
    return from_candid_variant_n60(_uploadFile, _downloadFile, value);
}
function from_candid_UserRole_n48(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: _UserRole): UserRole {
    return from_candid_variant_n49(_uploadFile, _downloadFile, value);
}
function from_candid_WorkflowResult_n73(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: _WorkflowResult): WorkflowResult {
    return from_candid_variant_n74(_uploadFile, _downloadFile, value);
}
function from_candid_WorkflowStatus_n39(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: _WorkflowStatus): WorkflowStatus {
    return from_candid_variant_n40(_uploadFile, _downloadFile, value);
}
function from_candid__ImmutableObjectStorageRefillResult_n4(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: __ImmutableObjectStorageRefillResult): _ImmutableObjectStorageRefillResult {
    return from_candid_record_n5(_uploadFile, _downloadFile, value);
}
function from_candid_opt_n18(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: [] | [string]): string | null {
    return value.length === 0 ? null : value[0];
}
async function from_candid_opt_n24(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: [] | [_ExternalBlob]): Promise<ExternalBlob | null> {
    return value.length === 0 ? null : await from_candid_ExternalBlob_n25(_uploadFile, _downloadFile, value[0]);
}
function from_candid_opt_n36(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: [] | [_Timestamp]): Timestamp | null {
    return value.length === 0 ? null : value[0];
}
function from_candid_opt_n37(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: [] | [Principal]): Principal | null {
    return value.length === 0 ? null : value[0];
}
function from_candid_opt_n38(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: [] | [bigint]): bigint | null {
    return value.length === 0 ? null : value[0];
}
async function from_candid_opt_n47(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: [] | [_Artikel]): Promise<Artikel | null> {
    return value.length === 0 ? null : await from_candid_Artikel_n22(_uploadFile, _downloadFile, value[0]);
}
async function from_candid_opt_n50(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: [] | [_Dokumen]): Promise<Dokumen | null> {
    return value.length === 0 ? null : await from_candid_Dokumen_n32(_uploadFile, _downloadFile, value[0]);
}
async function from_candid_opt_n51(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: [] | [_GaleriItem]): Promise<GaleriItem | null> {
    return value.length === 0 ? null : await from_candid_GaleriItem_n43(_uploadFile, _downloadFile, value[0]);
}
function from_candid_opt_n56(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: [] | [_TentangPage]): TentangPage | null {
    return value.length === 0 ? null : from_candid_TentangPage_n57(_uploadFile, _downloadFile, value[0]);
}
function from_candid_opt_n6(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: [] | [boolean]): boolean | null {
    return value.length === 0 ? null : value[0];
}
function from_candid_opt_n7(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: [] | [bigint]): bigint | null {
    return value.length === 0 ? null : value[0];
}
function from_candid_opt_n76(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: [] | [_Kategori]): Kategori | null {
    return value.length === 0 ? null : value[0];
}
function from_candid_record_n14(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: {
    principal: string;
    grantedMenus: Array<_MenuPermission>;
    nama: [] | [string];
    isActive: boolean;
    addedAt: _Timestamp;
    addedBy: string;
}): {
    principal: string;
    grantedMenus: Array<MenuPermission>;
    nama?: string;
    isActive: boolean;
    addedAt: Timestamp;
    addedBy: string;
} {
    return {
        principal: value.principal,
        grantedMenus: from_candid_vec_n15(_uploadFile, _downloadFile, value.grantedMenus),
        nama: record_opt_to_undefined(from_candid_opt_n18(_uploadFile, _downloadFile, value.nama)),
        isActive: value.isActive,
        addedAt: value.addedAt,
        addedBy: value.addedBy
    };
}
async function from_candid_record_n23(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: {
    id: _ArtikelId;
    status: string;
    tanggal: _Timestamp;
    tags: Array<string>;
    ringkasan: string;
    gambar: [] | [_ExternalBlob];
    judul: string;
    author: string;
    tanggalPublikasi: bigint;
    konten: string;
}): Promise<{
    id: ArtikelId;
    status: string;
    tanggal: Timestamp;
    tags: Array<string>;
    ringkasan: string;
    gambar?: ExternalBlob;
    judul: string;
    author: string;
    tanggalPublikasi: bigint;
    konten: string;
}> {
    return {
        id: value.id,
        status: value.status,
        tanggal: value.tanggal,
        tags: value.tags,
        ringkasan: value.ringkasan,
        gambar: record_opt_to_undefined(await from_candid_opt_n24(_uploadFile, _downloadFile, value.gambar)),
        judul: value.judul,
        author: value.author,
        tanggalPublikasi: value.tanggalPublikasi,
        konten: value.konten
    };
}
async function from_candid_record_n33(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: {
    id: _DokumenId;
    tag: Array<string>;
    kategoriId: _KategoriId;
    status: _StatusDokumen;
    tahun: bigint;
    judul: string;
    relasiHukum: [] | [string];
    reviewedAt: [] | [_Timestamp];
    reviewedBy: [] | [Principal];
    catatanKoreksi: string;
    abstrak: string;
    tanggalPenetapan: string;
    nomor: string;
    downloadCount: bigint;
    tanggalPengundangan: [] | [bigint];
    workflowStatus: _WorkflowStatus;
    filePdf: _ExternalBlob;
}): Promise<{
    id: DokumenId;
    tag: Array<string>;
    kategoriId: KategoriId;
    status: StatusDokumen;
    tahun: bigint;
    judul: string;
    relasiHukum?: string;
    reviewedAt?: Timestamp;
    reviewedBy?: Principal;
    catatanKoreksi: string;
    abstrak: string;
    tanggalPenetapan: string;
    nomor: string;
    downloadCount: bigint;
    tanggalPengundangan?: bigint;
    workflowStatus: WorkflowStatus;
    filePdf: ExternalBlob;
}> {
    return {
        id: value.id,
        tag: value.tag,
        kategoriId: value.kategoriId,
        status: from_candid_StatusDokumen_n34(_uploadFile, _downloadFile, value.status),
        tahun: value.tahun,
        judul: value.judul,
        relasiHukum: record_opt_to_undefined(from_candid_opt_n18(_uploadFile, _downloadFile, value.relasiHukum)),
        reviewedAt: record_opt_to_undefined(from_candid_opt_n36(_uploadFile, _downloadFile, value.reviewedAt)),
        reviewedBy: record_opt_to_undefined(from_candid_opt_n37(_uploadFile, _downloadFile, value.reviewedBy)),
        catatanKoreksi: value.catatanKoreksi,
        abstrak: value.abstrak,
        tanggalPenetapan: value.tanggalPenetapan,
        nomor: value.nomor,
        downloadCount: value.downloadCount,
        tanggalPengundangan: record_opt_to_undefined(from_candid_opt_n38(_uploadFile, _downloadFile, value.tanggalPengundangan)),
        workflowStatus: from_candid_WorkflowStatus_n39(_uploadFile, _downloadFile, value.workflowStatus),
        filePdf: await from_candid_ExternalBlob_n25(_uploadFile, _downloadFile, value.filePdf)
    };
}
async function from_candid_record_n44(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: {
    id: _GaleriItemId;
    album: string;
    createdAt: _Timestamp;
    createdBy: string;
    gambar: [] | [_ExternalBlob];
    judul: string;
    deskripsi: string;
}): Promise<{
    id: GaleriItemId;
    album: string;
    createdAt: Timestamp;
    createdBy: string;
    gambar?: ExternalBlob;
    judul: string;
    deskripsi: string;
}> {
    return {
        id: value.id,
        album: value.album,
        createdAt: value.createdAt,
        createdBy: value.createdBy,
        gambar: record_opt_to_undefined(await from_candid_opt_n24(_uploadFile, _downloadFile, value.gambar)),
        judul: value.judul,
        deskripsi: value.deskripsi
    };
}
function from_candid_record_n5(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: {
    success: [] | [boolean];
    topped_up_amount: [] | [bigint];
}): {
    success?: boolean;
    topped_up_amount?: bigint;
} {
    return {
        success: record_opt_to_undefined(from_candid_opt_n6(_uploadFile, _downloadFile, value.success)),
        topped_up_amount: record_opt_to_undefined(from_candid_opt_n7(_uploadFile, _downloadFile, value.topped_up_amount))
    };
}
function from_candid_record_n53(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: {
    grantedMenus: Array<_MenuPermission>;
    isAdmin: boolean;
}): {
    grantedMenus: Array<MenuPermission>;
    isAdmin: boolean;
} {
    return {
        grantedMenus: from_candid_vec_n15(_uploadFile, _downloadFile, value.grantedMenus),
        isAdmin: value.isAdmin
    };
}
function from_candid_record_n58(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: {
    id: bigint;
    slug: _TentangSlug;
    judul: string;
    updatedAt: _Timestamp;
    updatedBy: string;
    konten: _TentangContent;
}): {
    id: bigint;
    slug: TentangSlug;
    judul: string;
    updatedAt: Timestamp;
    updatedBy: string;
    konten: TentangContent;
} {
    return {
        id: value.id,
        slug: from_candid_TentangSlug_n59(_uploadFile, _downloadFile, value.slug),
        judul: value.judul,
        updatedAt: value.updatedAt,
        updatedBy: value.updatedBy,
        konten: from_candid_TentangContent_n61(_uploadFile, _downloadFile, value.konten)
    };
}
function from_candid_record_n62(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: {
    blocks: Array<_TentangBlock>;
}): {
    blocks: Array<TentangBlock>;
} {
    return {
        blocks: from_candid_vec_n63(_uploadFile, _downloadFile, value.blocks)
    };
}
function from_candid_variant_n12(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: {
    ok: _AdminUser;
} | {
    err: string;
}): {
    __kind__: "ok";
    ok: AdminUser;
} | {
    __kind__: "err";
    err: string;
} {
    return "ok" in value ? {
        __kind__: "ok",
        ok: from_candid_AdminUser_n13(_uploadFile, _downloadFile, value.ok)
    } : "err" in value ? {
        __kind__: "err",
        err: value.err
    } : value;
}
function from_candid_variant_n17(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: {
    galeri: null;
} | {
    userManagement: null;
} | {
    kategori: null;
} | {
    dokumen: null;
} | {
    statistik: null;
} | {
    berita: null;
} | {
    tentang: null;
}): MenuPermission {
    return "galeri" in value ? MenuPermission.galeri : "userManagement" in value ? MenuPermission.userManagement : "kategori" in value ? MenuPermission.kategori : "dokumen" in value ? MenuPermission.dokumen : "statistik" in value ? MenuPermission.statistik : "berita" in value ? MenuPermission.berita : "tentang" in value ? MenuPermission.tentang : value;
}
function from_candid_variant_n35(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: {
    Berlaku: null;
} | {
    Dicabut: null;
} | {
    TidakBerlaku: null;
}): StatusDokumen {
    return "Berlaku" in value ? StatusDokumen.Berlaku : "Dicabut" in value ? StatusDokumen.Dicabut : "TidakBerlaku" in value ? StatusDokumen.TidakBerlaku : value;
}
function from_candid_variant_n40(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: {
    PendingReview: null;
} | {
    Draft: null;
} | {
    Published: null;
}): WorkflowStatus {
    return "PendingReview" in value ? WorkflowStatus.PendingReview : "Draft" in value ? WorkflowStatus.Draft : "Published" in value ? WorkflowStatus.Published : value;
}
function from_candid_variant_n49(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: {
    admin: null;
} | {
    user: null;
} | {
    guest: null;
}): UserRole {
    return "admin" in value ? UserRole.admin : "user" in value ? UserRole.user : "guest" in value ? UserRole.guest : value;
}
function from_candid_variant_n60(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: {
    visiMisi: null;
} | {
    sejarah: null;
} | {
    fungsi: null;
} | {
    struktur: null;
} | {
    dasarHukum: null;
}): TentangSlug {
    return "visiMisi" in value ? TentangSlug.visiMisi : "sejarah" in value ? TentangSlug.sejarah : "fungsi" in value ? TentangSlug.fungsi : "struktur" in value ? TentangSlug.struktur : "dasarHukum" in value ? TentangSlug.dasarHukum : value;
}
function from_candid_variant_n65(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: {
    daftarItem: Array<string>;
} | {
    paragraf: string;
}): {
    __kind__: "daftarItem";
    daftarItem: Array<string>;
} | {
    __kind__: "paragraf";
    paragraf: string;
} {
    return "daftarItem" in value ? {
        __kind__: "daftarItem",
        daftarItem: value.daftarItem
    } : "paragraf" in value ? {
        __kind__: "paragraf",
        paragraf: value.paragraf
    } : value;
}
function from_candid_variant_n74(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: {
    ok: string;
} | {
    err: string;
}): {
    __kind__: "ok";
    ok: string;
} | {
    __kind__: "err";
    err: string;
} {
    return "ok" in value ? {
        __kind__: "ok",
        ok: value.ok
    } : "err" in value ? {
        __kind__: "err",
        err: value.err
    } : value;
}
function from_candid_variant_n75(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: {
    ok: null;
} | {
    err: string;
}): {
    __kind__: "ok";
    ok: null;
} | {
    __kind__: "err";
    err: string;
} {
    return "ok" in value ? {
        __kind__: "ok",
        ok: value.ok
    } : "err" in value ? {
        __kind__: "err",
        err: value.err
    } : value;
}
function from_candid_vec_n15(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: Array<_MenuPermission>): Array<MenuPermission> {
    return value.map((x)=>from_candid_MenuPermission_n16(_uploadFile, _downloadFile, x));
}
function from_candid_vec_n63(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: Array<_TentangBlock>): Array<TentangBlock> {
    return value.map((x)=>from_candid_TentangBlock_n64(_uploadFile, _downloadFile, x));
}
function from_candid_vec_n66(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: Array<_AdminUser>): Array<AdminUser> {
    return value.map((x)=>from_candid_AdminUser_n13(_uploadFile, _downloadFile, x));
}
async function from_candid_vec_n67(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: Array<_Artikel>): Promise<Array<Artikel>> {
    return await Promise.all(value.map(async (x)=>await from_candid_Artikel_n22(_uploadFile, _downloadFile, x)));
}
async function from_candid_vec_n70(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: Array<_Dokumen>): Promise<Array<Dokumen>> {
    return await Promise.all(value.map(async (x)=>await from_candid_Dokumen_n32(_uploadFile, _downloadFile, x)));
}
async function from_candid_vec_n71(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: Array<_GaleriItem>): Promise<Array<GaleriItem>> {
    return await Promise.all(value.map(async (x)=>await from_candid_GaleriItem_n43(_uploadFile, _downloadFile, x)));
}
function from_candid_vec_n72(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: Array<_TentangPage>): Array<TentangPage> {
    return value.map((x)=>from_candid_TentangPage_n57(_uploadFile, _downloadFile, x));
}
async function to_candid_ArtikelInput_n19(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: ArtikelInput): Promise<_ArtikelInput> {
    return await to_candid_record_n20(_uploadFile, _downloadFile, value);
}
async function to_candid_DokumenInput_n26(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: DokumenInput): Promise<_DokumenInput> {
    return await to_candid_record_n27(_uploadFile, _downloadFile, value);
}
async function to_candid_ExternalBlob_n21(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: ExternalBlob): Promise<_ExternalBlob> {
    return await _uploadFile(value);
}
function to_candid_FilterDokumen_n68(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: FilterDokumen): _FilterDokumen {
    return to_candid_record_n69(_uploadFile, _downloadFile, value);
}
async function to_candid_GaleriItemInput_n41(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: GaleriItemInput): Promise<_GaleriItemInput> {
    return await to_candid_record_n42(_uploadFile, _downloadFile, value);
}
function to_candid_MenuPermission_n10(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: MenuPermission): _MenuPermission {
    return to_candid_variant_n11(_uploadFile, _downloadFile, value);
}
function to_candid_StatusDokumen_n28(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: StatusDokumen): _StatusDokumen {
    return to_candid_variant_n29(_uploadFile, _downloadFile, value);
}
function to_candid_TentangBlock_n82(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: TentangBlock): _TentangBlock {
    return to_candid_variant_n83(_uploadFile, _downloadFile, value);
}
function to_candid_TentangContent_n79(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: TentangContent): _TentangContent {
    return to_candid_record_n80(_uploadFile, _downloadFile, value);
}
function to_candid_TentangPageInput_n77(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: TentangPageInput): _TentangPageInput {
    return to_candid_record_n78(_uploadFile, _downloadFile, value);
}
function to_candid_TentangSlug_n54(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: TentangSlug): _TentangSlug {
    return to_candid_variant_n55(_uploadFile, _downloadFile, value);
}
function to_candid_UserRole_n45(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: UserRole): _UserRole {
    return to_candid_variant_n46(_uploadFile, _downloadFile, value);
}
function to_candid_WorkflowStatus_n30(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: WorkflowStatus): _WorkflowStatus {
    return to_candid_variant_n31(_uploadFile, _downloadFile, value);
}
function to_candid__ImmutableObjectStorageRefillInformation_n2(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: _ImmutableObjectStorageRefillInformation): __ImmutableObjectStorageRefillInformation {
    return to_candid_record_n3(_uploadFile, _downloadFile, value);
}
function to_candid_opt_n1(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: _ImmutableObjectStorageRefillInformation | null): [] | [__ImmutableObjectStorageRefillInformation] {
    return value === null ? candid_none() : candid_some(to_candid__ImmutableObjectStorageRefillInformation_n2(_uploadFile, _downloadFile, value));
}
function to_candid_opt_n8(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: string | null): [] | [string] {
    return value === null ? candid_none() : candid_some(value);
}
async function to_candid_record_n20(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: {
    status: string;
    tags: Array<string>;
    ringkasan: string;
    gambar?: ExternalBlob;
    judul: string;
    author: string;
    tanggalPublikasi?: bigint;
    konten: string;
}): Promise<{
    status: string;
    tags: Array<string>;
    ringkasan: string;
    gambar: [] | [_ExternalBlob];
    judul: string;
    author: string;
    tanggalPublikasi: [] | [bigint];
    konten: string;
}> {
    return {
        status: value.status,
        tags: value.tags,
        ringkasan: value.ringkasan,
        gambar: value.gambar ? candid_some(await to_candid_ExternalBlob_n21(_uploadFile, _downloadFile, value.gambar)) : candid_none(),
        judul: value.judul,
        author: value.author,
        tanggalPublikasi: value.tanggalPublikasi ? candid_some(value.tanggalPublikasi) : candid_none(),
        konten: value.konten
    };
}
async function to_candid_record_n27(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: {
    tag: Array<string>;
    kategoriId: KategoriId;
    status: StatusDokumen;
    tahun: bigint;
    judul: string;
    relasiHukum?: string;
    abstrak: string;
    tanggalPenetapan: string;
    nomor: string;
    tanggalPengundangan?: bigint;
    workflowStatus?: WorkflowStatus;
    filePdf: ExternalBlob;
}): Promise<{
    tag: Array<string>;
    kategoriId: _KategoriId;
    status: _StatusDokumen;
    tahun: bigint;
    judul: string;
    relasiHukum: [] | [string];
    abstrak: string;
    tanggalPenetapan: string;
    nomor: string;
    tanggalPengundangan: [] | [bigint];
    workflowStatus: [] | [_WorkflowStatus];
    filePdf: _ExternalBlob;
}> {
    return {
        tag: value.tag,
        kategoriId: value.kategoriId,
        status: to_candid_StatusDokumen_n28(_uploadFile, _downloadFile, value.status),
        tahun: value.tahun,
        judul: value.judul,
        relasiHukum: value.relasiHukum ? candid_some(value.relasiHukum) : candid_none(),
        abstrak: value.abstrak,
        tanggalPenetapan: value.tanggalPenetapan,
        nomor: value.nomor,
        tanggalPengundangan: value.tanggalPengundangan ? candid_some(value.tanggalPengundangan) : candid_none(),
        workflowStatus: value.workflowStatus ? candid_some(to_candid_WorkflowStatus_n30(_uploadFile, _downloadFile, value.workflowStatus)) : candid_none(),
        filePdf: await to_candid_ExternalBlob_n21(_uploadFile, _downloadFile, value.filePdf)
    };
}
function to_candid_record_n3(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: {
    proposed_top_up_amount?: bigint;
}): {
    proposed_top_up_amount: [] | [bigint];
} {
    return {
        proposed_top_up_amount: value.proposed_top_up_amount ? candid_some(value.proposed_top_up_amount) : candid_none()
    };
}
async function to_candid_record_n42(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: {
    album: string;
    gambar?: ExternalBlob;
    judul: string;
    deskripsi: string;
}): Promise<{
    album: string;
    gambar: [] | [_ExternalBlob];
    judul: string;
    deskripsi: string;
}> {
    return {
        album: value.album,
        gambar: value.gambar ? candid_some(await to_candid_ExternalBlob_n21(_uploadFile, _downloadFile, value.gambar)) : candid_none(),
        judul: value.judul,
        deskripsi: value.deskripsi
    };
}
function to_candid_record_n69(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: {
    kategoriId?: KategoriId;
    status?: StatusDokumen;
    tahun?: bigint;
    katakunci?: string;
    workflowStatus?: WorkflowStatus;
}): {
    kategoriId: [] | [_KategoriId];
    status: [] | [_StatusDokumen];
    tahun: [] | [bigint];
    katakunci: [] | [string];
    workflowStatus: [] | [_WorkflowStatus];
} {
    return {
        kategoriId: value.kategoriId ? candid_some(value.kategoriId) : candid_none(),
        status: value.status ? candid_some(to_candid_StatusDokumen_n28(_uploadFile, _downloadFile, value.status)) : candid_none(),
        tahun: value.tahun ? candid_some(value.tahun) : candid_none(),
        katakunci: value.katakunci ? candid_some(value.katakunci) : candid_none(),
        workflowStatus: value.workflowStatus ? candid_some(to_candid_WorkflowStatus_n30(_uploadFile, _downloadFile, value.workflowStatus)) : candid_none()
    };
}
function to_candid_record_n78(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: {
    judul: string;
    konten: TentangContent;
}): {
    judul: string;
    konten: _TentangContent;
} {
    return {
        judul: value.judul,
        konten: to_candid_TentangContent_n79(_uploadFile, _downloadFile, value.konten)
    };
}
function to_candid_record_n80(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: {
    blocks: Array<TentangBlock>;
}): {
    blocks: Array<_TentangBlock>;
} {
    return {
        blocks: to_candid_vec_n81(_uploadFile, _downloadFile, value.blocks)
    };
}
function to_candid_variant_n11(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: MenuPermission): {
    galeri: null;
} | {
    userManagement: null;
} | {
    kategori: null;
} | {
    dokumen: null;
} | {
    statistik: null;
} | {
    berita: null;
} | {
    tentang: null;
} {
    return value == MenuPermission.galeri ? {
        galeri: null
    } : value == MenuPermission.userManagement ? {
        userManagement: null
    } : value == MenuPermission.kategori ? {
        kategori: null
    } : value == MenuPermission.dokumen ? {
        dokumen: null
    } : value == MenuPermission.statistik ? {
        statistik: null
    } : value == MenuPermission.berita ? {
        berita: null
    } : value == MenuPermission.tentang ? {
        tentang: null
    } : value;
}
function to_candid_variant_n29(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: StatusDokumen): {
    Berlaku: null;
} | {
    Dicabut: null;
} | {
    TidakBerlaku: null;
} {
    return value == StatusDokumen.Berlaku ? {
        Berlaku: null
    } : value == StatusDokumen.Dicabut ? {
        Dicabut: null
    } : value == StatusDokumen.TidakBerlaku ? {
        TidakBerlaku: null
    } : value;
}
function to_candid_variant_n31(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: WorkflowStatus): {
    PendingReview: null;
} | {
    Draft: null;
} | {
    Published: null;
} {
    return value == WorkflowStatus.PendingReview ? {
        PendingReview: null
    } : value == WorkflowStatus.Draft ? {
        Draft: null
    } : value == WorkflowStatus.Published ? {
        Published: null
    } : value;
}
function to_candid_variant_n46(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: UserRole): {
    admin: null;
} | {
    user: null;
} | {
    guest: null;
} {
    return value == UserRole.admin ? {
        admin: null
    } : value == UserRole.user ? {
        user: null
    } : value == UserRole.guest ? {
        guest: null
    } : value;
}
function to_candid_variant_n55(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: TentangSlug): {
    visiMisi: null;
} | {
    sejarah: null;
} | {
    fungsi: null;
} | {
    struktur: null;
} | {
    dasarHukum: null;
} {
    return value == TentangSlug.visiMisi ? {
        visiMisi: null
    } : value == TentangSlug.sejarah ? {
        sejarah: null
    } : value == TentangSlug.fungsi ? {
        fungsi: null
    } : value == TentangSlug.struktur ? {
        struktur: null
    } : value == TentangSlug.dasarHukum ? {
        dasarHukum: null
    } : value;
}
function to_candid_variant_n83(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: {
    __kind__: "daftarItem";
    daftarItem: Array<string>;
} | {
    __kind__: "paragraf";
    paragraf: string;
}): {
    daftarItem: Array<string>;
} | {
    paragraf: string;
} {
    return value.__kind__ === "daftarItem" ? {
        daftarItem: value.daftarItem
    } : value.__kind__ === "paragraf" ? {
        paragraf: value.paragraf
    } : value;
}
function to_candid_vec_n81(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: Array<TentangBlock>): Array<_TentangBlock> {
    return value.map((x)=>to_candid_TentangBlock_n82(_uploadFile, _downloadFile, x));
}
function to_candid_vec_n9(_uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, value: Array<MenuPermission>): Array<_MenuPermission> {
    return value.map((x)=>to_candid_MenuPermission_n10(_uploadFile, _downloadFile, x));
}
export interface CreateActorOptions {
    agent?: Agent;
    agentOptions?: HttpAgentOptions;
    actorOptions?: ActorConfig;
    processError?: (error: unknown) => never;
}
export function createActor(canisterId: string, _uploadFile: (file: ExternalBlob) => Promise<Uint8Array>, _downloadFile: (file: Uint8Array) => Promise<ExternalBlob>, options: CreateActorOptions = {}): Backend {
    const agent = options.agent || HttpAgent.createSync({
        ...options.agentOptions
    });
    if (options.agent && options.agentOptions) {
        console.warn("Detected both agent and agentOptions passed to createActor. Ignoring agentOptions and proceeding with the provided agent.");
    }
    const actor = Actor.createActor<_SERVICE>(idlFactory, {
        agent,
        canisterId: canisterId,
        ...options.actorOptions
    });
    return new Backend(actor, _uploadFile, _downloadFile, options.processError);
}
