// @caffeine/template-frontend/dist/assets/AdminTentang-D17vPKz9.js
import { c as createLucideIcon, r as reactExports, j as jsxRuntimeExports, B as BookOpen, v as LoaderCircle } from "./index-B7-RSq-m.js";
import { u as ue } from "./index-B0YpMaU8.js";
import { p as useTentangPages, q as useUpdateTentangPage } from "./useBackend-xFEMRG87.js";
import { C as CircleAlert } from "./circle-alert-D-S0YTKb.js";
import { C as Clock } from "./clock-CMHdisaV.js";
import { C as CircleCheck } from "./circle-check-H4K-d60y.js";
import { P as Plus } from "./plus-C7LHpCDy.js";
import { T as Trash2 } from "./trash-2-CnhMEsa2.js";
import "./useMutation-D--BDafu.js";
/**
 * @license lucide-react v0.511.0 - ISC
 *
 * This source code is licensed under the ISC license.
 * See the LICENSE file in the root directory of this source tree.
 */
const __iconNode = [
  [
    "path",
    {
      d: "M15.2 3a2 2 0 0 1 1.4.6l3.8 3.8a2 2 0 0 1 .6 1.4V19a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2z",
      key: "1c8476"
    }
  ],
  ["path", { d: "M17 21v-7a1 1 0 0 0-1-1H8a1 1 0 0 0-1 1v7", key: "1ydtos" }],
  ["path", { d: "M7 3v4a1 1 0 0 0 1 1h7", key: "t51u73" }]
];
const Save = createLucideIcon("save", __iconNode);
const TABS = [
  { slug: "sejarah", label: "Sejarah" },
  { slug: "visiMisi", label: "Visi & Misi" },
  { slug: "dasarHukum", label: "Dasar Hukum" },
  { slug: "fungsi", label: "Fungsi JDIH" },
  { slug: "struktur", label: "Struktur Organisasi" }
];
const DEFAULT_CONTENT = {
  sejarah: [
    {
      __kind__: "paragraf",
      paragraf: "JDIH (Jaringan Dokumentasi dan Informasi Hukum) Kampus Universitas 17 Agustus 1945 Banyuwangi dibentuk sebagai bagian dari upaya universitas dalam mendukung sistem dokumentasi hukum nasional yang diatur melalui Peraturan Presiden Nomor 33 Tahun 2012 tentang Jaringan Dokumentasi dan Informasi Hukum Nasional."
    },
    {
      __kind__: "paragraf",
      paragraf: "Pembentukan JDIH UNTAG Banyuwangi juga dilandasi oleh Peraturan Menteri Hukum dan HAM Nomor 8 Tahun 2019 tentang Standar Pengelolaan Dokumen dan Informasi Hukum."
    },
    {
      __kind__: "paragraf",
      paragraf: "Portal ini hadir sebagai wujud komitmen UNTAG Banyuwangi terhadap transparansi tata kelola hukum dan kemudahan akses informasi regulasi bagi seluruh civitas akademika."
    },
    {
      __kind__: "paragraf",
      paragraf: "Sejak didirikan, JDIH UNTAG Banyuwangi berkomitmen untuk terus memperbarui dan memperluas koleksi dokumen hukum agar seluruh civitas akademika dapat mengakses regulasi kampus secara mudah, cepat, dan gratis."
    }
  ],
  visiMisi: [
    {
      __kind__: "paragraf",
      paragraf: "Menjadi pusat dokumentasi dan informasi hukum kampus yang terpercaya, mudah diakses, dan bermanfaat bagi civitas akademika Universitas 17 Agustus 1945 Banyuwangi."
    },
    {
      __kind__: "daftarItem",
      daftarItem: [
        "Membangun sistem dokumentasi hukum kampus yang terstruktur, lengkap, dan mudah diakses",
        "Melakukan pengelolaan dan pemutakhiran data produk hukum secara berkala dan konsisten",
        "Menyebarluaskan informasi hukum kepada seluruh civitas akademika UNTAG Banyuwangi",
        "Mendukung penegakan hukum dan tata kelola universitas yang transparan dan akuntabel",
        "Menjalin koordinasi dan sinkronisasi dengan JDIHN pusat dan jaringan JDIH perguruan tinggi"
      ]
    }
  ],
  dasarHukum: [
    {
      __kind__: "daftarItem",
      daftarItem: [
        "Peraturan Presiden Nomor 33 Tahun 2012 tentang Jaringan Dokumentasi dan Informasi Hukum Nasional",
        "Peraturan Menteri Hukum dan HAM Nomor 8 Tahun 2019 tentang Standar Pengelolaan Dokumen dan Informasi Hukum",
        "Undang-Undang Nomor 12 Tahun 2011 tentang Pembentukan Peraturan Perundang-undangan",
        "Statuta Universitas 17 Agustus 1945 Banyuwangi",
        "Keputusan Rektor UNTAG Banyuwangi tentang Pembentukan JDIH Kampus"
      ]
    }
  ],
  fungsi: [
    {
      __kind__: "daftarItem",
      daftarItem: [
        "Mendokumentasikan seluruh produk hukum yang diterbitkan oleh Universitas 17 Agustus 1945 Banyuwangi",
        "Menyediakan layanan informasi hukum yang mudah diakses oleh civitas akademika dan masyarakat umum",
        "Menjaga ketertiban, kelengkapan, dan kemutakhiran data produk hukum kampus",
        "Mendukung integrasi dan sinkronisasi data dengan Jaringan Dokumentasi dan Informasi Hukum Nasional (JDIHN)",
        "Meningkatkan transparansi dan akuntabilitas tata kelola hukum universitas",
        "Menyediakan sarana publikasi dan sosialisasi regulasi internal kampus"
      ]
    }
  ],
  struktur: [
    {
      __kind__: "daftarItem",
      daftarItem: [
        "Jabatan: Penanggung Jawab | Nama: Prof. Dr. H. Teguh Sulistyo, S.H., M.H. | Unit: Rektor UNTAG Banyuwangi",
        "Jabatan: Ketua JDIH | Nama: Dr. Hj. Sri Wahyuni, S.H., M.H. | Unit: Wakil Rektor Bidang Akademik",
        "Jabatan: Koordinator Teknis | Nama: Agus Prasetyo, S.H., M.H. | Unit: Kepala Biro Hukum & Kerjasama",
        "Jabatan: Pengelola Dokumen | Nama: Dewi Rahmawati, S.H. | Unit: Staf Biro Hukum",
        "Jabatan: Anggota | Nama: Rian Hidayat, S.Kom. | Unit: UPT Teknologi Informasi",
        "Jabatan: Anggota | Nama: Siti Nurhayati, A.Md. | Unit: Staf Biro Akademik"
      ]
    }
  ]
};
const DEFAULT_TITLES = {
  sejarah: "Sejarah & Latar Belakang",
  visiMisi: "Visi & Misi",
  dasarHukum: "Dasar Hukum",
  fungsi: "Fungsi JDIH",
  struktur: "Struktur Organisasi"
};
function formatUpdatedAt(ts) {
  const ms = typeof ts === "bigint" ? Number(ts / 1000000n) : typeof ts === "string" ? new Date(ts).getTime() : ts;
  return new Date(ms).toLocaleString("id-ID", {
    day: "2-digit",
    month: "long",
    year: "numeric",
    hour: "2-digit",
    minute: "2-digit"
  });
}
function truncatePrincipal(p) {
  if (!p) return "—";
  if (p.length <= 20) return p;
  return `${p.slice(0, 8)}...${p.slice(-6)}`;
}
function SejarahEditor({ blocks, onChange }) {
  const paragraphs = blocks.filter((b) => b.__kind__ === "paragraf").map((b) => b.paragraf);
  const setParagraph = (idx, val) => {
    const updated = [...paragraphs];
    updated[idx] = val;
    onChange(
      updated.map((p) => ({ __kind__: "paragraf", paragraf: p }))
    );
  };
  return /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "space-y-4", children: paragraphs.map((p, idx) => (
    // biome-ignore lint/suspicious/noArrayIndexKey: ordered paragraphs
    /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { children: [
      /* @__PURE__ */ jsxRuntimeExports.jsxs(
        "label",
        {
          htmlFor: `sejarah-paragraf-${idx}`,
          className: "block text-xs font-semibold text-foreground mb-1",
          children: [
            "Paragraf ",
            idx + 1
          ]
        }
      ),
      /* @__PURE__ */ jsxRuntimeExports.jsx(
        "textarea",
        {
          id: `sejarah-paragraf-${idx}`,
          className: "w-full border border-input rounded-lg px-3 py-2 text-sm text-foreground bg-background resize-none focus:outline-none focus:ring-2 focus:ring-red-500",
          rows: 4,
          value: p,
          onChange: (e) => setParagraph(idx, e.target.value),
          "data-ocid": `admin_tentang.sejarah.paragraf.${idx + 1}`
        }
      )
    ] }, idx)
  )) });
}
function VisiMisiEditor({ blocks, onChange }) {
  const visiBlock = blocks.find((b) => b.__kind__ === "paragraf");
  const misiBlock = blocks.find((b) => b.__kind__ === "daftarItem");
  const visi = (visiBlock == null ? void 0 : visiBlock.paragraf) ?? "";
  const misi = (misiBlock == null ? void 0 : misiBlock.daftarItem) ?? [];
  const update = (newVisi, newMisi) => {
    onChange([
      { __kind__: "paragraf", paragraf: newVisi },
      { __kind__: "daftarItem", daftarItem: newMisi }
    ]);
  };
  const setMisiItem = (idx, val) => {
    const updated = [...misi];
    updated[idx] = val;
    update(visi, updated);
  };
  const addMisi = () => update(visi, [...misi, ""]);
  const removeMisi = (idx) => update(
    visi,
    misi.filter((_, i) => i !== idx)
  );
  return /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "space-y-6", children: [
    /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { children: [
      /* @__PURE__ */ jsxRuntimeExports.jsx(
        "label",
        {
          htmlFor: "visi-input",
          className: "block text-xs font-semibold text-foreground mb-1",
          children: "Visi"
        }
      ),
      /* @__PURE__ */ jsxRuntimeExports.jsx(
        "textarea",
        {
          id: "visi-input",
          className: "w-full border border-input rounded-lg px-3 py-2 text-sm text-foreground bg-background resize-none focus:outline-none focus:ring-2 focus:ring-red-500",
          rows: 3,
          value: visi,
          onChange: (e) => update(e.target.value, misi),
          "data-ocid": "admin_tentang.visi_misi.visi"
        }
      )
    ] }),
    /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { children: [
      /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex items-center justify-between mb-2", children: [
        /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "block text-xs font-semibold text-foreground", children: "Misi" }),
        /* @__PURE__ */ jsxRuntimeExports.jsxs(
          "button",
          {
            type: "button",
            onClick: addMisi,
            className: "flex items-center gap-1 text-xs text-red-600 hover:text-red-700 font-medium",
            "data-ocid": "admin_tentang.visi_misi.add_misi_button",
            children: [
              /* @__PURE__ */ jsxRuntimeExports.jsx(Plus, { size: 14 }),
              " Tambah Misi"
            ]
          }
        )
      ] }),
      /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "space-y-2", children: misi.map((item, idx) => (
        // biome-ignore lint/suspicious/noArrayIndexKey: ordered list
        /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex items-start gap-2", children: [
          /* @__PURE__ */ jsxRuntimeExports.jsxs("span", { className: "text-xs text-muted-foreground mt-2.5 w-5 flex-shrink-0", children: [
            idx + 1,
            "."
          ] }),
          /* @__PURE__ */ jsxRuntimeExports.jsx(
            "input",
            {
              className: "flex-1 border border-input rounded-lg px-3 py-2 text-sm text-foreground bg-background focus:outline-none focus:ring-2 focus:ring-red-500",
              value: item,
              onChange: (e) => setMisiItem(idx, e.target.value),
              placeholder: `Misi ${idx + 1}`,
              "data-ocid": `admin_tentang.visi_misi.misi.${idx + 1}`
            }
          ),
          /* @__PURE__ */ jsxRuntimeExports.jsx(
            "button",
            {
              type: "button",
              onClick: () => removeMisi(idx),
              className: "mt-2 text-muted-foreground hover:text-red-600 transition-colors",
              "aria-label": "Hapus misi",
              "data-ocid": `admin_tentang.visi_misi.delete_misi_button.${idx + 1}`,
              children: /* @__PURE__ */ jsxRuntimeExports.jsx(Trash2, { size: 16 })
            }
          )
        ] }, idx)
      )) })
    ] })
  ] });
}
function DaftarEditor({
  blocks,
  onChange,
  addLabel,
  placeholder,
  ocidPrefix
}) {
  const daftarBlock = blocks.find((b) => b.__kind__ === "daftarItem");
  const items = (daftarBlock == null ? void 0 : daftarBlock.daftarItem) ?? [];
  const update = (newItems) => {
    onChange([{ __kind__: "daftarItem", daftarItem: newItems }]);
  };
  return /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "space-y-3", children: [
    /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex items-center justify-between", children: [
      /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "text-xs font-semibold text-foreground", children: "Daftar Item" }),
      /* @__PURE__ */ jsxRuntimeExports.jsxs(
        "button",
        {
          type: "button",
          onClick: () => update([...items, ""]),
          className: "flex items-center gap-1 text-xs text-red-600 hover:text-red-700 font-medium",
          "data-ocid": `${ocidPrefix}.add_button`,
          children: [
            /* @__PURE__ */ jsxRuntimeExports.jsx(Plus, { size: 14 }),
            " ",
            addLabel
          ]
        }
      )
    ] }),
    /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "space-y-2", children: items.map((item, idx) => (
      // biome-ignore lint/suspicious/noArrayIndexKey: ordered list
      /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex items-start gap-2", children: [
        /* @__PURE__ */ jsxRuntimeExports.jsxs("span", { className: "text-xs text-muted-foreground mt-2.5 w-5 flex-shrink-0", children: [
          idx + 1,
          "."
        ] }),
        /* @__PURE__ */ jsxRuntimeExports.jsx(
          "input",
          {
            className: "flex-1 border border-input rounded-lg px-3 py-2 text-sm text-foreground bg-background focus:outline-none focus:ring-2 focus:ring-red-500",
            value: item,
            onChange: (e) => {
              const updated = [...items];
              updated[idx] = e.target.value;
              update(updated);
            },
            placeholder,
            "data-ocid": `${ocidPrefix}.item.${idx + 1}`
          }
        ),
        /* @__PURE__ */ jsxRuntimeExports.jsx(
          "button",
          {
            type: "button",
            onClick: () => update(items.filter((_, i) => i !== idx)),
            className: "mt-2 text-muted-foreground hover:text-red-600 transition-colors",
            "aria-label": "Hapus item",
            "data-ocid": `${ocidPrefix}.delete_button.${idx + 1}`,
            children: /* @__PURE__ */ jsxRuntimeExports.jsx(Trash2, { size: 16 })
          }
        )
      ] }, idx)
    )) })
  ] });
}
function StrukturEditor({ blocks, onChange }) {
  const daftarBlock = blocks.find((b) => b.__kind__ === "daftarItem");
  const rawItems = (daftarBlock == null ? void 0 : daftarBlock.daftarItem) ?? [];
  const parseItem = (raw) => {
    const parts = {};
    for (const seg of raw.split(" | ")) {
      const [k, ...rest] = seg.split(": ");
      if (k && rest.length) parts[k.trim()] = rest.join(": ").trim();
    }
    return {
      jabatan: parts.Jabatan ?? "",
      nama: parts.Nama ?? "",
      unit: parts.Unit ?? ""
    };
  };
  const serializeItem = (jabatan, nama, unit) => `Jabatan: ${jabatan} | Nama: ${nama} | Unit: ${unit}`;
  const members = rawItems.map(parseItem);
  const update = (idx, field, val) => {
    const updated = [...members];
    updated[idx] = { ...updated[idx], [field]: val };
    onChange([
      {
        __kind__: "daftarItem",
        daftarItem: updated.map(
          (m) => serializeItem(m.jabatan, m.nama, m.unit)
        )
      }
    ]);
  };
  const addMember = () => {
    onChange([
      {
        __kind__: "daftarItem",
        daftarItem: [...rawItems, serializeItem("", "", "")]
      }
    ]);
  };
  const removeMember = (idx) => {
    onChange([
      {
        __kind__: "daftarItem",
        daftarItem: rawItems.filter((_, i) => i !== idx)
      }
    ]);
  };
  return /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "space-y-3", children: [
    /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex items-center justify-between", children: [
      /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "text-xs font-semibold text-foreground", children: "Anggota Struktur" }),
      /* @__PURE__ */ jsxRuntimeExports.jsxs(
        "button",
        {
          type: "button",
          onClick: addMember,
          className: "flex items-center gap-1 text-xs text-red-600 hover:text-red-700 font-medium",
          "data-ocid": "admin_tentang.struktur.add_button",
          children: [
            /* @__PURE__ */ jsxRuntimeExports.jsx(Plus, { size: 14 }),
            " Tambah Anggota"
          ]
        }
      )
    ] }),
    /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "space-y-3", children: members.map((m, idx) => /* @__PURE__ */ jsxRuntimeExports.jsxs(
      "div",
      {
        className: "border border-border rounded-lg p-3 bg-background space-y-2",
        children: [
          /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex items-center justify-between", children: [
            /* @__PURE__ */ jsxRuntimeExports.jsxs("span", { className: "text-xs font-medium text-muted-foreground", children: [
              "Anggota ",
              idx + 1
            ] }),
            /* @__PURE__ */ jsxRuntimeExports.jsx(
              "button",
              {
                type: "button",
                onClick: () => removeMember(idx),
                className: "text-muted-foreground hover:text-red-600 transition-colors",
                "aria-label": "Hapus anggota",
                "data-ocid": `admin_tentang.struktur.delete_button.${idx + 1}`,
                children: /* @__PURE__ */ jsxRuntimeExports.jsx(Trash2, { size: 15 })
              }
            )
          ] }),
          /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "grid grid-cols-1 sm:grid-cols-3 gap-2", children: [
            /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { children: [
              /* @__PURE__ */ jsxRuntimeExports.jsx(
                "label",
                {
                  htmlFor: `jabatan-${idx}`,
                  className: "text-xs text-muted-foreground",
                  children: "Jabatan"
                }
              ),
              /* @__PURE__ */ jsxRuntimeExports.jsx(
                "input",
                {
                  id: `jabatan-${idx}`,
                  className: "w-full border border-input rounded-md px-2 py-1.5 text-sm text-foreground bg-background focus:outline-none focus:ring-2 focus:ring-red-500 mt-0.5",
                  value: m.jabatan,
                  onChange: (e) => update(idx, "jabatan", e.target.value),
                  placeholder: "Jabatan",
                  "data-ocid": `admin_tentang.struktur.jabatan.${idx + 1}`
                }
              )
            ] }),
            /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { children: [
              /* @__PURE__ */ jsxRuntimeExports.jsx(
                "label",
                {
                  htmlFor: `nama-${idx}`,
                  className: "text-xs text-muted-foreground",
                  children: "Nama"
                }
              ),
              /* @__PURE__ */ jsxRuntimeExports.jsx(
                "input",
                {
                  id: `nama-${idx}`,
                  className: "w-full border border-input rounded-md px-2 py-1.5 text-sm text-foreground bg-background focus:outline-none focus:ring-2 focus:ring-red-500 mt-0.5",
                  value: m.nama,
                  onChange: (e) => update(idx, "nama", e.target.value),
                  placeholder: "Nama lengkap",
                  "data-ocid": `admin_tentang.struktur.nama.${idx + 1}`
                }
              )
            ] }),
            /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { children: [
              /* @__PURE__ */ jsxRuntimeExports.jsx(
                "label",
                {
                  htmlFor: `unit-${idx}`,
                  className: "text-xs text-muted-foreground",
                  children: "Unit"
                }
              ),
              /* @__PURE__ */ jsxRuntimeExports.jsx(
                "input",
                {
                  id: `unit-${idx}`,
                  className: "w-full border border-input rounded-md px-2 py-1.5 text-sm text-foreground bg-background focus:outline-none focus:ring-2 focus:ring-red-500 mt-0.5",
                  value: m.unit,
                  onChange: (e) => update(idx, "unit", e.target.value),
                  placeholder: "Unit / Jabatan Struktural",
                  "data-ocid": `admin_tentang.struktur.unit.${idx + 1}`
                }
              )
            ] })
          ] })
        ]
      },
      `member-${idx}-${m.jabatan.slice(0, 8)}`
    )) })
  ] });
}
function TabPanel({ slug, page }) {
  var _a, _b;
  const defaultBlocks = DEFAULT_CONTENT[slug];
  const [judul, setJudul] = reactExports.useState((page == null ? void 0 : page.judul) ?? DEFAULT_TITLES[slug]);
  const [blocks, setBlocks] = reactExports.useState(
    ((_b = (_a = page == null ? void 0 : page.konten) == null ? void 0 : _a.blocks) == null ? void 0 : _b.length) ? page.konten.blocks : defaultBlocks
  );
  reactExports.useEffect(() => {
    var _a2, _b2;
    if (page) {
      setJudul(page.judul);
      if ((_b2 = (_a2 = page.konten) == null ? void 0 : _a2.blocks) == null ? void 0 : _b2.length) setBlocks(page.konten.blocks);
    }
  }, [page]);
  const mutation = useUpdateTentangPage();
  const handleSave = () => {
    mutation.mutate(
      { slug, input: { judul, konten: { blocks } } },
      {
        onSuccess: () => ue.success("Konten berhasil disimpan!"),
        onError: (e) => ue.error(
          `Gagal menyimpan: ${e instanceof Error ? e.message : "Kesalahan tidak diketahui"}`
        )
      }
    );
  };
  return /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "space-y-6", "data-ocid": `admin_tentang.${slug}.panel`, children: [
    page && /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex flex-wrap items-center gap-4 text-xs text-muted-foreground bg-muted/30 border border-border rounded-lg px-4 py-2.5", children: [
      /* @__PURE__ */ jsxRuntimeExports.jsxs("span", { className: "flex items-center gap-1.5", children: [
        /* @__PURE__ */ jsxRuntimeExports.jsx(Clock, { size: 13 }),
        " Terakhir diperbarui:",
        " ",
        formatUpdatedAt(page.updatedAt)
      ] }),
      /* @__PURE__ */ jsxRuntimeExports.jsxs("span", { className: "flex items-center gap-1.5", children: [
        "oleh",
        " ",
        /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "font-mono font-medium text-foreground", children: truncatePrincipal(page.updatedBy ? String(page.updatedBy) : null) })
      ] })
    ] }),
    /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { children: [
      /* @__PURE__ */ jsxRuntimeExports.jsx(
        "label",
        {
          htmlFor: `judul-${slug}`,
          className: "block text-xs font-semibold text-foreground mb-1",
          children: "Judul Halaman"
        }
      ),
      /* @__PURE__ */ jsxRuntimeExports.jsx(
        "input",
        {
          id: `judul-${slug}`,
          className: "w-full border border-input rounded-lg px-3 py-2 text-sm text-foreground bg-background focus:outline-none focus:ring-2 focus:ring-red-500",
          value: judul,
          onChange: (e) => setJudul(e.target.value),
          placeholder: "Judul halaman",
          "data-ocid": `admin_tentang.${slug}.judul`
        }
      )
    ] }),
    /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "border border-border rounded-xl bg-card p-5", children: [
      slug === "sejarah" && /* @__PURE__ */ jsxRuntimeExports.jsx(SejarahEditor, { blocks, onChange: setBlocks }),
      slug === "visiMisi" && /* @__PURE__ */ jsxRuntimeExports.jsx(VisiMisiEditor, { blocks, onChange: setBlocks }),
      slug === "dasarHukum" && /* @__PURE__ */ jsxRuntimeExports.jsx(
        DaftarEditor,
        {
          blocks,
          onChange: setBlocks,
          addLabel: "Tambah Regulasi",
          placeholder: "Contoh: Peraturan Presiden Nomor 33 Tahun 2012...",
          ocidPrefix: "admin_tentang.dasar_hukum"
        }
      ),
      slug === "fungsi" && /* @__PURE__ */ jsxRuntimeExports.jsx(
        DaftarEditor,
        {
          blocks,
          onChange: setBlocks,
          addLabel: "Tambah Fungsi",
          placeholder: "Contoh: Mendokumentasikan seluruh produk hukum...",
          ocidPrefix: "admin_tentang.fungsi"
        }
      ),
      slug === "struktur" && /* @__PURE__ */ jsxRuntimeExports.jsx(StrukturEditor, { blocks, onChange: setBlocks })
    ] }),
    /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex items-center justify-between pt-2 border-t border-border", children: [
      mutation.isSuccess && /* @__PURE__ */ jsxRuntimeExports.jsxs("span", { className: "flex items-center gap-1.5 text-xs text-green-700", children: [
        /* @__PURE__ */ jsxRuntimeExports.jsx(CircleCheck, { size: 14 }),
        " Tersimpan"
      ] }),
      mutation.isError && /* @__PURE__ */ jsxRuntimeExports.jsxs("span", { className: "flex items-center gap-1.5 text-xs text-red-600", children: [
        /* @__PURE__ */ jsxRuntimeExports.jsx(CircleAlert, { size: 14 }),
        " Gagal menyimpan"
      ] }),
      !mutation.isSuccess && !mutation.isError && /* @__PURE__ */ jsxRuntimeExports.jsx("span", {}),
      /* @__PURE__ */ jsxRuntimeExports.jsxs(
        "button",
        {
          type: "button",
          onClick: handleSave,
          disabled: mutation.isPending,
          className: "flex items-center gap-2 px-5 py-2 bg-red-600 text-white text-sm font-semibold rounded-lg hover:bg-red-700 disabled:opacity-60 transition-colors",
          "data-ocid": `admin_tentang.${slug}.save_button`,
          children: [
            mutation.isPending ? /* @__PURE__ */ jsxRuntimeExports.jsx(LoaderCircle, { size: 16, className: "animate-spin" }) : /* @__PURE__ */ jsxRuntimeExports.jsx(Save, { size: 16 }),
            "Simpan Perubahan"
          ]
        }
      )
    ] })
  ] });
}
function AdminTentang() {
  const [activeTab, setActiveTab] = reactExports.useState("sejarah");
  const { data: pages, isLoading } = useTentangPages();
  const getPage = (slug) => {
    if (!pages) return null;
    const found = pages.find((p) => {
      const slugKey = String(p.slug);
      return slugKey === slug || p.slug.__kind__ === slug;
    });
    return found ?? null;
  };
  return /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "p-6 max-w-4xl mx-auto", "data-ocid": "admin_tentang.page", children: [
    /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex items-center gap-3 mb-6", children: [
      /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "w-10 h-10 bg-red-100 rounded-xl flex items-center justify-center flex-shrink-0", children: /* @__PURE__ */ jsxRuntimeExports.jsx(BookOpen, { className: "w-5 h-5 text-red-700" }) }),
      /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { children: [
        /* @__PURE__ */ jsxRuntimeExports.jsx("h1", { className: "text-xl font-bold text-foreground", children: "Kelola Tentang JDIH" }),
        /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-xs text-muted-foreground mt-0.5", children: 'Edit konten halaman publik "Tentang JDIH" — perubahan langsung tersimpan ke database' })
      ] })
    ] }),
    /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "border border-border rounded-xl overflow-hidden bg-card", children: [
      /* @__PURE__ */ jsxRuntimeExports.jsx(
        "div",
        {
          className: "flex overflow-x-auto border-b border-border bg-muted/30",
          role: "tablist",
          children: TABS.map((tab) => /* @__PURE__ */ jsxRuntimeExports.jsx(
            "button",
            {
              type: "button",
              role: "tab",
              "aria-selected": activeTab === tab.slug,
              onClick: () => setActiveTab(tab.slug),
              className: `flex-shrink-0 px-4 py-3 text-sm font-medium border-b-2 transition-colors ${activeTab === tab.slug ? "border-red-600 text-red-700 bg-background" : "border-transparent text-muted-foreground hover:text-foreground hover:bg-muted/50"}`,
              "data-ocid": `admin_tentang.tab.${tab.slug}`,
              children: tab.label
            },
            tab.slug
          ))
        }
      ),
      /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "p-6", children: isLoading ? /* @__PURE__ */ jsxRuntimeExports.jsxs(
        "div",
        {
          className: "flex items-center gap-2 text-sm text-muted-foreground py-8 justify-center",
          "data-ocid": "admin_tentang.loading_state",
          children: [
            /* @__PURE__ */ jsxRuntimeExports.jsx(LoaderCircle, { size: 18, className: "animate-spin text-red-600" }),
            "Memuat konten..."
          ]
        }
      ) : /* @__PURE__ */ jsxRuntimeExports.jsx(
        TabPanel,
        {
          slug: activeTab,
          page: getPage(activeTab)
        },
        activeTab
      ) })
    ] }),
    /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "mt-4 flex items-start gap-3 p-4 bg-gray-50 border border-gray-200 rounded-lg text-sm text-gray-700", children: [
      /* @__PURE__ */ jsxRuntimeExports.jsx(CircleAlert, { size: 16, className: "flex-shrink-0 mt-0.5" }),
      /* @__PURE__ */ jsxRuntimeExports.jsxs("p", { children: [
        /* @__PURE__ */ jsxRuntimeExports.jsx("strong", { children: "Tips:" }),
        ' Setiap tab menyimpan secara terpisah. Klik "Simpan Perubahan" di setiap tab untuk menyimpan konten halaman tersebut. Konten yang disimpan langsung tampil di halaman publik.'
      ] })
    ] })
  ] });
}
export {
  AdminTentang
};
