// @caffeine/template-frontend/dist/assets/AdminDokumen-DqV8Z0N5.js
import { c as createLucideIcon, t as useQueryClient, p as usePermissions, r as reactExports, j as jsxRuntimeExports, a as Button, e as SkeletonList, v as LoaderCircle, h as api } from "./index-B7-RSq-m.js";
import { A as AlertDialog, a as AlertDialogContent, b as AlertDialogHeader, c as AlertDialogTitle, d as AlertDialogDescription, e as AlertDialogFooter, f as AlertDialogCancel, g as AlertDialogAction } from "./alert-dialog-lTbm3t0t.js";
import { D as Dialog, a as DialogContent, b as DialogHeader, c as DialogTitle, d as DialogFooter } from "./dialog-CIgZYBSq.js";
import { I as Input } from "./input-B3nX7k9Z.js";
import { L as Label } from "./label-eF2rHvnY.js";
import { S as Select, a as SelectTrigger, b as SelectValue, c as SelectContent, d as SelectItem } from "./select-37c5ocl_.js";
import { T as Textarea, U as Upload } from "./textarea-DRo76BEL.js";
import { u as ue } from "./index-B0YpMaU8.js";
import { E as EmptyState } from "./EmptyState-Bzc48I5U.js";
import { E as ErrorState } from "./ErrorState-o51Yfgm5.js";
import { W as WORKFLOW_STATUS_LABEL, K as KategoriTag, S as StatusBadge, b as WorkflowStatusBadge, T as TAHUN_OPTIONS } from "./StatusBadge-CanDaSra.js";
import { i as useDokumenAdmin, d as useKategori, j as useDeleteDokumen, k as useSubmitForReview, l as usePublishDokumen, m as useReturnToDraft } from "./useBackend-xFEMRG87.js";
import { P as Plus } from "./plus-C7LHpCDy.js";
import { S as Search } from "./search-DeNGI9Ut.js";
import { U as UserCheck } from "./user-check-BV1RhZlN.js";
import { P as Pencil } from "./pencil-CwCkT-7p.js";
import { T as Trash2 } from "./trash-2-CnhMEsa2.js";
import { C as Clock } from "./clock-CMHdisaV.js";
import { R as RotateCcw } from "./rotate-ccw-DC-KaPvS.js";
import "./index-BcOIygc1.js";
import "./index-CUotUQbC.js";
import "./circle-alert-D-S0YTKb.js";
import "./badge-KXcDp_20.js";
import "./useMutation-D--BDafu.js";
/**
 * @license lucide-react v0.511.0 - ISC
 *
 * This source code is licensed under the ISC license.
 * See the LICENSE file in the root directory of this source tree.
 */
const __iconNode$1 = [
  ["path", { d: "M21.801 10A10 10 0 1 1 17 3.335", key: "yps3ct" }],
  ["path", { d: "m9 11 3 3L22 4", key: "1pflzl" }]
];
const CircleCheckBig = createLucideIcon("circle-check-big", __iconNode$1);
/**
 * @license lucide-react v0.511.0 - ISC
 *
 * This source code is licensed under the ISC license.
 * See the LICENSE file in the root directory of this source tree.
 */
const __iconNode = [
  [
    "path",
    {
      d: "M14.536 21.686a.5.5 0 0 0 .937-.024l6.5-19a.496.496 0 0 0-.635-.635l-19 6.5a.5.5 0 0 0-.024.937l7.93 3.18a2 2 0 0 1 1.112 1.11z",
      key: "1ffxy3"
    }
  ],
  ["path", { d: "m21.854 2.147-10.94 10.939", key: "12cjpa" }]
];
const Send = createLucideIcon("send", __iconNode);
function truncatePrincipal(principal) {
  if (!principal) return "-";
  if (principal.length <= 12) return principal;
  return `${principal.slice(0, 12)}...`;
}
const EMPTY_FORM = {
  judul: "",
  nomor: "",
  kategoriId: "",
  tahun: String((/* @__PURE__ */ new Date()).getFullYear()),
  tanggalPenetapan: "",
  tanggalPengundangan: "",
  relasiHukum: "",
  status: "Berlaku",
  abstrak: "",
  tags: "",
  file: null
};
function validate(form, isEdit) {
  const errs = {};
  if (!form.judul.trim()) errs.judul = "Judul wajib diisi";
  if (!form.nomor.trim()) errs.nomor = "Nomor dokumen wajib diisi";
  if (!form.kategoriId) errs.kategoriId = "Pilih kategori";
  if (!form.tanggalPenetapan)
    errs.tanggalPenetapan = "Tanggal penetapan wajib diisi";
  if (!form.abstrak.trim()) errs.abstrak = "Abstrak wajib diisi";
  if (!isEdit && !form.file) errs.file = "File PDF wajib diunggah";
  return errs;
}
const WORKFLOW_FILTER_OPTIONS = [
  { value: "all", label: "Semua Status Alur" },
  { value: "Draft", label: WORKFLOW_STATUS_LABEL.Draft },
  { value: "PendingReview", label: WORKFLOW_STATUS_LABEL.PendingReview },
  { value: "Published", label: WORKFLOW_STATUS_LABEL.Published }
];
function ReturnToDraftForm({ docId, onClose }) {
  const [catatan, setCatatan] = reactExports.useState("");
  const returnToDraft = useReturnToDraft();
  async function handleSubmit() {
    if (!catatan.trim()) {
      ue.error("Catatan koreksi wajib diisi");
      return;
    }
    try {
      await returnToDraft.mutateAsync({ id: docId, catatan: catatan.trim() });
      ue.success("Dokumen dikembalikan ke Draf");
      onClose();
    } catch (err) {
      ue.error(
        err instanceof Error ? err.message : "Gagal mengembalikan ke draf"
      );
    }
  }
  return /* @__PURE__ */ jsxRuntimeExports.jsxs(
    "div",
    {
      className: "mt-2 rounded-lg border border-yellow-300 bg-yellow-50 p-3 space-y-2",
      "data-ocid": "admin_dokumen.return_draft_form",
      children: [
        /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-xs font-semibold text-yellow-800", children: "Catatan Koreksi untuk Operator" }),
        /* @__PURE__ */ jsxRuntimeExports.jsx(
          Textarea,
          {
            value: catatan,
            onChange: (e) => setCatatan(e.target.value),
            placeholder: "Tuliskan alasan pengembalian atau koreksi yang perlu diperbaiki...",
            rows: 3,
            className: "text-xs resize-none border-yellow-300 focus:border-yellow-500",
            "data-ocid": "admin_dokumen.catatan_koreksi_textarea"
          }
        ),
        /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex gap-2 justify-end", children: [
          /* @__PURE__ */ jsxRuntimeExports.jsx(
            Button,
            {
              variant: "outline",
              size: "sm",
              onClick: onClose,
              disabled: returnToDraft.isPending,
              className: "text-xs h-7",
              "data-ocid": "admin_dokumen.return_draft_cancel_button",
              children: "Batal"
            }
          ),
          /* @__PURE__ */ jsxRuntimeExports.jsxs(
            Button,
            {
              size: "sm",
              onClick: handleSubmit,
              disabled: returnToDraft.isPending || !catatan.trim(),
              className: "text-xs h-7 bg-yellow-600 hover:bg-yellow-700 text-white",
              "data-ocid": "admin_dokumen.return_draft_confirm_button",
              children: [
                returnToDraft.isPending ? /* @__PURE__ */ jsxRuntimeExports.jsx(LoaderCircle, { className: "w-3 h-3 animate-spin mr-1" }) : null,
                "Kembalikan ke Draf"
              ]
            }
          )
        ] })
      ]
    }
  );
}
function WorkflowActions({
  docId,
  workflowStatus,
  isSuperAdmin,
  idx
}) {
  const [showReturnForm, setShowReturnForm] = reactExports.useState(false);
  const submitForReview = useSubmitForReview();
  const publishDokumen = usePublishDokumen();
  if (workflowStatus === "Published") return null;
  if (workflowStatus === "Draft") {
    return /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "mt-2", children: /* @__PURE__ */ jsxRuntimeExports.jsxs(
      Button,
      {
        variant: "outline",
        size: "sm",
        className: "text-xs h-7 border-foreground/30 gap-1.5",
        disabled: submitForReview.isPending,
        onClick: async () => {
          try {
            await submitForReview.mutateAsync(String(docId));
            ue.success("Dokumen diajukan untuk review");
          } catch (err) {
            ue.error(
              err instanceof Error ? err.message : "Gagal mengajukan untuk review"
            );
          }
        },
        "data-ocid": `admin_dokumen.ajukan_review_button.${idx}`,
        children: [
          submitForReview.isPending ? /* @__PURE__ */ jsxRuntimeExports.jsx(LoaderCircle, { className: "w-3 h-3 animate-spin" }) : /* @__PURE__ */ jsxRuntimeExports.jsx(Send, { className: "w-3 h-3" }),
          "Ajukan Review"
        ]
      }
    ) });
  }
  if (workflowStatus === "PendingReview") {
    if (!isSuperAdmin) {
      return /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "mt-2", children: /* @__PURE__ */ jsxRuntimeExports.jsxs(
        "span",
        {
          className: "inline-flex items-center gap-1.5 text-xs text-yellow-700 bg-yellow-50 border border-yellow-200 rounded px-2 py-1",
          "data-ocid": `admin_dokumen.menunggu_persetujuan.${idx}`,
          children: [
            /* @__PURE__ */ jsxRuntimeExports.jsx(Clock, { className: "w-3 h-3" }),
            "Menunggu Persetujuan Super Admin"
          ]
        }
      ) });
    }
    return /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "mt-2 space-y-2", children: [
      /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex flex-wrap gap-2", children: [
        /* @__PURE__ */ jsxRuntimeExports.jsxs(
          Button,
          {
            size: "sm",
            className: "text-xs h-7 bg-green-600 hover:bg-green-700 text-white gap-1.5",
            disabled: publishDokumen.isPending,
            onClick: async () => {
              try {
                await publishDokumen.mutateAsync(String(docId));
                ue.success("Dokumen berhasil diterbitkan");
              } catch (err) {
                ue.error(
                  err instanceof Error ? err.message : "Gagal menerbitkan dokumen"
                );
              }
            },
            "data-ocid": `admin_dokumen.terbitkan_button.${idx}`,
            children: [
              publishDokumen.isPending ? /* @__PURE__ */ jsxRuntimeExports.jsx(LoaderCircle, { className: "w-3 h-3 animate-spin" }) : /* @__PURE__ */ jsxRuntimeExports.jsx(CircleCheckBig, { className: "w-3 h-3" }),
              "Terbitkan"
            ]
          }
        ),
        /* @__PURE__ */ jsxRuntimeExports.jsxs(
          Button,
          {
            variant: "outline",
            size: "sm",
            className: "text-xs h-7 border-yellow-400 text-yellow-700 hover:bg-yellow-50 gap-1.5",
            onClick: () => setShowReturnForm((v) => !v),
            "data-ocid": `admin_dokumen.kembalikan_draft_button.${idx}`,
            children: [
              /* @__PURE__ */ jsxRuntimeExports.jsx(RotateCcw, { className: "w-3 h-3" }),
              "Kembalikan ke Draf"
            ]
          }
        )
      ] }),
      showReturnForm && /* @__PURE__ */ jsxRuntimeExports.jsx(
        ReturnToDraftForm,
        {
          docId: String(docId),
          onClose: () => setShowReturnForm(false)
        }
      )
    ] });
  }
  return null;
}
function AdminDokumen() {
  const qc = useQueryClient();
  const { isAdmin, grantedMenus } = usePermissions();
  const [search, setSearch] = reactExports.useState("");
  const [workflowFilter, setWorkflowFilter] = reactExports.useState(
    "all"
  );
  const {
    data: dokumen,
    isLoading,
    isError,
    refetch
  } = useDokumenAdmin({
    query: search,
    workflowStatus: workflowFilter === "all" ? void 0 : workflowFilter
  });
  const { data: kategoriList } = useKategori();
  const deleteDokumen = useDeleteDokumen();
  const [showForm, setShowForm] = reactExports.useState(false);
  const [editId, setEditId] = reactExports.useState(null);
  const [editAudit, setEditAudit] = reactExports.useState({
    createdBy: "",
    lastModifiedBy: ""
  });
  const [editCatatanKoreksi, setEditCatatanKoreksi] = reactExports.useState("");
  const [form, setForm] = reactExports.useState(EMPTY_FORM);
  const [errors, setErrors] = reactExports.useState({});
  const [saving, setSaving] = reactExports.useState(false);
  const [uploadProgress, setUploadProgress] = reactExports.useState(0);
  const [deleteTarget, setDeleteTarget] = reactExports.useState(null);
  const fileRef = reactExports.useRef(null);
  function openCreate() {
    setForm(EMPTY_FORM);
    setErrors({});
    setEditId(null);
    setEditAudit({ createdBy: "", lastModifiedBy: "" });
    setEditCatatanKoreksi("");
    setShowForm(true);
  }
  function openEdit(doc) {
    let tglPengundangan = "";
    if (doc.tanggalPengundangan != null) {
      const d = new Date(Number(doc.tanggalPengundangan));
      tglPengundangan = d.toISOString().slice(0, 10);
    }
    setForm({
      judul: doc.judul,
      nomor: doc.nomor,
      kategoriId: String(doc.kategoriId),
      tahun: String(doc.tahun),
      tanggalPenetapan: doc.tanggalPenetapan,
      tanggalPengundangan: tglPengundangan,
      relasiHukum: doc.relasiHukum ?? "",
      status: doc.status,
      abstrak: doc.abstrak,
      tags: (doc.tag ?? []).join(", "),
      file: null
    });
    setErrors({});
    setEditId(doc.id);
    setEditAudit({
      createdBy: doc.createdBy ?? "",
      lastModifiedBy: doc.lastModifiedBy ?? ""
    });
    setEditCatatanKoreksi(doc.catatanKoreksi ?? "");
    setShowForm(true);
  }
  function setField(key, val) {
    setForm((prev) => ({ ...prev, [key]: val }));
    if (errors[key]) {
      setErrors((prev) => ({ ...prev, [key]: void 0 }));
    }
  }
  async function handleSave() {
    const errs = validate(form, !!editId);
    if (Object.keys(errs).length > 0) {
      setErrors(errs);
      return;
    }
    setSaving(true);
    setUploadProgress(0);
    try {
      const tagsArr = form.tags.split(",").map((t) => t.trim()).filter(Boolean);
      const input = {
        judul: form.judul.trim(),
        nomor: form.nomor.trim(),
        kategoriId: Number(form.kategoriId),
        tahun: Number(form.tahun),
        tanggalPenetapan: form.tanggalPenetapan,
        tanggalPengundangan: form.tanggalPengundangan || void 0,
        relasiHukum: form.relasiHukum.trim() || void 0,
        status: form.status,
        abstrak: form.abstrak.trim(),
        tag: tagsArr,
        filePdf: form.file || void 0
      };
      if (editId) {
        await api.dokumen.update(String(editId), input);
        ue.success("Dokumen berhasil diperbarui");
      } else {
        if (!form.file) return;
        await api.dokumen.add(input);
        ue.success("Dokumen berhasil ditambahkan");
      }
      qc.invalidateQueries({ queryKey: ["dokumen"] });
      qc.invalidateQueries({ queryKey: ["dokumenAdmin"] });
      qc.invalidateQueries({ queryKey: ["statistik"] });
      setShowForm(false);
    } catch (err) {
      console.error(err);
      ue.error("Gagal menyimpan dokumen. Silakan coba lagi.");
    } finally {
      setSaving(false);
      setUploadProgress(0);
    }
  }
  async function handleDelete(id) {
    try {
      await deleteDokumen.mutateAsync(String(id));
      ue.success("Dokumen berhasil dihapus");
    } catch {
      ue.error("Gagal menghapus dokumen");
    } finally {
      setDeleteTarget(null);
    }
  }
  return /* @__PURE__ */ jsxRuntimeExports.jsxs(jsxRuntimeExports.Fragment, { children: [
    /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "container mx-auto px-4 py-6", children: [
      /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex flex-col sm:flex-row sm:items-center sm:justify-between gap-3 mb-6", children: [
        /* @__PURE__ */ jsxRuntimeExports.jsx("h1", { className: "font-display font-bold text-xl text-foreground", children: "Kelola Dokumen" }),
        /* @__PURE__ */ jsxRuntimeExports.jsxs(
          Button,
          {
            onClick: openCreate,
            className: "bg-accent hover:bg-accent/90 text-accent-foreground gap-2 text-sm",
            "data-ocid": "admin_dokumen.tambah_button",
            children: [
              /* @__PURE__ */ jsxRuntimeExports.jsx(Plus, { className: "w-4 h-4" }),
              "Tambah Dokumen"
            ]
          }
        )
      ] }),
      /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex flex-col sm:flex-row gap-3 mb-5", children: [
        /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "relative flex-1 max-w-sm", children: [
          /* @__PURE__ */ jsxRuntimeExports.jsx(Search, { className: "absolute left-3 top-1/2 -translate-y-1/2 w-4 h-4 text-muted-foreground" }),
          /* @__PURE__ */ jsxRuntimeExports.jsx(
            Input,
            {
              placeholder: "Cari dokumen...",
              value: search,
              onChange: (e) => setSearch(e.target.value),
              className: "pl-9 text-sm",
              "data-ocid": "admin_dokumen.search_input"
            }
          )
        ] }),
        /* @__PURE__ */ jsxRuntimeExports.jsxs(
          Select,
          {
            value: workflowFilter,
            onValueChange: (v) => setWorkflowFilter(v),
            children: [
              /* @__PURE__ */ jsxRuntimeExports.jsx(
                SelectTrigger,
                {
                  className: "w-[200px] text-sm",
                  "data-ocid": "admin_dokumen.workflow_filter_select",
                  children: /* @__PURE__ */ jsxRuntimeExports.jsx(SelectValue, { placeholder: "Semua Status Alur" })
                }
              ),
              /* @__PURE__ */ jsxRuntimeExports.jsx(SelectContent, { children: WORKFLOW_FILTER_OPTIONS.map((opt) => /* @__PURE__ */ jsxRuntimeExports.jsx(SelectItem, { value: opt.value, children: opt.label }, opt.value)) })
            ]
          }
        )
      ] }),
      isLoading ? /* @__PURE__ */ jsxRuntimeExports.jsx(SkeletonList, { count: 5 }) : isError ? /* @__PURE__ */ jsxRuntimeExports.jsx(ErrorState, { onRetry: () => refetch() }) : !dokumen || dokumen.length === 0 ? /* @__PURE__ */ jsxRuntimeExports.jsx(
        EmptyState,
        {
          title: "Belum ada dokumen",
          description: "Tambahkan dokumen hukum pertama Anda.",
          actionLabel: "Tambah Dokumen",
          onAction: openCreate,
          "data-ocid": "admin_dokumen.empty_state"
        }
      ) : /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "space-y-3", "data-ocid": "admin_dokumen.dokumen_list", children: dokumen.map((doc, idx) => {
        const docAny = doc;
        const createdBy = docAny.createdBy ?? "";
        const lastModifiedBy = docAny.lastModifiedBy ?? "";
        const catatanKoreksi = docAny.catatanKoreksi ?? "";
        const workflowStatus = docAny.workflowStatus ?? "Draft";
        return /* @__PURE__ */ jsxRuntimeExports.jsx(
          "div",
          {
            className: "bg-card border border-border rounded-lg p-4",
            "data-ocid": `admin_dokumen.dokumen_item.${idx + 1}`,
            children: /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex items-start gap-4", children: [
              /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex-1 min-w-0", children: [
                /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex flex-wrap items-center gap-2 mb-1", children: [
                  /* @__PURE__ */ jsxRuntimeExports.jsx(KategoriTag, { nama: doc.kategoriNama ?? "-" }),
                  /* @__PURE__ */ jsxRuntimeExports.jsx(StatusBadge, { status: doc.status }),
                  /* @__PURE__ */ jsxRuntimeExports.jsx(WorkflowStatusBadge, { status: workflowStatus })
                ] }),
                /* @__PURE__ */ jsxRuntimeExports.jsx("h3", { className: "font-medium text-foreground text-sm leading-snug truncate", children: doc.judul }),
                /* @__PURE__ */ jsxRuntimeExports.jsxs("p", { className: "text-muted-foreground text-xs mt-0.5", children: [
                  doc.nomor,
                  " · ",
                  doc.tahun
                ] }),
                /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex flex-wrap gap-3 mt-1.5 text-xs text-muted-foreground", children: [
                  /* @__PURE__ */ jsxRuntimeExports.jsxs(
                    "span",
                    {
                      className: "flex items-center gap-1",
                      title: createdBy || "—",
                      children: [
                        /* @__PURE__ */ jsxRuntimeExports.jsx(UserCheck, { className: "w-3 h-3" }),
                        "Dibuat:",
                        " ",
                        /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "font-mono", children: createdBy ? truncatePrincipal(createdBy) : "—" })
                      ]
                    }
                  ),
                  lastModifiedBy && /* @__PURE__ */ jsxRuntimeExports.jsxs(
                    "span",
                    {
                      className: "flex items-center gap-1",
                      title: lastModifiedBy,
                      children: [
                        "Diubah:",
                        " ",
                        /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "font-mono", children: truncatePrincipal(lastModifiedBy) })
                      ]
                    }
                  )
                ] }),
                /* @__PURE__ */ jsxRuntimeExports.jsx(
                  WorkflowActions,
                  {
                    docId: doc.id,
                    workflowStatus,
                    isSuperAdmin: isAdmin,
                    idx: idx + 1
                  }
                )
              ] }),
              /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex items-center gap-2 shrink-0", children: [
                /* @__PURE__ */ jsxRuntimeExports.jsx(
                  Button,
                  {
                    variant: "ghost",
                    size: "icon",
                    className: "h-8 w-8 border border-border",
                    onClick: () => openEdit({
                      ...doc,
                      tanggalPenetapan: doc.tanggalPenetapan ?? "",
                      tanggalPengundangan: doc.tanggalPengundangan ? new Date(doc.tanggalPengundangan).getTime() : void 0,
                      tag: doc.tag,
                      catatanKoreksi,
                      createdBy,
                      lastModifiedBy
                    }),
                    "data-ocid": `admin_dokumen.edit_button.${idx + 1}`,
                    children: /* @__PURE__ */ jsxRuntimeExports.jsx(Pencil, { className: "w-3.5 h-3.5" })
                  }
                ),
                /* @__PURE__ */ jsxRuntimeExports.jsx(
                  Button,
                  {
                    variant: "ghost",
                    size: "icon",
                    className: "h-8 w-8 border border-border text-accent hover:text-accent hover:bg-accent/5",
                    onClick: () => setDeleteTarget({ id: doc.id, judul: doc.judul }),
                    "data-ocid": `admin_dokumen.delete_button.${idx + 1}`,
                    children: /* @__PURE__ */ jsxRuntimeExports.jsx(Trash2, { className: "w-3.5 h-3.5" })
                  }
                )
              ] })
            ] })
          },
          doc.id
        );
      }) })
    ] }),
    /* @__PURE__ */ jsxRuntimeExports.jsx(Dialog, { open: showForm, onOpenChange: (v) => !saving && setShowForm(v), children: /* @__PURE__ */ jsxRuntimeExports.jsxs(
      DialogContent,
      {
        className: "max-w-lg max-h-[90vh] overflow-y-auto",
        "data-ocid": "admin_dokumen.dialog",
        children: [
          /* @__PURE__ */ jsxRuntimeExports.jsx(DialogHeader, { children: /* @__PURE__ */ jsxRuntimeExports.jsx(DialogTitle, { className: "font-display", children: editId ? "Edit Dokumen" : "Tambah Dokumen" }) }),
          /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "space-y-4 py-2", children: [
            editId && editCatatanKoreksi && /* @__PURE__ */ jsxRuntimeExports.jsxs(
              "div",
              {
                className: "rounded-lg border border-yellow-300 bg-yellow-50 p-3",
                "data-ocid": "admin_dokumen.catatan_koreksi_panel",
                children: [
                  /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-xs font-semibold text-yellow-800 mb-1", children: "Catatan Koreksi dari Super Admin" }),
                  /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-xs text-yellow-700 whitespace-pre-wrap", children: editCatatanKoreksi })
                ]
              }
            ),
            /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { children: [
              /* @__PURE__ */ jsxRuntimeExports.jsxs(Label, { htmlFor: "dok-judul", className: "text-sm font-medium", children: [
                "Judul ",
                /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "text-accent", children: "*" })
              ] }),
              /* @__PURE__ */ jsxRuntimeExports.jsx(
                Input,
                {
                  id: "dok-judul",
                  value: form.judul,
                  onChange: (e) => setField("judul", e.target.value),
                  placeholder: "Masukkan judul dokumen",
                  className: "mt-1 text-sm",
                  "data-ocid": "admin_dokumen.judul_input"
                }
              ),
              errors.judul && /* @__PURE__ */ jsxRuntimeExports.jsx(
                "p",
                {
                  className: "text-xs text-accent mt-1",
                  "data-ocid": "admin_dokumen.judul.field_error",
                  children: errors.judul
                }
              )
            ] }),
            /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { children: [
              /* @__PURE__ */ jsxRuntimeExports.jsxs(Label, { htmlFor: "dok-nomor", className: "text-sm font-medium", children: [
                "Nomor Dokumen ",
                /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "text-accent", children: "*" })
              ] }),
              /* @__PURE__ */ jsxRuntimeExports.jsx(
                Input,
                {
                  id: "dok-nomor",
                  value: form.nomor,
                  onChange: (e) => setField("nomor", e.target.value),
                  placeholder: "cth. SK/001/UNTAG-BWI/2024",
                  className: "mt-1 text-sm",
                  "data-ocid": "admin_dokumen.nomor_input"
                }
              ),
              errors.nomor && /* @__PURE__ */ jsxRuntimeExports.jsx(
                "p",
                {
                  className: "text-xs text-accent mt-1",
                  "data-ocid": "admin_dokumen.nomor.field_error",
                  children: errors.nomor
                }
              )
            ] }),
            /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "grid grid-cols-2 gap-3", children: [
              /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { children: [
                /* @__PURE__ */ jsxRuntimeExports.jsxs(Label, { className: "text-sm font-medium", children: [
                  "Kategori ",
                  /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "text-accent", children: "*" })
                ] }),
                /* @__PURE__ */ jsxRuntimeExports.jsxs(
                  Select,
                  {
                    value: form.kategoriId,
                    onValueChange: (v) => setField("kategoriId", v),
                    children: [
                      /* @__PURE__ */ jsxRuntimeExports.jsx(
                        SelectTrigger,
                        {
                          className: "mt-1 text-sm",
                          "data-ocid": "admin_dokumen.kategori_select",
                          children: /* @__PURE__ */ jsxRuntimeExports.jsx(SelectValue, { placeholder: "Pilih kategori" })
                        }
                      ),
                      /* @__PURE__ */ jsxRuntimeExports.jsx(SelectContent, { children: (kategoriList ?? []).map((k) => /* @__PURE__ */ jsxRuntimeExports.jsx(SelectItem, { value: String(k.id), children: k.nama }, k.id)) })
                    ]
                  }
                ),
                errors.kategoriId && /* @__PURE__ */ jsxRuntimeExports.jsx(
                  "p",
                  {
                    className: "text-xs text-accent mt-1",
                    "data-ocid": "admin_dokumen.kategori.field_error",
                    children: errors.kategoriId
                  }
                )
              ] }),
              /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { children: [
                /* @__PURE__ */ jsxRuntimeExports.jsx(Label, { className: "text-sm font-medium", children: "Tahun" }),
                /* @__PURE__ */ jsxRuntimeExports.jsxs(
                  Select,
                  {
                    value: form.tahun,
                    onValueChange: (v) => setField("tahun", v),
                    children: [
                      /* @__PURE__ */ jsxRuntimeExports.jsx(
                        SelectTrigger,
                        {
                          className: "mt-1 text-sm",
                          "data-ocid": "admin_dokumen.tahun_select",
                          children: /* @__PURE__ */ jsxRuntimeExports.jsx(SelectValue, {})
                        }
                      ),
                      /* @__PURE__ */ jsxRuntimeExports.jsx(SelectContent, { children: TAHUN_OPTIONS.map((y) => /* @__PURE__ */ jsxRuntimeExports.jsx(SelectItem, { value: String(y), children: y }, String(y))) })
                    ]
                  }
                )
              ] })
            ] }),
            /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "grid grid-cols-2 gap-3", children: [
              /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { children: [
                /* @__PURE__ */ jsxRuntimeExports.jsxs(Label, { htmlFor: "dok-tgl", className: "text-sm font-medium", children: [
                  "Tanggal Penetapan ",
                  /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "text-accent", children: "*" })
                ] }),
                /* @__PURE__ */ jsxRuntimeExports.jsx(
                  Input,
                  {
                    id: "dok-tgl",
                    type: "date",
                    value: form.tanggalPenetapan,
                    onChange: (e) => setField("tanggalPenetapan", e.target.value),
                    className: "mt-1 text-sm",
                    "data-ocid": "admin_dokumen.tanggal_input"
                  }
                ),
                errors.tanggalPenetapan && /* @__PURE__ */ jsxRuntimeExports.jsx(
                  "p",
                  {
                    className: "text-xs text-accent mt-1",
                    "data-ocid": "admin_dokumen.tanggal.field_error",
                    children: errors.tanggalPenetapan
                  }
                )
              ] }),
              /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { children: [
                /* @__PURE__ */ jsxRuntimeExports.jsx(Label, { className: "text-sm font-medium", children: "Status" }),
                /* @__PURE__ */ jsxRuntimeExports.jsxs(
                  Select,
                  {
                    value: form.status,
                    onValueChange: (v) => setField("status", v),
                    children: [
                      /* @__PURE__ */ jsxRuntimeExports.jsx(
                        SelectTrigger,
                        {
                          className: "mt-1 text-sm",
                          "data-ocid": "admin_dokumen.status_select",
                          children: /* @__PURE__ */ jsxRuntimeExports.jsx(SelectValue, {})
                        }
                      ),
                      /* @__PURE__ */ jsxRuntimeExports.jsxs(SelectContent, { children: [
                        /* @__PURE__ */ jsxRuntimeExports.jsx(SelectItem, { value: "Berlaku", children: "Berlaku" }),
                        /* @__PURE__ */ jsxRuntimeExports.jsx(SelectItem, { value: "TidakBerlaku", children: "Tidak Berlaku" }),
                        /* @__PURE__ */ jsxRuntimeExports.jsx(SelectItem, { value: "Dicabut", children: "Dicabut" })
                      ] })
                    ]
                  }
                )
              ] })
            ] }),
            /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { children: [
              /* @__PURE__ */ jsxRuntimeExports.jsxs(
                Label,
                {
                  htmlFor: "dok-tgl-pengundangan",
                  className: "text-sm font-medium",
                  children: [
                    "Tanggal Pengundangan",
                    " ",
                    /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "text-muted-foreground font-normal", children: "(opsional)" })
                  ]
                }
              ),
              /* @__PURE__ */ jsxRuntimeExports.jsx(
                Input,
                {
                  id: "dok-tgl-pengundangan",
                  type: "date",
                  value: form.tanggalPengundangan,
                  onChange: (e) => setField("tanggalPengundangan", e.target.value),
                  className: "mt-1 text-sm",
                  "data-ocid": "admin_dokumen.tanggal_pengundangan_input"
                }
              )
            ] }),
            /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { children: [
              /* @__PURE__ */ jsxRuntimeExports.jsxs(Label, { htmlFor: "dok-relasi-hukum", className: "text-sm font-medium", children: [
                "Relasi Dengan Produk Hukum Lain",
                " ",
                /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "text-muted-foreground font-normal", children: "(opsional)" })
              ] }),
              /* @__PURE__ */ jsxRuntimeExports.jsx(
                Textarea,
                {
                  id: "dok-relasi-hukum",
                  value: form.relasiHukum,
                  onChange: (e) => setField("relasiHukum", e.target.value),
                  placeholder: "Contoh: Mencabut Peraturan Rektor No. 5 Tahun 2018",
                  rows: 2,
                  className: "mt-1 text-sm resize-none",
                  "data-ocid": "admin_dokumen.relasi_hukum_textarea"
                }
              )
            ] }),
            /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { children: [
              /* @__PURE__ */ jsxRuntimeExports.jsxs(Label, { htmlFor: "dok-abstrak", className: "text-sm font-medium", children: [
                "Abstrak ",
                /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "text-accent", children: "*" })
              ] }),
              /* @__PURE__ */ jsxRuntimeExports.jsx(
                Textarea,
                {
                  id: "dok-abstrak",
                  value: form.abstrak,
                  onChange: (e) => setField("abstrak", e.target.value),
                  placeholder: "Ringkasan isi dokumen...",
                  rows: 3,
                  className: "mt-1 text-sm resize-none",
                  "data-ocid": "admin_dokumen.abstrak_textarea"
                }
              ),
              errors.abstrak && /* @__PURE__ */ jsxRuntimeExports.jsx(
                "p",
                {
                  className: "text-xs text-accent mt-1",
                  "data-ocid": "admin_dokumen.abstrak.field_error",
                  children: errors.abstrak
                }
              )
            ] }),
            /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { children: [
              /* @__PURE__ */ jsxRuntimeExports.jsx(Label, { htmlFor: "dok-tags", className: "text-sm font-medium", children: "Tag (pisahkan dengan koma)" }),
              /* @__PURE__ */ jsxRuntimeExports.jsx(
                Input,
                {
                  id: "dok-tags",
                  value: form.tags,
                  onChange: (e) => setField("tags", e.target.value),
                  placeholder: "cth. statuta, akademik, 2024",
                  className: "mt-1 text-sm",
                  "data-ocid": "admin_dokumen.tags_input"
                }
              )
            ] }),
            /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { children: [
              /* @__PURE__ */ jsxRuntimeExports.jsxs(Label, { className: "text-sm font-medium", children: [
                "File PDF ",
                !editId && /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "text-accent", children: "*" }),
                editId && /* @__PURE__ */ jsxRuntimeExports.jsxs("span", { className: "text-muted-foreground font-normal", children: [
                  " ",
                  "(kosongkan jika tidak mengubah)"
                ] })
              ] }),
              /* @__PURE__ */ jsxRuntimeExports.jsxs(
                "button",
                {
                  type: "button",
                  className: "mt-1 w-full border-2 border-dashed border-border rounded-lg p-4 text-center cursor-pointer hover:border-primary/40 transition-colors",
                  onClick: () => {
                    var _a;
                    return (_a = fileRef.current) == null ? void 0 : _a.click();
                  },
                  "data-ocid": "admin_dokumen.upload_button",
                  children: [
                    /* @__PURE__ */ jsxRuntimeExports.jsx(Upload, { className: "w-5 h-5 text-muted-foreground mx-auto mb-2" }),
                    form.file ? /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-sm text-foreground font-medium", children: form.file.name }) : /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-xs text-muted-foreground", children: "Klik untuk pilih file PDF" })
                  ]
                }
              ),
              /* @__PURE__ */ jsxRuntimeExports.jsx(
                "input",
                {
                  ref: fileRef,
                  type: "file",
                  accept: ".pdf,application/pdf",
                  className: "hidden",
                  onChange: (e) => {
                    var _a;
                    const f = ((_a = e.target.files) == null ? void 0 : _a[0]) ?? null;
                    setField("file", f);
                  },
                  "data-ocid": "admin_dokumen.file_input"
                }
              ),
              errors.file && /* @__PURE__ */ jsxRuntimeExports.jsx(
                "p",
                {
                  className: "text-xs text-accent mt-1",
                  "data-ocid": "admin_dokumen.file.field_error",
                  children: errors.file
                }
              ),
              saving && uploadProgress > 0 && uploadProgress < 100 && /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "mt-2", children: [
                /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "h-1.5 bg-muted rounded-full overflow-hidden", children: /* @__PURE__ */ jsxRuntimeExports.jsx(
                  "div",
                  {
                    className: "h-full bg-primary rounded-full transition-all",
                    style: { width: `${uploadProgress}%` }
                  }
                ) }),
                /* @__PURE__ */ jsxRuntimeExports.jsxs("p", { className: "text-xs text-muted-foreground mt-1", children: [
                  "Mengunggah... ",
                  uploadProgress,
                  "%"
                ] })
              ] })
            ] }),
            editId && /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "rounded-lg bg-muted/50 border border-border p-3 space-y-1.5", children: [
              /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-xs font-semibold text-muted-foreground uppercase tracking-wide mb-2", children: "Informasi Perubahan" }),
              /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex items-start gap-2 text-xs text-muted-foreground", children: [
                /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "shrink-0 w-32", children: "Dibuat oleh:" }),
                /* @__PURE__ */ jsxRuntimeExports.jsx(
                  "span",
                  {
                    className: "font-mono break-all text-foreground",
                    title: editAudit.createdBy || "—",
                    children: editAudit.createdBy || "—"
                  }
                )
              ] }),
              /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex items-start gap-2 text-xs text-muted-foreground", children: [
                /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "shrink-0 w-32", children: "Terakhir diubah:" }),
                /* @__PURE__ */ jsxRuntimeExports.jsx(
                  "span",
                  {
                    className: "font-mono break-all text-foreground",
                    title: editAudit.lastModifiedBy || "—",
                    children: editAudit.lastModifiedBy || "—"
                  }
                )
              ] })
            ] })
          ] }),
          /* @__PURE__ */ jsxRuntimeExports.jsxs(DialogFooter, { className: "gap-2", children: [
            /* @__PURE__ */ jsxRuntimeExports.jsx(
              Button,
              {
                variant: "outline",
                onClick: () => setShowForm(false),
                disabled: saving,
                "data-ocid": "admin_dokumen.cancel_button",
                children: "Batal"
              }
            ),
            /* @__PURE__ */ jsxRuntimeExports.jsxs(
              Button,
              {
                onClick: handleSave,
                disabled: saving,
                className: "bg-accent hover:bg-accent/90 text-accent-foreground gap-2",
                "data-ocid": "admin_dokumen.submit_button",
                children: [
                  saving && /* @__PURE__ */ jsxRuntimeExports.jsx(LoaderCircle, { className: "w-4 h-4 animate-spin" }),
                  saving ? "Menyimpan..." : "Simpan"
                ]
              }
            )
          ] })
        ]
      }
    ) }),
    /* @__PURE__ */ jsxRuntimeExports.jsx(
      AlertDialog,
      {
        open: !!deleteTarget,
        onOpenChange: (v) => !v && setDeleteTarget(null),
        children: /* @__PURE__ */ jsxRuntimeExports.jsxs(AlertDialogContent, { "data-ocid": "admin_dokumen.delete_dialog", children: [
          /* @__PURE__ */ jsxRuntimeExports.jsxs(AlertDialogHeader, { children: [
            /* @__PURE__ */ jsxRuntimeExports.jsx(AlertDialogTitle, { children: "Hapus Dokumen?" }),
            /* @__PURE__ */ jsxRuntimeExports.jsxs(AlertDialogDescription, { children: [
              "Tindakan ini tidak dapat dibatalkan. Dokumen",
              " ",
              /* @__PURE__ */ jsxRuntimeExports.jsx("strong", { children: deleteTarget == null ? void 0 : deleteTarget.judul }),
              " akan dihapus secara permanen."
            ] })
          ] }),
          /* @__PURE__ */ jsxRuntimeExports.jsxs(AlertDialogFooter, { children: [
            /* @__PURE__ */ jsxRuntimeExports.jsx(AlertDialogCancel, { "data-ocid": "admin_dokumen.delete_cancel_button", children: "Batal" }),
            /* @__PURE__ */ jsxRuntimeExports.jsx(
              AlertDialogAction,
              {
                onClick: () => deleteTarget && handleDelete(deleteTarget.id),
                className: "bg-accent hover:bg-accent/90 text-accent-foreground",
                "data-ocid": "admin_dokumen.delete_confirm_button",
                children: deleteDokumen.isPending ? /* @__PURE__ */ jsxRuntimeExports.jsx(LoaderCircle, { className: "w-4 h-4 animate-spin" }) : "Hapus"
              }
            )
          ] })
        ] })
      }
    )
  ] });
}
export {
  AdminDokumen
};
