// @caffeine/template-frontend/dist/assets/AdminStatistik-Cc2lNbWM.js
import { c as createLucideIcon, F as FileText, B as BookOpen, N as Newspaper, j as jsxRuntimeExports, a as Button, g as LoadingSpinner } from "./index-B7-RSq-m.js";
import { R as RefreshCw, E as ErrorState } from "./ErrorState-o51Yfgm5.js";
import { P as PageHeader } from "./PageHeader-CmrF1ENQ.js";
import { u as useStatistik } from "./useBackend-xFEMRG87.js";
import { D as Download } from "./download-BnCFvid2.js";
import "./circle-alert-D-S0YTKb.js";
import "./BreadcrumbNav---FAxtpV.js";
import "./chevron-right-RI9IPXg2.js";
import "./useMutation-D--BDafu.js";
/**
 * @license lucide-react v0.511.0 - ISC
 *
 * This source code is licensed under the ISC license.
 * See the LICENSE file in the root directory of this source tree.
 */
const __iconNode = [
  ["path", { d: "M3 3v16a2 2 0 0 0 2 2h16", key: "c24i48" }],
  ["path", { d: "M18 17V9", key: "2bz60n" }],
  ["path", { d: "M13 17V5", key: "1frdt8" }],
  ["path", { d: "M8 17v-3", key: "17ska0" }]
];
const ChartColumn = createLucideIcon("chart-column", __iconNode);
function AdminStatistik() {
  const {
    data: statistik,
    isLoading,
    isError,
    refetch,
    isFetching
  } = useStatistik();
  const STAT_CARDS = [
    {
      icon: FileText,
      label: "Total Dokumen",
      value: Number((statistik == null ? void 0 : statistik.totalDokumen) ?? 0),
      colorBg: "bg-primary/10",
      colorText: "text-primary",
      ocid: "admin_statistik.stat.dokumen"
    },
    {
      icon: Download,
      label: "Total Unduhan",
      value: Number((statistik == null ? void 0 : statistik.totalUnduhan) ?? 0),
      colorBg: "bg-foreground/5",
      colorText: "text-foreground",
      ocid: "admin_statistik.stat.unduhan"
    },
    {
      icon: BookOpen,
      label: "Kategori",
      value: Number((statistik == null ? void 0 : statistik.totalKategori) ?? 0),
      colorBg: "bg-muted",
      colorText: "text-muted-foreground",
      ocid: "admin_statistik.stat.kategori"
    },
    {
      icon: Newspaper,
      label: "Artikel Berita",
      value: Number((statistik == null ? void 0 : statistik.totalArtikel) ?? 0),
      colorBg: "bg-accent/10",
      colorText: "text-accent",
      ocid: "admin_statistik.stat.artikel"
    }
  ];
  return /* @__PURE__ */ jsxRuntimeExports.jsxs(jsxRuntimeExports.Fragment, { children: [
    /* @__PURE__ */ jsxRuntimeExports.jsx(
      PageHeader,
      {
        title: "Statistik",
        description: "Data penggunaan dan ringkasan konten portal JDIH",
        breadcrumbs: [{ label: "Admin", to: "/admin" }, { label: "Statistik" }],
        actions: /* @__PURE__ */ jsxRuntimeExports.jsxs(
          Button,
          {
            variant: "outline",
            size: "sm",
            onClick: () => refetch(),
            disabled: isFetching,
            className: "gap-2 text-sm border-border",
            "data-ocid": "admin_statistik.refresh_button",
            children: [
              /* @__PURE__ */ jsxRuntimeExports.jsx(
                RefreshCw,
                {
                  className: `w-4 h-4 ${isFetching ? "animate-spin" : ""}`
                }
              ),
              "Perbarui"
            ]
          }
        )
      }
    ),
    /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "container mx-auto px-4 py-8", children: isLoading ? /* @__PURE__ */ jsxRuntimeExports.jsx(
      "div",
      {
        className: "flex items-center justify-center py-24",
        "data-ocid": "admin_statistik.loading_state",
        children: /* @__PURE__ */ jsxRuntimeExports.jsx(LoadingSpinner, { size: "lg", text: "Memuat statistik..." })
      }
    ) : isError ? /* @__PURE__ */ jsxRuntimeExports.jsx(ErrorState, { onRetry: () => refetch() }) : /* @__PURE__ */ jsxRuntimeExports.jsxs(jsxRuntimeExports.Fragment, { children: [
      /* @__PURE__ */ jsxRuntimeExports.jsx(
        "div",
        {
          className: "grid grid-cols-2 md:grid-cols-4 gap-4 mb-8",
          "data-ocid": "admin_statistik.stats_section",
          children: STAT_CARDS.map((stat, idx) => /* @__PURE__ */ jsxRuntimeExports.jsxs(
            "div",
            {
              className: "bg-card border border-border rounded-xl p-5",
              "data-ocid": `admin_statistik.stat_card.${idx + 1}`,
              children: [
                /* @__PURE__ */ jsxRuntimeExports.jsx(
                  "div",
                  {
                    className: `w-10 h-10 rounded-lg flex items-center justify-center mb-3 ${stat.colorBg} ${stat.colorText}`,
                    children: /* @__PURE__ */ jsxRuntimeExports.jsx(stat.icon, { className: "w-5 h-5" })
                  }
                ),
                /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "text-2xl font-display font-bold text-foreground mb-1", children: stat.value.toLocaleString("id-ID") }),
                /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "text-xs text-muted-foreground", children: stat.label })
              ]
            },
            stat.label
          ))
        }
      ),
      /* @__PURE__ */ jsxRuntimeExports.jsxs(
        "div",
        {
          className: "bg-card border border-border rounded-xl p-6",
          "data-ocid": "admin_statistik.kategori_chart",
          children: [
            /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex items-center gap-3 mb-6", children: [
              /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "w-9 h-9 rounded-lg bg-primary/10 flex items-center justify-center", children: /* @__PURE__ */ jsxRuntimeExports.jsx(ChartColumn, { className: "w-5 h-5 text-primary" }) }),
              /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { children: [
                /* @__PURE__ */ jsxRuntimeExports.jsx("h2", { className: "font-display font-semibold text-foreground text-base", children: "Dokumen per Kategori" }),
                /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-xs text-muted-foreground", children: "Distribusi dokumen berdasarkan kategori hukum" })
              ] })
            ] }),
            !(statistik == null ? void 0 : statistik.dokumenPerKategori) || statistik.dokumenPerKategori.length === 0 ? /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-sm text-muted-foreground text-center py-8", children: "Belum ada data kategori" }) : /* @__PURE__ */ jsxRuntimeExports.jsx(
              "div",
              {
                className: "space-y-5",
                "data-ocid": "admin_statistik.kategori_bars",
                children: (() => {
                  const maxVal = Math.max(
                    ...statistik.dokumenPerKategori.map((d) => d.jumlah),
                    1
                  );
                  const totalDok = statistik.dokumenPerKategori.reduce(
                    (sum, d) => sum + d.jumlah,
                    0
                  );
                  return statistik.dokumenPerKategori.map((item, idx) => {
                    const pct = Math.round(item.jumlah / maxVal * 100);
                    const porsi = totalDok > 0 ? Math.round(item.jumlah / totalDok * 100) : 0;
                    return /* @__PURE__ */ jsxRuntimeExports.jsxs(
                      "div",
                      {
                        "data-ocid": `admin_statistik.kategori_bar.${idx + 1}`,
                        children: [
                          /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex items-center justify-between text-sm mb-2", children: [
                            /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "text-foreground font-medium", children: item.kategoriNama }),
                            /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex items-center gap-3 text-xs text-muted-foreground", children: [
                              /* @__PURE__ */ jsxRuntimeExports.jsxs("span", { className: "font-mono", children: [
                                item.jumlah,
                                " dokumen"
                              ] }),
                              /* @__PURE__ */ jsxRuntimeExports.jsxs("span", { className: "bg-muted rounded px-1.5 py-0.5", children: [
                                porsi,
                                "%"
                              ] })
                            ] })
                          ] }),
                          /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "h-3 bg-muted rounded-full overflow-hidden", children: /* @__PURE__ */ jsxRuntimeExports.jsx(
                            "div",
                            {
                              className: "h-full bg-primary rounded-full transition-all duration-700",
                              style: { width: `${pct}%` }
                            }
                          ) })
                        ]
                      },
                      item.kategoriId
                    );
                  });
                })()
              }
            )
          ]
        }
      )
    ] }) })
  ] });
}
export {
  AdminStatistik
};
