// @caffeine/template-frontend/dist/assets/AdminDashboard-ko5nAU8s.js
import { c as createLucideIcon, p as usePermissions, j as jsxRuntimeExports, F as FileText, q as FolderOpen, N as Newspaper, b as Link, s as useQuery, h as api } from "./index-B7-RSq-m.js";
import { S as Skeleton } from "./skeleton-Z-jxXZYe.js";
import { u as useStatistik, b as useBerita, h as useActivityLog } from "./useBackend-xFEMRG87.js";
import { D as Download } from "./download-BnCFvid2.js";
import { P as Plus } from "./plus-C7LHpCDy.js";
import "./useMutation-D--BDafu.js";
/**
 * @license lucide-react v0.511.0 - ISC
 *
 * This source code is licensed under the ISC license.
 * See the LICENSE file in the root directory of this source tree.
 */
const __iconNode$1 = [
  [
    "path",
    {
      d: "M22 12h-2.48a2 2 0 0 0-1.93 1.46l-2.35 8.36a.25.25 0 0 1-.48 0L9.24 2.18a.25.25 0 0 0-.48 0l-2.35 8.36A2 2 0 0 1 4.49 12H2",
      key: "169zse"
    }
  ]
];
const Activity = createLucideIcon("activity", __iconNode$1);
/**
 * @license lucide-react v0.511.0 - ISC
 *
 * This source code is licensed under the ISC license.
 * See the LICENSE file in the root directory of this source tree.
 */
const __iconNode = [
  ["path", { d: "M10.268 21a2 2 0 0 0 3.464 0", key: "vwvbt9" }],
  [
    "path",
    {
      d: "M3.262 15.326A1 1 0 0 0 4 17h16a1 1 0 0 0 .74-1.673C19.41 13.956 18 12.499 18 8A6 6 0 0 0 6 8c0 4.499-1.411 5.956-2.738 7.326",
      key: "11g9vi"
    }
  ]
];
const Bell = createLucideIcon("bell", __iconNode);
function formatTimestamp(ts) {
  const ms = typeof ts === "bigint" ? Number(ts / 1000000n) : typeof ts === "string" ? new Date(ts).getTime() : ts;
  const date = new Date(ms);
  return date.toLocaleString("id-ID", {
    day: "2-digit",
    month: "short",
    year: "numeric",
    hour: "2-digit",
    minute: "2-digit"
  });
}
function truncatePrincipal(principal) {
  if (principal.length <= 12) return principal;
  return `${principal.slice(0, 5)}...${principal.slice(-5)}`;
}
function ActionBadge({ action }) {
  const styles = {
    Tambah: "bg-red-100 text-red-700 border border-red-200",
    Edit: "bg-zinc-100 text-zinc-700 border border-zinc-200",
    Hapus: "bg-black text-white"
  };
  const cls = styles[action] ?? "bg-zinc-100 text-zinc-700 border border-zinc-200";
  return /* @__PURE__ */ jsxRuntimeExports.jsx(
    "span",
    {
      className: `inline-block px-2 py-0.5 rounded text-xs font-semibold ${cls}`,
      children: action
    }
  );
}
function StatCard({ label, value, icon: Icon, accent, ocid }) {
  const iconCls = accent === "red" ? "bg-red-50 text-red-600" : "bg-zinc-900 text-white";
  return /* @__PURE__ */ jsxRuntimeExports.jsxs(
    "div",
    {
      className: "bg-white rounded-xl shadow-sm border border-zinc-100 p-5 flex items-center gap-4",
      "data-ocid": ocid,
      children: [
        /* @__PURE__ */ jsxRuntimeExports.jsx(
          "div",
          {
            className: `w-11 h-11 rounded-lg flex items-center justify-center flex-shrink-0 ${iconCls}`,
            children: /* @__PURE__ */ jsxRuntimeExports.jsx(Icon, { className: "w-5 h-5" })
          }
        ),
        /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "min-w-0", children: [
          /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "text-2xl font-bold text-zinc-900 leading-tight", children: typeof value === "number" ? value.toLocaleString("id-ID") : value }),
          /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "text-xs text-zinc-500 mt-0.5 truncate", children: label })
        ] })
      ]
    }
  );
}
function PendingReviewBanner() {
  const { data, isLoading } = useQuery({
    queryKey: ["pendingReviewCount"],
    queryFn: () => api.dokumen.pendingCount(),
    staleTime: 3e4,
    refetchInterval: 6e4
  });
  if (isLoading) {
    return /* @__PURE__ */ jsxRuntimeExports.jsx(
      Skeleton,
      {
        className: "h-20 w-full rounded-xl",
        "data-ocid": "dashboard.pending_review.loading_state"
      }
    );
  }
  const count = (data == null ? void 0 : data.count) ?? 0;
  if (count === 0) {
    return /* @__PURE__ */ jsxRuntimeExports.jsxs(
      "div",
      {
        className: "flex items-center gap-3 px-5 py-3 rounded-xl border border-zinc-200 bg-zinc-50 text-zinc-400",
        "data-ocid": "dashboard.pending_review.empty_state",
        children: [
          /* @__PURE__ */ jsxRuntimeExports.jsx(Bell, { className: "w-4 h-4 flex-shrink-0" }),
          /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "text-sm", children: "Tidak ada dokumen menunggu review" })
        ]
      }
    );
  }
  return /* @__PURE__ */ jsxRuntimeExports.jsxs(
    "div",
    {
      className: "flex items-center justify-between gap-4 px-5 py-4 rounded-xl border border-orange-200 bg-orange-50",
      "data-ocid": "dashboard.pending_review.card",
      children: [
        /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex items-center gap-3 min-w-0", children: [
          /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "w-10 h-10 rounded-lg bg-orange-100 flex items-center justify-center flex-shrink-0", children: /* @__PURE__ */ jsxRuntimeExports.jsx(Bell, { className: "w-5 h-5 text-orange-600" }) }),
          /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "min-w-0", children: [
            /* @__PURE__ */ jsxRuntimeExports.jsxs("p", { className: "text-sm font-semibold text-orange-900", children: [
              count.toLocaleString("id-ID"),
              " Dokumen Menunggu Review"
            ] }),
            /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-xs text-orange-600 mt-0.5", children: "Perlu diperiksa dan diverifikasi sebelum diterbitkan" })
          ] })
        ] }),
        /* @__PURE__ */ jsxRuntimeExports.jsx(
          "a",
          {
            href: "/admin/dokumen?filter=pending-review",
            "data-ocid": "dashboard.pending_review.link",
            className: "flex-shrink-0 px-4 py-2 rounded-lg bg-orange-600 hover:bg-orange-700 text-white text-xs font-semibold transition-colors whitespace-nowrap",
            children: "Lihat Dokumen"
          }
        )
      ]
    }
  );
}
function ActivityRow({ log, index }) {
  return /* @__PURE__ */ jsxRuntimeExports.jsxs(
    "div",
    {
      className: "flex items-start gap-3 py-3 border-b border-zinc-100 last:border-0",
      "data-ocid": `dashboard.activity.item.${index}`,
      children: [
        /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "mt-0.5", children: /* @__PURE__ */ jsxRuntimeExports.jsx(ActionBadge, { action: log.action }) }),
        /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex-1 min-w-0", children: [
          /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-sm text-zinc-800 font-medium truncate", children: log.targetTitle }),
          /* @__PURE__ */ jsxRuntimeExports.jsxs("p", { className: "text-xs text-zinc-400 mt-0.5", children: [
            /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "text-zinc-500", children: log.targetType }),
            " · ",
            /* @__PURE__ */ jsxRuntimeExports.jsx("span", { title: log.performedBy, className: "font-mono", children: truncatePrincipal(log.performedBy) })
          ] })
        ] }),
        /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "text-xs text-zinc-400 whitespace-nowrap flex-shrink-0 pt-0.5", children: formatTimestamp(log.performedAt) })
      ]
    }
  );
}
function KategoriBar({ nama, count, percentage, index }) {
  return /* @__PURE__ */ jsxRuntimeExports.jsxs(
    "div",
    {
      className: "flex items-center gap-3",
      "data-ocid": `dashboard.chart.item.${index}`,
      children: [
        /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "w-32 text-xs text-zinc-600 text-right truncate flex-shrink-0", children: nama }),
        /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "flex-1 bg-zinc-100 rounded-full h-3 overflow-hidden", children: /* @__PURE__ */ jsxRuntimeExports.jsx(
          "div",
          {
            className: "h-full bg-red-600 rounded-full transition-all duration-700",
            style: { width: `${percentage}%` }
          }
        ) }),
        /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "w-10 text-xs text-zinc-500 text-right flex-shrink-0", children: count })
      ]
    }
  );
}
function AdminDashboard() {
  const { data: statistik, isLoading: loadingStats } = useStatistik();
  const { data: beritaList, isLoading: loadingBerita } = useBerita();
  const { data: activityLog, isLoading: loadingActivity } = useActivityLog();
  const { isAdmin } = usePermissions();
  const totalBerita = (beritaList == null ? void 0 : beritaList.length) ?? 0;
  const perKategori = (statistik == null ? void 0 : statistik.dokumenPerKategori) ?? [];
  const totalDokumenKategori = perKategori.reduce((s, k) => s + k.jumlah, 0) || 1;
  const chartData = perKategori.slice(0, 7);
  return /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "p-6 space-y-6 bg-zinc-50 min-h-full", children: [
    /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { "data-ocid": "dashboard.page", children: [
      /* @__PURE__ */ jsxRuntimeExports.jsx("h1", { className: "text-xl font-bold text-zinc-900", children: "Dashboard" }),
      /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-sm text-zinc-500 mt-1", children: "Ringkasan data portal JDIH UNTAG Banyuwangi" })
    ] }),
    isAdmin && /* @__PURE__ */ jsxRuntimeExports.jsx("div", { "data-ocid": "dashboard.pending_review_section", children: /* @__PURE__ */ jsxRuntimeExports.jsx(PendingReviewBanner, {}) }),
    /* @__PURE__ */ jsxRuntimeExports.jsx(
      "div",
      {
        className: "grid grid-cols-2 lg:grid-cols-4 gap-4",
        "data-ocid": "dashboard.stats_section",
        children: loadingStats ? Array.from({ length: 4 }, (_, i) => `stat-skel-${i}`).map((id) => /* @__PURE__ */ jsxRuntimeExports.jsx(Skeleton, { className: "h-24 rounded-xl" }, id)) : /* @__PURE__ */ jsxRuntimeExports.jsxs(jsxRuntimeExports.Fragment, { children: [
          /* @__PURE__ */ jsxRuntimeExports.jsx(
            StatCard,
            {
              label: "Total Dokumen",
              value: (statistik == null ? void 0 : statistik.totalDokumen) ?? 0,
              icon: FileText,
              accent: "red",
              ocid: "dashboard.stat.dokumen"
            }
          ),
          /* @__PURE__ */ jsxRuntimeExports.jsx(
            StatCard,
            {
              label: "Total Unduhan",
              value: (statistik == null ? void 0 : statistik.totalUnduhan) ?? 0,
              icon: Download,
              accent: "black",
              ocid: "dashboard.stat.unduhan"
            }
          ),
          /* @__PURE__ */ jsxRuntimeExports.jsx(
            StatCard,
            {
              label: "Total Kategori",
              value: (statistik == null ? void 0 : statistik.totalKategori) ?? 0,
              icon: FolderOpen,
              accent: "red",
              ocid: "dashboard.stat.kategori"
            }
          ),
          /* @__PURE__ */ jsxRuntimeExports.jsx(
            StatCard,
            {
              label: "Total Berita",
              value: loadingBerita ? "—" : totalBerita,
              icon: Newspaper,
              accent: "black",
              ocid: "dashboard.stat.berita"
            }
          )
        ] })
      }
    ),
    /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "grid grid-cols-1 lg:grid-cols-3 gap-6", children: [
      /* @__PURE__ */ jsxRuntimeExports.jsxs(
        "div",
        {
          className: "lg:col-span-2 bg-white rounded-xl shadow-sm border border-zinc-100 p-5",
          "data-ocid": "dashboard.activity_section",
          children: [
            /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex items-center gap-2 mb-4", children: [
              /* @__PURE__ */ jsxRuntimeExports.jsx(Activity, { className: "w-4 h-4 text-red-600" }),
              /* @__PURE__ */ jsxRuntimeExports.jsx("h2", { className: "text-base font-semibold text-zinc-900", children: "Aktivitas Terbaru" })
            ] }),
            loadingActivity ? /* @__PURE__ */ jsxRuntimeExports.jsx(
              "div",
              {
                className: "space-y-3",
                "data-ocid": "dashboard.activity.loading_state",
                children: Array.from({ length: 5 }, (_, i) => `act-skel-${i}`).map(
                  (id) => /* @__PURE__ */ jsxRuntimeExports.jsx(Skeleton, { className: "h-12 w-full rounded-lg" }, id)
                )
              }
            ) : !activityLog || activityLog.length === 0 ? /* @__PURE__ */ jsxRuntimeExports.jsxs(
              "div",
              {
                className: "flex flex-col items-center justify-center py-10 text-center",
                "data-ocid": "dashboard.activity.empty_state",
                children: [
                  /* @__PURE__ */ jsxRuntimeExports.jsx(Activity, { className: "w-10 h-10 text-zinc-300 mb-2" }),
                  /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-sm text-zinc-400", children: "Belum ada aktivitas" })
                ]
              }
            ) : /* @__PURE__ */ jsxRuntimeExports.jsx("div", { children: activityLog.slice(0, 10).map((log, i) => /* @__PURE__ */ jsxRuntimeExports.jsx(ActivityRow, { log, index: i + 1 }, String(log.id))) })
          ]
        }
      ),
      /* @__PURE__ */ jsxRuntimeExports.jsxs(
        "div",
        {
          className: "bg-white rounded-xl shadow-sm border border-zinc-100 p-5",
          "data-ocid": "dashboard.quick_actions_section",
          children: [
            /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex items-center gap-2 mb-4", children: [
              /* @__PURE__ */ jsxRuntimeExports.jsx(Plus, { className: "w-4 h-4 text-red-600" }),
              /* @__PURE__ */ jsxRuntimeExports.jsx("h2", { className: "text-base font-semibold text-zinc-900", children: "Aksi Cepat" })
            ] }),
            /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "space-y-2", children: [
              /* @__PURE__ */ jsxRuntimeExports.jsxs(
                Link,
                {
                  to: "/admin/dokumen",
                  "data-ocid": "dashboard.tambah_dokumen_button",
                  className: "flex items-center gap-3 w-full px-4 py-3 rounded-lg border border-zinc-200 text-sm font-medium text-zinc-800 hover:border-red-300 hover:bg-red-50 hover:text-red-700 transition-colors group",
                  children: [
                    /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "w-8 h-8 rounded-md bg-red-50 text-red-600 flex items-center justify-center group-hover:bg-red-100 transition-colors", children: /* @__PURE__ */ jsxRuntimeExports.jsx(FileText, { className: "w-4 h-4" }) }),
                    "Tambah Dokumen"
                  ]
                }
              ),
              /* @__PURE__ */ jsxRuntimeExports.jsxs(
                Link,
                {
                  to: "/admin/berita",
                  "data-ocid": "dashboard.tambah_berita_button",
                  className: "flex items-center gap-3 w-full px-4 py-3 rounded-lg border border-zinc-200 text-sm font-medium text-zinc-800 hover:border-red-300 hover:bg-red-50 hover:text-red-700 transition-colors group",
                  children: [
                    /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "w-8 h-8 rounded-md bg-zinc-100 text-zinc-600 flex items-center justify-center group-hover:bg-red-100 group-hover:text-red-600 transition-colors", children: /* @__PURE__ */ jsxRuntimeExports.jsx(Newspaper, { className: "w-4 h-4" }) }),
                    "Tambah Berita"
                  ]
                }
              ),
              /* @__PURE__ */ jsxRuntimeExports.jsxs(
                Link,
                {
                  to: "/admin/kategori",
                  "data-ocid": "dashboard.kelola_kategori_button",
                  className: "flex items-center gap-3 w-full px-4 py-3 rounded-lg border border-zinc-200 text-sm font-medium text-zinc-800 hover:border-red-300 hover:bg-red-50 hover:text-red-700 transition-colors group",
                  children: [
                    /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "w-8 h-8 rounded-md bg-zinc-100 text-zinc-600 flex items-center justify-center group-hover:bg-red-100 group-hover:text-red-600 transition-colors", children: /* @__PURE__ */ jsxRuntimeExports.jsx(FolderOpen, { className: "w-4 h-4" }) }),
                    "Kelola Kategori"
                  ]
                }
              )
            ] }),
            /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "mt-5 pt-4 border-t border-zinc-100", children: [
              /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-xs text-zinc-400 mb-3 font-medium uppercase tracking-wide", children: "Ringkasan Dokumen" }),
              loadingStats ? /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "space-y-2", children: [
                /* @__PURE__ */ jsxRuntimeExports.jsx(Skeleton, { className: "h-4 w-full rounded" }),
                /* @__PURE__ */ jsxRuntimeExports.jsx(Skeleton, { className: "h-4 w-4/5 rounded" })
              ] }) : /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "space-y-1.5", children: [
                /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex justify-between text-xs", children: [
                  /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "text-zinc-500", children: "Total dokumen" }),
                  /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "font-semibold text-zinc-800", children: ((statistik == null ? void 0 : statistik.totalDokumen) ?? 0).toLocaleString("id-ID") })
                ] }),
                /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex justify-between text-xs", children: [
                  /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "text-zinc-500", children: "Total unduhan" }),
                  /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "font-semibold text-zinc-800", children: ((statistik == null ? void 0 : statistik.totalUnduhan) ?? 0).toLocaleString("id-ID") })
                ] }),
                /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex justify-between text-xs", children: [
                  /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "text-zinc-500", children: "Total berita" }),
                  /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "font-semibold text-zinc-800", children: totalBerita })
                ] })
              ] })
            ] })
          ]
        }
      )
    ] }),
    /* @__PURE__ */ jsxRuntimeExports.jsxs(
      "div",
      {
        className: "bg-white rounded-xl shadow-sm border border-zinc-100 p-5",
        "data-ocid": "dashboard.chart_section",
        children: [
          /* @__PURE__ */ jsxRuntimeExports.jsx("h2", { className: "text-base font-semibold text-zinc-900 mb-5", children: "Dokumen per Kategori" }),
          loadingStats ? /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "space-y-3", "data-ocid": "dashboard.chart.loading_state", children: Array.from({ length: 5 }, (_, i) => `chart-skel-${i}`).map(
            (id) => /* @__PURE__ */ jsxRuntimeExports.jsx(Skeleton, { className: "h-5 w-full rounded" }, id)
          ) }) : chartData.length === 0 ? /* @__PURE__ */ jsxRuntimeExports.jsxs(
            "div",
            {
              className: "flex flex-col items-center justify-center py-8 text-center",
              "data-ocid": "dashboard.chart.empty_state",
              children: [
                /* @__PURE__ */ jsxRuntimeExports.jsx(FolderOpen, { className: "w-8 h-8 text-zinc-300 mb-2" }),
                /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-sm text-zinc-400", children: "Belum ada data kategori" })
              ]
            }
          ) : /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "space-y-3", children: chartData.map((k, i) => {
            const pct = Math.round(k.jumlah / totalDokumenKategori * 100);
            return /* @__PURE__ */ jsxRuntimeExports.jsx(
              KategoriBar,
              {
                nama: k.kategoriNama,
                count: k.jumlah,
                percentage: pct,
                index: i + 1
              },
              k.kategoriId
            );
          }) })
        ]
      }
    )
  ] });
}
export {
  AdminDashboard
};
