// @caffeine/template-frontend/dist/assets/AdminKategori-Cj3PZwat.js
import { t as useQueryClient, r as reactExports, j as jsxRuntimeExports, a as Button, e as SkeletonList, v as LoaderCircle, h as api } from "./index-B7-RSq-m.js";
import { A as AlertDialog, a as AlertDialogContent, b as AlertDialogHeader, c as AlertDialogTitle, d as AlertDialogDescription, e as AlertDialogFooter, f as AlertDialogCancel, g as AlertDialogAction } from "./alert-dialog-lTbm3t0t.js";
import { D as Dialog, a as DialogContent, b as DialogHeader, c as DialogTitle, d as DialogFooter } from "./dialog-CIgZYBSq.js";
import { I as Input } from "./input-B3nX7k9Z.js";
import { L as Label } from "./label-eF2rHvnY.js";
import { u as ue } from "./index-B0YpMaU8.js";
import { E as EmptyState } from "./EmptyState-Bzc48I5U.js";
import { E as ErrorState } from "./ErrorState-o51Yfgm5.js";
import { P as PageHeader } from "./PageHeader-CmrF1ENQ.js";
import { d as useKategori, n as useDeleteKategori } from "./useBackend-xFEMRG87.js";
import { P as Plus } from "./plus-C7LHpCDy.js";
import { P as Pencil } from "./pencil-CwCkT-7p.js";
import { T as Trash2 } from "./trash-2-CnhMEsa2.js";
import "./index-BcOIygc1.js";
import "./circle-alert-D-S0YTKb.js";
import "./BreadcrumbNav---FAxtpV.js";
import "./chevron-right-RI9IPXg2.js";
import "./useMutation-D--BDafu.js";
function validate(form) {
  const errs = {};
  if (!form.nama.trim()) errs.nama = "Nama kategori wajib diisi";
  return errs;
}
function AdminKategori() {
  const qc = useQueryClient();
  const { data: kategoriList, isLoading, isError, refetch } = useKategori();
  const deleteKategori = useDeleteKategori();
  const [showForm, setShowForm] = reactExports.useState(false);
  const [editId, setEditId] = reactExports.useState(null);
  const [form, setForm] = reactExports.useState({ nama: "" });
  const [errors, setErrors] = reactExports.useState({});
  const [saving, setSaving] = reactExports.useState(false);
  const [deleteTarget, setDeleteTarget] = reactExports.useState(null);
  function openCreate() {
    setForm({ nama: "" });
    setErrors({});
    setEditId(null);
    setShowForm(true);
  }
  function openEdit(id, nama) {
    setForm({ nama });
    setErrors({});
    setEditId(id);
    setShowForm(true);
  }
  function setField(val) {
    setForm({ nama: val });
    if (errors.nama) setErrors({});
  }
  async function handleSave() {
    const errs = validate(form);
    if (Object.keys(errs).length > 0) {
      setErrors(errs);
      return;
    }
    setSaving(true);
    try {
      if (editId) {
        await api.kategori.update(Number(editId), form.nama.trim());
        ue.success("Kategori berhasil diperbarui");
      } else {
        await api.kategori.add(form.nama.trim());
        ue.success("Kategori berhasil ditambahkan");
      }
      qc.invalidateQueries({ queryKey: ["kategori"] });
      qc.invalidateQueries({ queryKey: ["statistik"] });
      setShowForm(false);
    } catch {
      ue.error("Gagal menyimpan kategori. Silakan coba lagi.");
    } finally {
      setSaving(false);
    }
  }
  async function handleDelete(id) {
    try {
      await deleteKategori.mutateAsync(Number(id));
      ue.success("Kategori berhasil dihapus");
    } catch {
      ue.error("Gagal menghapus kategori");
    } finally {
      setDeleteTarget(null);
    }
  }
  return /* @__PURE__ */ jsxRuntimeExports.jsxs(jsxRuntimeExports.Fragment, { children: [
    /* @__PURE__ */ jsxRuntimeExports.jsx(
      PageHeader,
      {
        title: "Kelola Kategori",
        description: "Kelola jenis produk hukum yang tersedia di portal",
        breadcrumbs: [{ label: "Admin", to: "/admin" }, { label: "Kategori" }],
        actions: /* @__PURE__ */ jsxRuntimeExports.jsxs(
          Button,
          {
            onClick: openCreate,
            className: "bg-accent hover:bg-accent/90 text-accent-foreground gap-2 text-sm",
            "data-ocid": "admin_kategori.tambah_button",
            children: [
              /* @__PURE__ */ jsxRuntimeExports.jsx(Plus, { className: "w-4 h-4" }),
              "Tambah Kategori"
            ]
          }
        )
      }
    ),
    /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "container mx-auto px-4 py-6", children: isLoading ? /* @__PURE__ */ jsxRuntimeExports.jsx(SkeletonList, { count: 4 }) : isError ? /* @__PURE__ */ jsxRuntimeExports.jsx(ErrorState, { onRetry: () => refetch() }) : !kategoriList || kategoriList.length === 0 ? /* @__PURE__ */ jsxRuntimeExports.jsx(
      EmptyState,
      {
        title: "Belum ada kategori",
        description: "Tambahkan kategori produk hukum pertama.",
        actionLabel: "Tambah Kategori",
        onAction: openCreate,
        "data-ocid": "admin_kategori.empty_state"
      }
    ) : /* @__PURE__ */ jsxRuntimeExports.jsx(
      "div",
      {
        className: "grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-3 gap-4",
        "data-ocid": "admin_kategori.kategori_list",
        children: kategoriList.map((kat, idx) => /* @__PURE__ */ jsxRuntimeExports.jsxs(
          "div",
          {
            className: "bg-card border border-border rounded-xl p-5 transition-all hover:border-primary/30 hover:shadow-sm",
            "data-ocid": `admin_kategori.kategori_card.${idx + 1}`,
            children: [
              /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex items-start justify-between gap-2 mb-2", children: [
                /* @__PURE__ */ jsxRuntimeExports.jsx("h3", { className: "font-semibold text-foreground text-sm leading-snug flex-1", children: kat.nama }),
                /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex items-center gap-1 shrink-0", children: [
                  /* @__PURE__ */ jsxRuntimeExports.jsx(
                    Button,
                    {
                      variant: "ghost",
                      size: "icon",
                      className: "h-7 w-7 hover:bg-secondary",
                      onClick: () => openEdit(kat.id, kat.nama),
                      "data-ocid": `admin_kategori.edit_button.${idx + 1}`,
                      children: /* @__PURE__ */ jsxRuntimeExports.jsx(Pencil, { className: "w-3.5 h-3.5" })
                    }
                  ),
                  /* @__PURE__ */ jsxRuntimeExports.jsx(
                    Button,
                    {
                      variant: "ghost",
                      size: "icon",
                      className: "h-7 w-7 text-accent hover:text-accent hover:bg-accent/5",
                      onClick: () => setDeleteTarget({ id: kat.id, nama: kat.nama }),
                      "data-ocid": `admin_kategori.delete_button.${idx + 1}`,
                      children: /* @__PURE__ */ jsxRuntimeExports.jsx(Trash2, { className: "w-3.5 h-3.5" })
                    }
                  )
                ] })
              ] }),
              kat.deskripsi && /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-xs text-muted-foreground mb-3 line-clamp-2", children: kat.deskripsi }),
              /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "text-xs font-medium text-primary bg-primary/5 rounded px-2 py-1 inline-block", children: [
                kat.jumlahDokumen,
                " dokumen"
              ] })
            ]
          },
          kat.id
        ))
      }
    ) }),
    /* @__PURE__ */ jsxRuntimeExports.jsx(Dialog, { open: showForm, onOpenChange: (v) => !saving && setShowForm(v), children: /* @__PURE__ */ jsxRuntimeExports.jsxs(DialogContent, { className: "max-w-sm", "data-ocid": "admin_kategori.dialog", children: [
      /* @__PURE__ */ jsxRuntimeExports.jsx(DialogHeader, { children: /* @__PURE__ */ jsxRuntimeExports.jsx(DialogTitle, { className: "font-display", children: editId ? "Edit Kategori" : "Tambah Kategori" }) }),
      /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "py-2", children: [
        /* @__PURE__ */ jsxRuntimeExports.jsxs(Label, { htmlFor: "kat-nama", className: "text-sm font-medium", children: [
          "Nama Kategori ",
          /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "text-accent", children: "*" })
        ] }),
        /* @__PURE__ */ jsxRuntimeExports.jsx(
          Input,
          {
            id: "kat-nama",
            value: form.nama,
            onChange: (e) => setField(e.target.value),
            placeholder: "cth. Peraturan Rektor",
            className: "mt-1 text-sm",
            "data-ocid": "admin_kategori.nama_input",
            onKeyDown: (e) => e.key === "Enter" && handleSave()
          }
        ),
        errors.nama && /* @__PURE__ */ jsxRuntimeExports.jsx(
          "p",
          {
            className: "text-xs text-accent mt-1",
            "data-ocid": "admin_kategori.nama.field_error",
            children: errors.nama
          }
        )
      ] }),
      /* @__PURE__ */ jsxRuntimeExports.jsxs(DialogFooter, { className: "gap-2", children: [
        /* @__PURE__ */ jsxRuntimeExports.jsx(
          Button,
          {
            variant: "outline",
            onClick: () => setShowForm(false),
            disabled: saving,
            "data-ocid": "admin_kategori.cancel_button",
            children: "Batal"
          }
        ),
        /* @__PURE__ */ jsxRuntimeExports.jsxs(
          Button,
          {
            onClick: handleSave,
            disabled: saving,
            className: "bg-accent hover:bg-accent/90 text-accent-foreground gap-2",
            "data-ocid": "admin_kategori.submit_button",
            children: [
              saving && /* @__PURE__ */ jsxRuntimeExports.jsx(LoaderCircle, { className: "w-4 h-4 animate-spin" }),
              saving ? "Menyimpan..." : "Simpan"
            ]
          }
        )
      ] })
    ] }) }),
    /* @__PURE__ */ jsxRuntimeExports.jsx(
      AlertDialog,
      {
        open: !!deleteTarget,
        onOpenChange: (v) => !v && setDeleteTarget(null),
        children: /* @__PURE__ */ jsxRuntimeExports.jsxs(AlertDialogContent, { "data-ocid": "admin_kategori.delete_dialog", children: [
          /* @__PURE__ */ jsxRuntimeExports.jsxs(AlertDialogHeader, { children: [
            /* @__PURE__ */ jsxRuntimeExports.jsx(AlertDialogTitle, { children: "Hapus Kategori?" }),
            /* @__PURE__ */ jsxRuntimeExports.jsxs(AlertDialogDescription, { children: [
              "Kategori ",
              /* @__PURE__ */ jsxRuntimeExports.jsx("strong", { children: deleteTarget == null ? void 0 : deleteTarget.nama }),
              " akan dihapus. Dokumen yang terkait kategori ini mungkin terpengaruh."
            ] })
          ] }),
          /* @__PURE__ */ jsxRuntimeExports.jsxs(AlertDialogFooter, { children: [
            /* @__PURE__ */ jsxRuntimeExports.jsx(AlertDialogCancel, { "data-ocid": "admin_kategori.delete_cancel_button", children: "Batal" }),
            /* @__PURE__ */ jsxRuntimeExports.jsx(
              AlertDialogAction,
              {
                onClick: () => deleteTarget && handleDelete(deleteTarget.id),
                className: "bg-accent hover:bg-accent/90 text-accent-foreground",
                "data-ocid": "admin_kategori.delete_confirm_button",
                children: deleteKategori.isPending ? /* @__PURE__ */ jsxRuntimeExports.jsx(LoaderCircle, { className: "w-4 h-4 animate-spin" }) : "Hapus"
              }
            )
          ] })
        ] })
      }
    )
  ] });
}
export {
  AdminKategori
};
