// @caffeine/template-frontend/dist/assets/AdminBerita-DfQNu8i-.js
import { p as usePermissions, t as useQueryClient, r as reactExports, j as jsxRuntimeExports, a as Button, e as SkeletonList, I as Image, v as LoaderCircle, X, h as api } from "./index-B7-RSq-m.js";
import { A as AlertDialog, a as AlertDialogContent, b as AlertDialogHeader, c as AlertDialogTitle, d as AlertDialogDescription, e as AlertDialogFooter, f as AlertDialogCancel, g as AlertDialogAction } from "./alert-dialog-lTbm3t0t.js";
import { B as Badge } from "./badge-KXcDp_20.js";
import { D as Dialog, a as DialogContent, b as DialogHeader, c as DialogTitle, d as DialogFooter } from "./dialog-CIgZYBSq.js";
import { I as Input } from "./input-B3nX7k9Z.js";
import { L as Label } from "./label-eF2rHvnY.js";
import { S as Select, a as SelectTrigger, b as SelectValue, c as SelectContent, d as SelectItem } from "./select-37c5ocl_.js";
import { T as Textarea, U as Upload } from "./textarea-DRo76BEL.js";
import { u as ue } from "./index-B0YpMaU8.js";
import { E as EmptyState } from "./EmptyState-Bzc48I5U.js";
import { E as ErrorState } from "./ErrorState-o51Yfgm5.js";
import { b as useBerita, o as useDeleteBerita } from "./useBackend-xFEMRG87.js";
import { P as Plus } from "./plus-C7LHpCDy.js";
import { C as Calendar } from "./calendar-mZT-2HLQ.js";
import { U as User } from "./user-CWnTlCz9.js";
import { T as Tag } from "./tag-DZ8TZA2a.js";
import { U as UserCheck } from "./user-check-BV1RhZlN.js";
import { P as Pencil } from "./pencil-CwCkT-7p.js";
import { T as Trash2 } from "./trash-2-CnhMEsa2.js";
import "./index-BcOIygc1.js";
import "./index-CUotUQbC.js";
import "./circle-alert-D-S0YTKb.js";
import "./useMutation-D--BDafu.js";
function truncatePrincipal(principal) {
  if (!principal) return "-";
  if (principal.length <= 12) return principal;
  return `${principal.slice(0, 12)}...`;
}
const EMPTY_FORM = {
  judul: "",
  author: "",
  konten: "",
  ringkasan: "",
  tagsInput: "",
  status: "Terbit",
  tanggalPublikasi: ""
};
function validate(form) {
  const errs = {};
  if (!form.judul.trim()) errs.judul = "Judul wajib diisi";
  if (!form.author.trim()) errs.author = "Nama penulis wajib diisi";
  if (!form.konten.trim()) errs.konten = "Konten artikel wajib diisi";
  return errs;
}
function formatTanggal(ts) {
  if (!ts) return "-";
  try {
    let date;
    if (typeof ts === "string") {
      date = new Date(ts);
    } else {
      const ms = Number(typeof ts === "bigint" ? ts / 1000000n : ts);
      if (ms === 0) return "-";
      date = new Date(ms);
    }
    return date.toLocaleDateString("id-ID", {
      day: "numeric",
      month: "long",
      year: "numeric"
    });
  } catch {
    return "-";
  }
}
function parseTags(input) {
  return input.split(",").map((t) => t.trim()).filter(Boolean);
}
function ImageUploadWidget({
  imageFile,
  existingImageUrl,
  uploadProgress,
  isUploading,
  error,
  onFileSelect,
  onClearExisting
}) {
  const fileRef = reactExports.useRef(null);
  const previewUrl = imageFile ? URL.createObjectURL(imageFile) : null;
  const displayUrl = previewUrl ?? (existingImageUrl || null);
  return /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { children: [
    /* @__PURE__ */ jsxRuntimeExports.jsx(Label, { className: "text-sm font-medium", children: "Gambar Sampul" }),
    /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "mt-1", children: [
      displayUrl ? /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "relative rounded-lg overflow-hidden border border-border", children: [
        /* @__PURE__ */ jsxRuntimeExports.jsx(
          "img",
          {
            src: displayUrl,
            alt: "Pratinjau gambar",
            className: "w-full h-40 object-cover"
          }
        ),
        /* @__PURE__ */ jsxRuntimeExports.jsx(
          "button",
          {
            type: "button",
            onClick: () => {
              onFileSelect(null);
              onClearExisting();
              if (fileRef.current) fileRef.current.value = "";
            },
            className: "absolute top-2 right-2 bg-background/80 hover:bg-background rounded-full p-1 border border-border transition-colors",
            "aria-label": "Hapus gambar",
            "data-ocid": "admin_berita.gambar_remove_button",
            children: /* @__PURE__ */ jsxRuntimeExports.jsx(X, { className: "w-4 h-4 text-foreground" })
          }
        ),
        imageFile && /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "absolute bottom-0 left-0 right-0 bg-background/80 px-3 py-1.5 text-xs text-foreground truncate", children: imageFile.name })
      ] }) : /* @__PURE__ */ jsxRuntimeExports.jsxs(
        "button",
        {
          type: "button",
          className: "w-full border-2 border-dashed border-border rounded-lg p-5 text-center cursor-pointer hover:border-primary/40 transition-colors",
          onClick: () => {
            var _a;
            return (_a = fileRef.current) == null ? void 0 : _a.click();
          },
          "data-ocid": "admin_berita.gambar_upload_button",
          children: [
            isUploading ? /* @__PURE__ */ jsxRuntimeExports.jsx(LoaderCircle, { className: "w-6 h-6 text-muted-foreground mx-auto mb-1 animate-spin" }) : /* @__PURE__ */ jsxRuntimeExports.jsx(Image, { className: "w-6 h-6 text-muted-foreground mx-auto mb-1" }),
            /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-xs text-muted-foreground", children: "Klik untuk pilih gambar (JPG, PNG, WebP)" })
          ]
        }
      ),
      /* @__PURE__ */ jsxRuntimeExports.jsx(
        "input",
        {
          ref: fileRef,
          type: "file",
          accept: "image/*",
          className: "hidden",
          onChange: (e) => {
            var _a;
            const f = ((_a = e.target.files) == null ? void 0 : _a[0]) ?? null;
            onFileSelect(f);
          },
          "data-ocid": "admin_berita.gambar_file_input"
        }
      ),
      isUploading && uploadProgress > 0 && uploadProgress < 100 && /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "mt-2", children: [
        /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "h-1.5 bg-muted rounded-full overflow-hidden", children: /* @__PURE__ */ jsxRuntimeExports.jsx(
          "div",
          {
            className: "h-full bg-primary rounded-full transition-all",
            style: { width: `${uploadProgress}%` }
          }
        ) }),
        /* @__PURE__ */ jsxRuntimeExports.jsxs("p", { className: "text-xs text-muted-foreground mt-1", children: [
          "Mengunggah gambar... ",
          uploadProgress,
          "%"
        ] })
      ] }),
      displayUrl && /* @__PURE__ */ jsxRuntimeExports.jsxs(
        "button",
        {
          type: "button",
          className: "mt-2 flex items-center gap-1.5 text-xs text-muted-foreground hover:text-foreground transition-colors",
          onClick: () => {
            var _a;
            return (_a = fileRef.current) == null ? void 0 : _a.click();
          },
          "data-ocid": "admin_berita.gambar_change_button",
          children: [
            /* @__PURE__ */ jsxRuntimeExports.jsx(Upload, { className: "w-3.5 h-3.5" }),
            "Ganti gambar"
          ]
        }
      )
    ] }),
    error && /* @__PURE__ */ jsxRuntimeExports.jsx(
      "p",
      {
        className: "text-xs text-accent mt-1",
        "data-ocid": "admin_berita.gambar.field_error",
        children: error
      }
    )
  ] });
}
function AdminBerita() {
  const { isAdmin, grantedMenus } = usePermissions();
  const qc = useQueryClient();
  const { data: artikelList, isLoading, isError, refetch } = useBerita();
  const deleteBerita = useDeleteBerita();
  const [showForm, setShowForm] = reactExports.useState(false);
  const [editId, setEditId] = reactExports.useState(null);
  const [editAudit, setEditAudit] = reactExports.useState({
    createdBy: "",
    lastModifiedBy: ""
  });
  const [form, setForm] = reactExports.useState(EMPTY_FORM);
  const [errors, setErrors] = reactExports.useState({});
  const [saving, setSaving] = reactExports.useState(false);
  const [imageFile, setImageFile] = reactExports.useState(null);
  const [existingImageUrl, setExistingImageUrl] = reactExports.useState("");
  const [uploadProgress, setUploadProgress] = reactExports.useState(0);
  const [isUploading, setIsUploading] = reactExports.useState(false);
  const [deleteTarget, setDeleteTarget] = reactExports.useState(null);
  function openCreate() {
    setForm(EMPTY_FORM);
    setErrors({});
    setEditId(null);
    setEditAudit({ createdBy: "", lastModifiedBy: "" });
    setImageFile(null);
    setExistingImageUrl("");
    setUploadProgress(0);
    setShowForm(true);
  }
  function openEdit(id, artikel) {
    setForm({
      judul: artikel.judul,
      author: artikel.author,
      konten: artikel.konten,
      ringkasan: artikel.ringkasan ?? "",
      tagsInput: (artikel.tags ?? []).join(", "),
      status: artikel.status ?? "Terbit",
      tanggalPublikasi: ""
    });
    setErrors({});
    setEditId(id);
    setEditAudit({
      createdBy: artikel.createdBy ?? "",
      lastModifiedBy: artikel.lastModifiedBy ?? ""
    });
    setImageFile(null);
    setExistingImageUrl(artikel.gambar ?? "");
    setUploadProgress(0);
    setShowForm(true);
  }
  function setField(key, val) {
    setForm((prev) => ({ ...prev, [key]: val }));
    if (key in errors) setErrors((prev) => ({ ...prev, [key]: void 0 }));
  }
  function getSelectedImageFile() {
    return imageFile || void 0;
  }
  async function handleSave() {
    const errs = validate(form);
    if (Object.keys(errs).length > 0) {
      setErrors(errs);
      return;
    }
    setSaving(true);
    try {
      const gambar = getSelectedImageFile();
      const input = {
        judul: form.judul.trim(),
        author: form.author.trim(),
        konten: form.konten.trim(),
        ringkasan: form.ringkasan.trim(),
        gambar,
        tags: parseTags(form.tagsInput),
        status: form.status || "Terbit",
        tanggalPublikasi: form.tanggalPublikasi || void 0
      };
      if (editId) {
        await api.berita.update(editId, input);
        ue.success("Artikel berhasil diperbarui");
      } else {
        await api.berita.add(input);
        ue.success("Artikel berhasil ditambahkan");
      }
      qc.invalidateQueries({ queryKey: ["berita"] });
      qc.invalidateQueries({ queryKey: ["statistik"] });
      setShowForm(false);
    } catch {
      ue.error("Gagal menyimpan artikel. Silakan coba lagi.");
    } finally {
      setSaving(false);
      setIsUploading(false);
      setUploadProgress(0);
    }
  }
  async function handleDelete(id) {
    try {
      await deleteBerita.mutateAsync(id);
      ue.success("Artikel berhasil dihapus");
    } catch {
      ue.error("Gagal menghapus artikel");
    } finally {
      setDeleteTarget(null);
    }
  }
  const isBusy = saving || isUploading;
  return /* @__PURE__ */ jsxRuntimeExports.jsxs(jsxRuntimeExports.Fragment, { children: [
    /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "container mx-auto px-4 py-6", children: [
      /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex flex-col sm:flex-row sm:items-center sm:justify-between gap-3 mb-6", children: [
        /* @__PURE__ */ jsxRuntimeExports.jsx("h1", { className: "font-display font-bold text-xl text-foreground", children: "Kelola Berita" }),
        /* @__PURE__ */ jsxRuntimeExports.jsxs(
          Button,
          {
            onClick: openCreate,
            className: "bg-accent hover:bg-accent/90 text-accent-foreground gap-2 text-sm",
            "data-ocid": "admin_berita.tambah_button",
            children: [
              /* @__PURE__ */ jsxRuntimeExports.jsx(Plus, { className: "w-4 h-4" }),
              "Tambah Artikel"
            ]
          }
        )
      ] }),
      isLoading ? /* @__PURE__ */ jsxRuntimeExports.jsx(SkeletonList, { count: 4 }) : isError ? /* @__PURE__ */ jsxRuntimeExports.jsx(ErrorState, { onRetry: () => refetch() }) : !artikelList || artikelList.length === 0 ? /* @__PURE__ */ jsxRuntimeExports.jsx(
        EmptyState,
        {
          title: "Belum ada artikel berita",
          description: "Publikasikan artikel berita atau pengumuman pertama Anda.",
          actionLabel: "Tambah Artikel",
          onAction: openCreate,
          "data-ocid": "admin_berita.empty_state"
        }
      ) : /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "space-y-3", "data-ocid": "admin_berita.artikel_list", children: artikelList.map((artikel, idx) => {
        const artikelAny = artikel;
        const artikelStatus = artikelAny.status;
        const artikelGambarUrl = artikelAny.gambar;
        const createdBy = artikelAny.createdBy ?? "";
        const lastModifiedBy = artikelAny.lastModifiedBy ?? "";
        return /* @__PURE__ */ jsxRuntimeExports.jsxs(
          "div",
          {
            className: "bg-card border border-border rounded-lg p-4 flex items-center gap-4",
            "data-ocid": `admin_berita.artikel_item.${idx + 1}`,
            children: [
              artikelGambarUrl ? /* @__PURE__ */ jsxRuntimeExports.jsx(
                "img",
                {
                  src: artikelGambarUrl,
                  alt: artikel.judul,
                  className: "w-14 h-14 rounded object-cover border border-border shrink-0 hidden sm:block"
                }
              ) : /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "w-14 h-14 rounded border border-border bg-muted flex items-center justify-center shrink-0 hidden sm:block", children: /* @__PURE__ */ jsxRuntimeExports.jsx(Image, { className: "w-5 h-5 text-muted-foreground" }) }),
              /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex-1 min-w-0", children: [
                /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex items-center gap-2 flex-wrap", children: [
                  /* @__PURE__ */ jsxRuntimeExports.jsx("h3", { className: "font-medium text-foreground text-sm leading-snug line-clamp-1", children: artikel.judul }),
                  artikelStatus && /* @__PURE__ */ jsxRuntimeExports.jsx(
                    Badge,
                    {
                      variant: artikelStatus === "Terbit" ? "default" : "secondary",
                      className: artikelStatus === "Terbit" ? "text-xs bg-primary/10 text-primary border-primary/20 hover:bg-primary/10" : "text-xs bg-muted text-muted-foreground",
                      children: artikelStatus
                    }
                  )
                ] }),
                /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex flex-wrap gap-3 mt-1.5 text-xs text-muted-foreground", children: [
                  /* @__PURE__ */ jsxRuntimeExports.jsxs("span", { className: "flex items-center gap-1", children: [
                    /* @__PURE__ */ jsxRuntimeExports.jsx(Calendar, { className: "w-3 h-3" }),
                    formatTanggal(artikel.createdAt)
                  ] }),
                  /* @__PURE__ */ jsxRuntimeExports.jsxs("span", { className: "flex items-center gap-1", children: [
                    /* @__PURE__ */ jsxRuntimeExports.jsx(User, { className: "w-3 h-3" }),
                    artikel.author ?? "-"
                  ] }),
                  artikel.tags && artikel.tags.length > 0 && /* @__PURE__ */ jsxRuntimeExports.jsxs("span", { className: "flex items-center gap-1", children: [
                    /* @__PURE__ */ jsxRuntimeExports.jsx(Tag, { className: "w-3 h-3" }),
                    artikel.tags.slice(0, 2).join(", "),
                    artikel.tags.length > 2 && ` +${artikel.tags.length - 2}`
                  ] })
                ] }),
                /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex flex-wrap gap-3 mt-1 text-xs text-muted-foreground", children: [
                  /* @__PURE__ */ jsxRuntimeExports.jsxs(
                    "span",
                    {
                      className: "flex items-center gap-1",
                      title: createdBy || "—",
                      children: [
                        /* @__PURE__ */ jsxRuntimeExports.jsx(UserCheck, { className: "w-3 h-3" }),
                        "Dibuat:",
                        " ",
                        /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "font-mono", children: createdBy ? truncatePrincipal(createdBy) : "—" })
                      ]
                    }
                  ),
                  lastModifiedBy && /* @__PURE__ */ jsxRuntimeExports.jsxs(
                    "span",
                    {
                      className: "flex items-center gap-1",
                      title: lastModifiedBy,
                      children: [
                        "Diubah:",
                        " ",
                        /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "font-mono", children: truncatePrincipal(lastModifiedBy) })
                      ]
                    }
                  )
                ] }),
                artikel.ringkasan && /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-xs text-muted-foreground mt-1 line-clamp-1", children: artikel.ringkasan })
              ] }),
              /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex items-center gap-2 shrink-0", children: [
                /* @__PURE__ */ jsxRuntimeExports.jsx(
                  Button,
                  {
                    variant: "ghost",
                    size: "icon",
                    className: "h-8 w-8 border border-border",
                    onClick: () => openEdit(String(artikel.id), {
                      judul: artikel.judul,
                      author: artikel.author,
                      konten: artikel.konten,
                      ringkasan: artikel.ringkasan,
                      gambar: artikelGambarUrl,
                      tags: artikel.tags,
                      status: artikelStatus,
                      createdBy,
                      lastModifiedBy
                    }),
                    "data-ocid": `admin_berita.edit_button.${idx + 1}`,
                    children: /* @__PURE__ */ jsxRuntimeExports.jsx(Pencil, { className: "w-3.5 h-3.5" })
                  }
                ),
                /* @__PURE__ */ jsxRuntimeExports.jsx(
                  Button,
                  {
                    variant: "ghost",
                    size: "icon",
                    className: "h-8 w-8 border border-border text-accent hover:text-accent hover:bg-accent/5",
                    onClick: () => setDeleteTarget({
                      id: String(artikel.id),
                      judul: artikel.judul
                    }),
                    "data-ocid": `admin_berita.delete_button.${idx + 1}`,
                    children: /* @__PURE__ */ jsxRuntimeExports.jsx(Trash2, { className: "w-3.5 h-3.5" })
                  }
                )
              ] })
            ]
          },
          artikel.id
        );
      }) })
    ] }),
    /* @__PURE__ */ jsxRuntimeExports.jsx(Dialog, { open: showForm, onOpenChange: (v) => !isBusy && setShowForm(v), children: /* @__PURE__ */ jsxRuntimeExports.jsxs(
      DialogContent,
      {
        className: "max-w-2xl max-h-[90vh] overflow-y-auto",
        "data-ocid": "admin_berita.dialog",
        children: [
          /* @__PURE__ */ jsxRuntimeExports.jsx(DialogHeader, { children: /* @__PURE__ */ jsxRuntimeExports.jsx(DialogTitle, { className: "font-display text-base", children: editId ? "Edit Artikel" : "Tambah Artikel" }) }),
          /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "space-y-4 py-2", children: [
            /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { children: [
              /* @__PURE__ */ jsxRuntimeExports.jsxs(Label, { htmlFor: "berita-judul", className: "text-sm font-medium", children: [
                "Judul ",
                /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "text-accent", children: "*" })
              ] }),
              /* @__PURE__ */ jsxRuntimeExports.jsx(
                Input,
                {
                  id: "berita-judul",
                  value: form.judul,
                  onChange: (e) => setField("judul", e.target.value),
                  placeholder: "Judul artikel atau pengumuman",
                  className: "mt-1 text-sm",
                  "data-ocid": "admin_berita.judul_input"
                }
              ),
              errors.judul && /* @__PURE__ */ jsxRuntimeExports.jsx(
                "p",
                {
                  className: "text-xs text-accent mt-1",
                  "data-ocid": "admin_berita.judul.field_error",
                  children: errors.judul
                }
              )
            ] }),
            /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { children: [
              /* @__PURE__ */ jsxRuntimeExports.jsx(Label, { htmlFor: "berita-ringkasan", className: "text-sm font-medium", children: "Ringkasan" }),
              /* @__PURE__ */ jsxRuntimeExports.jsx(
                Textarea,
                {
                  id: "berita-ringkasan",
                  value: form.ringkasan,
                  onChange: (e) => setField("ringkasan", e.target.value),
                  placeholder: "Ringkasan singkat artikel (opsional)...",
                  rows: 3,
                  className: "mt-1 text-sm resize-none",
                  "data-ocid": "admin_berita.ringkasan_textarea"
                }
              )
            ] }),
            /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { children: [
              /* @__PURE__ */ jsxRuntimeExports.jsxs(Label, { htmlFor: "berita-konten", className: "text-sm font-medium", children: [
                "Konten ",
                /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "text-accent", children: "*" })
              ] }),
              /* @__PURE__ */ jsxRuntimeExports.jsx(
                Textarea,
                {
                  id: "berita-konten",
                  value: form.konten,
                  onChange: (e) => setField("konten", e.target.value),
                  placeholder: "Tulis isi artikel di sini...",
                  rows: 8,
                  className: "mt-1 text-sm",
                  "data-ocid": "admin_berita.konten_textarea"
                }
              ),
              errors.konten && /* @__PURE__ */ jsxRuntimeExports.jsx(
                "p",
                {
                  className: "text-xs text-accent mt-1",
                  "data-ocid": "admin_berita.konten.field_error",
                  children: errors.konten
                }
              )
            ] }),
            /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "grid grid-cols-1 sm:grid-cols-2 gap-4", children: [
              /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { children: [
                /* @__PURE__ */ jsxRuntimeExports.jsxs(Label, { htmlFor: "berita-author", className: "text-sm font-medium", children: [
                  "Penulis ",
                  /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "text-accent", children: "*" })
                ] }),
                /* @__PURE__ */ jsxRuntimeExports.jsx(
                  Input,
                  {
                    id: "berita-author",
                    value: form.author,
                    onChange: (e) => setField("author", e.target.value),
                    placeholder: "Nama penulis atau unit",
                    className: "mt-1 text-sm",
                    "data-ocid": "admin_berita.author_input"
                  }
                ),
                errors.author && /* @__PURE__ */ jsxRuntimeExports.jsx(
                  "p",
                  {
                    className: "text-xs text-accent mt-1",
                    "data-ocid": "admin_berita.author.field_error",
                    children: errors.author
                  }
                )
              ] }),
              /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { children: [
                /* @__PURE__ */ jsxRuntimeExports.jsx(Label, { htmlFor: "berita-tanggal", className: "text-sm font-medium", children: "Tanggal Publikasi" }),
                /* @__PURE__ */ jsxRuntimeExports.jsx(
                  Input,
                  {
                    id: "berita-tanggal",
                    type: "date",
                    value: form.tanggalPublikasi,
                    onChange: (e) => setField("tanggalPublikasi", e.target.value),
                    className: "mt-1 text-sm",
                    "data-ocid": "admin_berita.tanggal_input"
                  }
                )
              ] })
            ] }),
            /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "grid grid-cols-1 sm:grid-cols-2 gap-4 items-start", children: [
              /* @__PURE__ */ jsxRuntimeExports.jsx(
                ImageUploadWidget,
                {
                  imageFile,
                  existingImageUrl,
                  uploadProgress,
                  isUploading,
                  error: errors.gambar,
                  onFileSelect: (f) => {
                    setImageFile(f);
                    setErrors((prev) => ({ ...prev, gambar: void 0 }));
                  },
                  onClearExisting: () => {
                    setExistingImageUrl("");
                  }
                }
              ),
              /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { children: [
                /* @__PURE__ */ jsxRuntimeExports.jsx(Label, { htmlFor: "berita-status", className: "text-sm font-medium", children: "Status Publikasi" }),
                /* @__PURE__ */ jsxRuntimeExports.jsxs(
                  Select,
                  {
                    value: form.status,
                    onValueChange: (val) => setField("status", val),
                    children: [
                      /* @__PURE__ */ jsxRuntimeExports.jsx(
                        SelectTrigger,
                        {
                          id: "berita-status",
                          className: "mt-1 text-sm",
                          "data-ocid": "admin_berita.status_select",
                          children: /* @__PURE__ */ jsxRuntimeExports.jsx(SelectValue, { placeholder: "Pilih status" })
                        }
                      ),
                      /* @__PURE__ */ jsxRuntimeExports.jsxs(SelectContent, { children: [
                        /* @__PURE__ */ jsxRuntimeExports.jsx(SelectItem, { value: "Terbit", children: "Terbit" }),
                        /* @__PURE__ */ jsxRuntimeExports.jsx(SelectItem, { value: "Draf", children: "Draf" })
                      ] })
                    ]
                  }
                )
              ] })
            ] }),
            /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { children: [
              /* @__PURE__ */ jsxRuntimeExports.jsx(Label, { htmlFor: "berita-tags", className: "text-sm font-medium", children: "Tags" }),
              /* @__PURE__ */ jsxRuntimeExports.jsx(
                Input,
                {
                  id: "berita-tags",
                  value: form.tagsInput,
                  onChange: (e) => setField("tagsInput", e.target.value),
                  placeholder: "hukum, regulasi, kampus",
                  className: "mt-1 text-sm",
                  "data-ocid": "admin_berita.tags_input"
                }
              ),
              /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-xs text-muted-foreground mt-1", children: "Pisahkan dengan koma" })
            ] }),
            editId && /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "rounded-lg bg-muted/50 border border-border p-3 space-y-1.5", children: [
              /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-xs font-semibold text-muted-foreground uppercase tracking-wide mb-2", children: "Informasi Perubahan" }),
              /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex items-start gap-2 text-xs text-muted-foreground", children: [
                /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "shrink-0 w-32", children: "Dibuat oleh:" }),
                /* @__PURE__ */ jsxRuntimeExports.jsx(
                  "span",
                  {
                    className: "font-mono break-all text-foreground",
                    title: editAudit.createdBy || "—",
                    children: editAudit.createdBy || "—"
                  }
                )
              ] }),
              /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex items-start gap-2 text-xs text-muted-foreground", children: [
                /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "shrink-0 w-32", children: "Terakhir diubah:" }),
                /* @__PURE__ */ jsxRuntimeExports.jsx(
                  "span",
                  {
                    className: "font-mono break-all text-foreground",
                    title: editAudit.lastModifiedBy || "—",
                    children: editAudit.lastModifiedBy || "—"
                  }
                )
              ] })
            ] })
          ] }),
          /* @__PURE__ */ jsxRuntimeExports.jsxs(DialogFooter, { className: "gap-2 pt-2", children: [
            /* @__PURE__ */ jsxRuntimeExports.jsx(
              Button,
              {
                variant: "outline",
                onClick: () => setShowForm(false),
                disabled: isBusy,
                "data-ocid": "admin_berita.cancel_button",
                children: "Batal"
              }
            ),
            /* @__PURE__ */ jsxRuntimeExports.jsxs(
              Button,
              {
                onClick: handleSave,
                disabled: isBusy,
                className: "bg-accent hover:bg-accent/90 text-accent-foreground gap-2",
                "data-ocid": "admin_berita.submit_button",
                children: [
                  isBusy && /* @__PURE__ */ jsxRuntimeExports.jsx(LoaderCircle, { className: "w-4 h-4 animate-spin" }),
                  isUploading ? "Mengunggah gambar..." : saving ? "Menyimpan..." : "Simpan"
                ]
              }
            )
          ] })
        ]
      }
    ) }),
    /* @__PURE__ */ jsxRuntimeExports.jsx(
      AlertDialog,
      {
        open: !!deleteTarget,
        onOpenChange: (v) => !v && setDeleteTarget(null),
        children: /* @__PURE__ */ jsxRuntimeExports.jsxs(AlertDialogContent, { "data-ocid": "admin_berita.delete_dialog", children: [
          /* @__PURE__ */ jsxRuntimeExports.jsxs(AlertDialogHeader, { children: [
            /* @__PURE__ */ jsxRuntimeExports.jsx(AlertDialogTitle, { children: "Hapus Artikel?" }),
            /* @__PURE__ */ jsxRuntimeExports.jsxs(AlertDialogDescription, { children: [
              "Artikel ",
              /* @__PURE__ */ jsxRuntimeExports.jsx("strong", { children: deleteTarget == null ? void 0 : deleteTarget.judul }),
              " akan dihapus secara permanen."
            ] })
          ] }),
          /* @__PURE__ */ jsxRuntimeExports.jsxs(AlertDialogFooter, { children: [
            /* @__PURE__ */ jsxRuntimeExports.jsx(AlertDialogCancel, { "data-ocid": "admin_berita.delete_cancel_button", children: "Batal" }),
            /* @__PURE__ */ jsxRuntimeExports.jsx(
              AlertDialogAction,
              {
                onClick: () => deleteTarget && handleDelete(deleteTarget.id),
                className: "bg-accent hover:bg-accent/90 text-accent-foreground",
                "data-ocid": "admin_berita.delete_confirm_button",
                children: deleteBerita.isPending ? /* @__PURE__ */ jsxRuntimeExports.jsx(LoaderCircle, { className: "w-4 h-4 animate-spin" }) : "Hapus"
              }
            )
          ] })
        ] })
      }
    )
  ] });
}
export {
  AdminBerita
};
