// @caffeine/template-frontend/dist/assets/AdminGaleri-BKVfVnkj.js
import { t as useQueryClient, p as usePermissions, r as reactExports, j as jsxRuntimeExports, a as Button, e as SkeletonList, I as Image, v as LoaderCircle, X, h as api } from "./index-B7-RSq-m.js";
import { A as AlertDialog, a as AlertDialogContent, b as AlertDialogHeader, c as AlertDialogTitle, d as AlertDialogDescription, e as AlertDialogFooter, f as AlertDialogCancel, g as AlertDialogAction } from "./alert-dialog-lTbm3t0t.js";
import { D as Dialog, a as DialogContent, b as DialogHeader, c as DialogTitle, d as DialogFooter } from "./dialog-CIgZYBSq.js";
import { I as Input } from "./input-B3nX7k9Z.js";
import { L as Label } from "./label-eF2rHvnY.js";
import { T as Textarea, U as Upload } from "./textarea-DRo76BEL.js";
import { u as ue } from "./index-B0YpMaU8.js";
import { E as EmptyState } from "./EmptyState-Bzc48I5U.js";
import { E as ErrorState } from "./ErrorState-o51Yfgm5.js";
import { c as useGaleri, r as useDeleteGaleriItem } from "./useBackend-xFEMRG87.js";
import { P as Plus } from "./plus-C7LHpCDy.js";
import { P as Pencil } from "./pencil-CwCkT-7p.js";
import { T as Trash2 } from "./trash-2-CnhMEsa2.js";
import "./index-BcOIygc1.js";
import "./circle-alert-D-S0YTKb.js";
import "./useMutation-D--BDafu.js";
const EMPTY_FORM = { judul: "", deskripsi: "", album: "" };
const ALBUM_SUGGESTIONS = [
  "Kegiatan Akademik",
  "Wisuda",
  "Seminar",
  "Fasilitas Kampus",
  "Kegiatan Mahasiswa",
  "Dokumentasi Hukum"
];
function validate(form) {
  const errs = {};
  if (!form.judul.trim()) errs.judul = "Judul wajib diisi";
  if (!form.album.trim()) errs.album = "Album wajib diisi";
  return errs;
}
function GaleriImageWidget({
  imageFile,
  existingUrl,
  isUploading,
  uploadProgress,
  onFileSelect,
  onClearExisting
}) {
  const fileRef = reactExports.useRef(null);
  const previewUrl = imageFile ? URL.createObjectURL(imageFile) : null;
  const displayUrl = previewUrl ?? existingUrl ?? null;
  return /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { children: [
    /* @__PURE__ */ jsxRuntimeExports.jsx(Label, { className: "text-sm font-medium", children: "Foto" }),
    /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "mt-1", children: [
      displayUrl ? /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "relative rounded-lg overflow-hidden border border-border", children: [
        /* @__PURE__ */ jsxRuntimeExports.jsx(
          "img",
          {
            src: displayUrl,
            alt: "Pratinjau foto",
            className: "w-full h-44 object-cover"
          }
        ),
        /* @__PURE__ */ jsxRuntimeExports.jsx(
          "button",
          {
            type: "button",
            onClick: () => {
              onFileSelect(null);
              onClearExisting();
              if (fileRef.current) fileRef.current.value = "";
            },
            className: "absolute top-2 right-2 bg-background/80 hover:bg-background rounded-full p-1 border border-border transition-colors",
            "aria-label": "Hapus foto",
            "data-ocid": "admin_galeri.gambar_remove_button",
            children: /* @__PURE__ */ jsxRuntimeExports.jsx(X, { className: "w-4 h-4 text-foreground" })
          }
        ),
        imageFile && /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "absolute bottom-0 left-0 right-0 bg-background/80 px-3 py-1.5 text-xs text-foreground truncate", children: imageFile.name })
      ] }) : /* @__PURE__ */ jsxRuntimeExports.jsxs(
        "button",
        {
          type: "button",
          className: "w-full border-2 border-dashed border-border rounded-lg p-6 text-center cursor-pointer hover:border-primary/40 transition-colors",
          onClick: () => {
            var _a;
            return (_a = fileRef.current) == null ? void 0 : _a.click();
          },
          "data-ocid": "admin_galeri.gambar_upload_button",
          children: [
            isUploading ? /* @__PURE__ */ jsxRuntimeExports.jsx(LoaderCircle, { className: "w-7 h-7 text-muted-foreground mx-auto mb-2 animate-spin" }) : /* @__PURE__ */ jsxRuntimeExports.jsx(Image, { className: "w-7 h-7 text-muted-foreground mx-auto mb-2" }),
            /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-xs text-muted-foreground", children: "Klik untuk pilih foto (JPG, PNG, WebP)" })
          ]
        }
      ),
      /* @__PURE__ */ jsxRuntimeExports.jsx(
        "input",
        {
          ref: fileRef,
          type: "file",
          accept: "image/*",
          className: "hidden",
          onChange: (e) => {
            var _a;
            return onFileSelect(((_a = e.target.files) == null ? void 0 : _a[0]) ?? null);
          },
          "data-ocid": "admin_galeri.gambar_file_input"
        }
      ),
      isUploading && uploadProgress > 0 && uploadProgress < 100 && /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "mt-2", children: [
        /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "h-1.5 bg-muted rounded-full overflow-hidden", children: /* @__PURE__ */ jsxRuntimeExports.jsx(
          "div",
          {
            className: "h-full bg-primary rounded-full transition-all",
            style: { width: `${uploadProgress}%` }
          }
        ) }),
        /* @__PURE__ */ jsxRuntimeExports.jsxs("p", { className: "text-xs text-muted-foreground mt-1", children: [
          "Mengunggah foto... ",
          uploadProgress,
          "%"
        ] })
      ] }),
      displayUrl && /* @__PURE__ */ jsxRuntimeExports.jsxs(
        "button",
        {
          type: "button",
          className: "mt-2 flex items-center gap-1.5 text-xs text-muted-foreground hover:text-foreground transition-colors",
          onClick: () => {
            var _a;
            return (_a = fileRef.current) == null ? void 0 : _a.click();
          },
          "data-ocid": "admin_galeri.gambar_change_button",
          children: [
            /* @__PURE__ */ jsxRuntimeExports.jsx(Upload, { className: "w-3.5 h-3.5" }),
            "Ganti foto"
          ]
        }
      )
    ] })
  ] });
}
function AdminGaleri() {
  const qc = useQueryClient();
  const { isAdmin, grantedMenus } = usePermissions();
  const { data: galeriList, isLoading, isError, refetch } = useGaleri();
  const deleteGaleriItem = useDeleteGaleriItem();
  const [showForm, setShowForm] = reactExports.useState(false);
  const [editId, setEditId] = reactExports.useState(null);
  const [form, setForm] = reactExports.useState(EMPTY_FORM);
  const [errors, setErrors] = reactExports.useState({});
  const [saving, setSaving] = reactExports.useState(false);
  const [imageFile, setImageFile] = reactExports.useState(null);
  const [existingImageUrl, setExistingImageUrl] = reactExports.useState("");
  const [uploadProgress, setUploadProgress] = reactExports.useState(0);
  const [isUploading, setIsUploading] = reactExports.useState(false);
  const [deleteTarget, setDeleteTarget] = reactExports.useState(null);
  const canAccess = isAdmin || grantedMenus.includes("galeri");
  if (!canAccess) {
    return /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "container mx-auto px-4 py-12 text-center", children: /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-muted-foreground", children: "Anda tidak memiliki akses ke halaman ini." }) });
  }
  function openCreate() {
    setForm(EMPTY_FORM);
    setErrors({});
    setEditId(null);
    setImageFile(null);
    setExistingImageUrl("");
    setUploadProgress(0);
    setShowForm(true);
  }
  function openEdit(item) {
    setForm({
      judul: item.judul,
      deskripsi: item.deskripsi,
      album: item.album
    });
    setErrors({});
    setEditId(item.id);
    setImageFile(null);
    setExistingImageUrl(item.gambar ?? "");
    setUploadProgress(0);
    setShowForm(true);
  }
  function setField(key, val) {
    setForm((prev) => ({ ...prev, [key]: val }));
    if (key in errors) setErrors((prev) => ({ ...prev, [key]: void 0 }));
  }
  function getSelectedImageFile() {
    return imageFile || void 0;
  }
  async function handleSave() {
    const errs = validate(form);
    if (Object.keys(errs).length > 0) {
      setErrors(errs);
      return;
    }
    setSaving(true);
    try {
      const gambar = getSelectedImageFile();
      const input = {
        judul: form.judul.trim(),
        deskripsi: form.deskripsi.trim(),
        album: form.album.trim(),
        gambar
      };
      if (editId !== null) {
        await api.galeri.update(Number(editId), input);
        ue.success("Foto berhasil diperbarui");
      } else {
        await api.galeri.add(input);
        ue.success("Foto berhasil ditambahkan");
      }
      qc.invalidateQueries({ queryKey: ["galeri"] });
      setShowForm(false);
    } catch {
      ue.error("Gagal menyimpan foto. Silakan coba lagi.");
    } finally {
      setSaving(false);
      setIsUploading(false);
      setUploadProgress(0);
    }
  }
  async function handleDelete(id) {
    try {
      await deleteGaleriItem.mutateAsync(id);
      ue.success("Foto berhasil dihapus");
    } catch {
      ue.error("Gagal menghapus foto");
    } finally {
      setDeleteTarget(null);
    }
  }
  const isBusy = saving || isUploading;
  return /* @__PURE__ */ jsxRuntimeExports.jsxs(jsxRuntimeExports.Fragment, { children: [
    /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "container mx-auto px-4 py-6", children: [
      /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex flex-col sm:flex-row sm:items-center sm:justify-between gap-3 mb-6", children: [
        /* @__PURE__ */ jsxRuntimeExports.jsx("h1", { className: "font-display font-bold text-xl text-foreground", children: "Kelola Galeri" }),
        /* @__PURE__ */ jsxRuntimeExports.jsxs(
          Button,
          {
            onClick: openCreate,
            className: "bg-accent hover:bg-accent/90 text-accent-foreground gap-2 text-sm",
            "data-ocid": "admin_galeri.tambah_button",
            children: [
              /* @__PURE__ */ jsxRuntimeExports.jsx(Plus, { className: "w-4 h-4" }),
              "Tambah Foto"
            ]
          }
        )
      ] }),
      isLoading ? /* @__PURE__ */ jsxRuntimeExports.jsx(SkeletonList, { count: 4 }) : isError ? /* @__PURE__ */ jsxRuntimeExports.jsx(ErrorState, { onRetry: () => refetch() }) : !galeriList || galeriList.length === 0 ? /* @__PURE__ */ jsxRuntimeExports.jsx(
        EmptyState,
        {
          title: "Belum ada foto di galeri",
          description: "Unggah foto kegiatan atau dokumentasi kampus pertama Anda.",
          actionLabel: "Tambah Foto",
          onAction: openCreate,
          "data-ocid": "admin_galeri.empty_state"
        }
      ) : /* @__PURE__ */ jsxRuntimeExports.jsx(
        "div",
        {
          className: "grid grid-cols-2 sm:grid-cols-3 lg:grid-cols-4 gap-3",
          "data-ocid": "admin_galeri.foto_list",
          children: galeriList.map((item, idx) => {
            const imgUrl = item.gambar ?? null;
            return /* @__PURE__ */ jsxRuntimeExports.jsxs(
              "div",
              {
                className: "group relative bg-card border border-border rounded-lg overflow-hidden",
                "data-ocid": `admin_galeri.foto_item.${idx + 1}`,
                children: [
                  imgUrl ? /* @__PURE__ */ jsxRuntimeExports.jsx(
                    "img",
                    {
                      src: imgUrl,
                      alt: item.judul,
                      className: "w-full aspect-square object-cover"
                    }
                  ) : /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "w-full aspect-square bg-muted flex items-center justify-center", children: /* @__PURE__ */ jsxRuntimeExports.jsx(Image, { className: "w-10 h-10 text-muted-foreground/30" }) }),
                  /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "absolute inset-0 bg-foreground/0 group-hover:bg-foreground/50 transition-all duration-200" }),
                  /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "absolute inset-0 flex flex-col justify-between p-3 opacity-0 group-hover:opacity-100 transition-opacity duration-200", children: [
                    /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex justify-end gap-1.5", children: [
                      /* @__PURE__ */ jsxRuntimeExports.jsx(
                        "button",
                        {
                          type: "button",
                          onClick: () => openEdit(item),
                          className: "w-7 h-7 bg-background/90 hover:bg-background rounded flex items-center justify-center border border-border transition-colors",
                          "aria-label": "Edit",
                          "data-ocid": `admin_galeri.edit_button.${idx + 1}`,
                          children: /* @__PURE__ */ jsxRuntimeExports.jsx(Pencil, { className: "w-3.5 h-3.5 text-foreground" })
                        }
                      ),
                      /* @__PURE__ */ jsxRuntimeExports.jsx(
                        "button",
                        {
                          type: "button",
                          onClick: () => setDeleteTarget({ id: item.id, judul: item.judul }),
                          className: "w-7 h-7 bg-background/90 hover:bg-accent/10 rounded flex items-center justify-center border border-border transition-colors",
                          "aria-label": "Hapus",
                          "data-ocid": `admin_galeri.delete_button.${idx + 1}`,
                          children: /* @__PURE__ */ jsxRuntimeExports.jsx(Trash2, { className: "w-3.5 h-3.5 text-accent" })
                        }
                      )
                    ] }),
                    /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { children: [
                      /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "inline-block bg-primary text-primary-foreground text-xs px-2 py-0.5 rounded mb-1", children: item.album }),
                      /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-white text-xs font-semibold line-clamp-2", children: item.judul })
                    ] })
                  ] }),
                  /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "p-2 bg-card border-t border-border group-hover:opacity-0 transition-opacity", children: [
                    /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-foreground text-xs font-medium truncate", children: item.judul }),
                    /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-muted-foreground text-xs truncate", children: item.album })
                  ] })
                ]
              },
              item.id
            );
          })
        }
      )
    ] }),
    /* @__PURE__ */ jsxRuntimeExports.jsx(Dialog, { open: showForm, onOpenChange: (v) => !isBusy && setShowForm(v), children: /* @__PURE__ */ jsxRuntimeExports.jsxs(
      DialogContent,
      {
        className: "max-w-lg max-h-[90vh] overflow-y-auto",
        "data-ocid": "admin_galeri.dialog",
        children: [
          /* @__PURE__ */ jsxRuntimeExports.jsx(DialogHeader, { children: /* @__PURE__ */ jsxRuntimeExports.jsx(DialogTitle, { className: "font-display text-base", children: editId !== null ? "Edit Foto" : "Tambah Foto" }) }),
          /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "space-y-4 py-2", children: [
            /* @__PURE__ */ jsxRuntimeExports.jsx(
              GaleriImageWidget,
              {
                imageFile,
                existingUrl: existingImageUrl,
                isUploading,
                uploadProgress,
                onFileSelect: (f) => setImageFile(f),
                onClearExisting: () => {
                  setExistingImageUrl("");
                }
              }
            ),
            /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { children: [
              /* @__PURE__ */ jsxRuntimeExports.jsxs(Label, { htmlFor: "galeri-judul", className: "text-sm font-medium", children: [
                "Judul ",
                /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "text-accent", children: "*" })
              ] }),
              /* @__PURE__ */ jsxRuntimeExports.jsx(
                Input,
                {
                  id: "galeri-judul",
                  value: form.judul,
                  onChange: (e) => setField("judul", e.target.value),
                  placeholder: "Judul foto atau keterangan singkat",
                  className: "mt-1 text-sm",
                  "data-ocid": "admin_galeri.judul_input"
                }
              ),
              errors.judul && /* @__PURE__ */ jsxRuntimeExports.jsx(
                "p",
                {
                  className: "text-xs text-accent mt-1",
                  "data-ocid": "admin_galeri.judul.field_error",
                  children: errors.judul
                }
              )
            ] }),
            /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { children: [
              /* @__PURE__ */ jsxRuntimeExports.jsx(Label, { htmlFor: "galeri-deskripsi", className: "text-sm font-medium", children: "Deskripsi" }),
              /* @__PURE__ */ jsxRuntimeExports.jsx(
                Textarea,
                {
                  id: "galeri-deskripsi",
                  value: form.deskripsi,
                  onChange: (e) => setField("deskripsi", e.target.value),
                  placeholder: "Keterangan foto lebih lengkap (opsional)...",
                  rows: 3,
                  className: "mt-1 text-sm resize-none",
                  "data-ocid": "admin_galeri.deskripsi_textarea"
                }
              )
            ] }),
            /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { children: [
              /* @__PURE__ */ jsxRuntimeExports.jsxs(Label, { htmlFor: "galeri-album", className: "text-sm font-medium", children: [
                "Album ",
                /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "text-accent", children: "*" })
              ] }),
              /* @__PURE__ */ jsxRuntimeExports.jsx(
                Input,
                {
                  id: "galeri-album",
                  value: form.album,
                  onChange: (e) => setField("album", e.target.value),
                  placeholder: "Nama album atau kategori foto",
                  className: "mt-1 text-sm",
                  list: "album-suggestions",
                  "data-ocid": "admin_galeri.album_input"
                }
              ),
              /* @__PURE__ */ jsxRuntimeExports.jsx("datalist", { id: "album-suggestions", children: ALBUM_SUGGESTIONS.map((s) => /* @__PURE__ */ jsxRuntimeExports.jsx("option", { value: s }, s)) }),
              errors.album && /* @__PURE__ */ jsxRuntimeExports.jsx(
                "p",
                {
                  className: "text-xs text-accent mt-1",
                  "data-ocid": "admin_galeri.album.field_error",
                  children: errors.album
                }
              ),
              /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-xs text-muted-foreground mt-1", children: "Contoh: Kegiatan Akademik, Wisuda, Fasilitas Kampus" })
            ] })
          ] }),
          /* @__PURE__ */ jsxRuntimeExports.jsxs(DialogFooter, { className: "gap-2 pt-2", children: [
            /* @__PURE__ */ jsxRuntimeExports.jsx(
              Button,
              {
                variant: "outline",
                onClick: () => setShowForm(false),
                disabled: isBusy,
                "data-ocid": "admin_galeri.cancel_button",
                children: "Batal"
              }
            ),
            /* @__PURE__ */ jsxRuntimeExports.jsxs(
              Button,
              {
                onClick: handleSave,
                disabled: isBusy,
                className: "bg-accent hover:bg-accent/90 text-accent-foreground gap-2",
                "data-ocid": "admin_galeri.submit_button",
                children: [
                  isBusy && /* @__PURE__ */ jsxRuntimeExports.jsx(LoaderCircle, { className: "w-4 h-4 animate-spin" }),
                  isUploading ? "Mengunggah foto..." : saving ? "Menyimpan..." : "Simpan"
                ]
              }
            )
          ] })
        ]
      }
    ) }),
    /* @__PURE__ */ jsxRuntimeExports.jsx(
      AlertDialog,
      {
        open: !!deleteTarget,
        onOpenChange: (v) => !v && setDeleteTarget(null),
        children: /* @__PURE__ */ jsxRuntimeExports.jsxs(AlertDialogContent, { "data-ocid": "admin_galeri.delete_dialog", children: [
          /* @__PURE__ */ jsxRuntimeExports.jsxs(AlertDialogHeader, { children: [
            /* @__PURE__ */ jsxRuntimeExports.jsx(AlertDialogTitle, { children: "Hapus Foto?" }),
            /* @__PURE__ */ jsxRuntimeExports.jsxs(AlertDialogDescription, { children: [
              "Foto ",
              /* @__PURE__ */ jsxRuntimeExports.jsx("strong", { children: deleteTarget == null ? void 0 : deleteTarget.judul }),
              " akan dihapus secara permanen."
            ] })
          ] }),
          /* @__PURE__ */ jsxRuntimeExports.jsxs(AlertDialogFooter, { children: [
            /* @__PURE__ */ jsxRuntimeExports.jsx(AlertDialogCancel, { "data-ocid": "admin_galeri.delete_cancel_button", children: "Batal" }),
            /* @__PURE__ */ jsxRuntimeExports.jsx(
              AlertDialogAction,
              {
                onClick: () => deleteTarget && handleDelete(deleteTarget.id),
                className: "bg-accent hover:bg-accent/90 text-accent-foreground",
                "data-ocid": "admin_galeri.delete_confirm_button",
                children: deleteGaleriItem.isPending ? /* @__PURE__ */ jsxRuntimeExports.jsx(LoaderCircle, { className: "w-4 h-4 animate-spin" }) : "Hapus"
              }
            )
          ] })
        ] })
      }
    )
  ] });
}
export {
  AdminGaleri
};
