// @caffeine/template-frontend/dist/assets/Beranda-ChewJD2M.js
import { c as createLucideIcon, r as reactExports, u as useNavigate, F as FileText, B as BookOpen, N as Newspaper, j as jsxRuntimeExports, L as Layout, S as Scale, a as Button, b as Link, d as SkeletonCard, I as Image, X } from "./index-B7-RSq-m.js";
import { B as Badge } from "./badge-KXcDp_20.js";
import { D as Dialog, a as DialogContent } from "./dialog-CIgZYBSq.js";
import { I as Input } from "./input-B3nX7k9Z.js";
import { S as Skeleton } from "./skeleton-Z-jxXZYe.js";
import { J as JENIS_LABEL, K as KategoriTag, S as StatusBadge } from "./StatusBadge-CanDaSra.js";
import { u as useStatistik, a as useDokumen, b as useBerita, c as useGaleri } from "./useBackend-xFEMRG87.js";
import { D as Download } from "./download-BnCFvid2.js";
import { S as Search } from "./search-DeNGI9Ut.js";
import { C as ChevronRight } from "./chevron-right-RI9IPXg2.js";
import { A as ArrowRight } from "./arrow-right-pgjRVcXN.js";
import { C as Calendar } from "./calendar-mZT-2HLQ.js";
import "./useMutation-D--BDafu.js";
/**
 * @license lucide-react v0.511.0 - ISC
 *
 * This source code is licensed under the ISC license.
 * See the LICENSE file in the root directory of this source tree.
 */
const __iconNode$1 = [
  ["path", { d: "m14.5 12.5-8 8a2.119 2.119 0 1 1-3-3l8-8", key: "15492f" }],
  ["path", { d: "m16 16 6-6", key: "vzrcl6" }],
  ["path", { d: "m8 8 6-6", key: "18bi4p" }],
  ["path", { d: "m9 7 8 8", key: "5jnvq1" }],
  ["path", { d: "m21 11-8-8", key: "z4y7zo" }]
];
const Gavel = createLucideIcon("gavel", __iconNode$1);
/**
 * @license lucide-react v0.511.0 - ISC
 *
 * This source code is licensed under the ISC license.
 * See the LICENSE file in the root directory of this source tree.
 */
const __iconNode = [
  ["path", { d: "M16 7h6v6", key: "box55l" }],
  ["path", { d: "m22 7-8.5 8.5-5-5L2 17", key: "1t1m79" }]
];
const TrendingUp = createLucideIcon("trending-up", __iconNode);
const KATEGORI_ICONS = [Gavel, FileText, Scale, BookOpen, Scale, TrendingUp];
const SKELETON_KEYS = ["sa", "sb", "sc", "sd", "se", "sf"];
const BERITA_SKELETON_KEYS = ["ba", "bb", "bc"];
const GALERI_SKELETON_KEYS = ["ga", "gb", "gc", "gd", "ge", "gf"];
function GaleriPreviewCard({
  item,
  index,
  onOpen
}) {
  const imgUrl = item.gambar ?? null;
  return /* @__PURE__ */ jsxRuntimeExports.jsxs(
    "button",
    {
      type: "button",
      onClick: () => onOpen(item),
      className: "group relative aspect-square overflow-hidden rounded-lg border border-border bg-muted focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-primary",
      "data-ocid": `beranda.galeri_card.${index}`,
      children: [
        imgUrl ? /* @__PURE__ */ jsxRuntimeExports.jsx(
          "img",
          {
            src: imgUrl,
            alt: item.judul,
            className: "w-full h-full object-cover transition-transform duration-300 group-hover:scale-105",
            onError: (e) => {
              e.currentTarget.style.display = "none";
            }
          }
        ) : /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "w-full h-full flex items-center justify-center bg-secondary/40", children: /* @__PURE__ */ jsxRuntimeExports.jsx(Image, { className: "w-10 h-10 text-muted-foreground/40" }) }),
        /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "absolute inset-0 bg-foreground/0 group-hover:bg-foreground/50 transition-all duration-300 flex flex-col justify-end p-3 opacity-0 group-hover:opacity-100", children: [
          /* @__PURE__ */ jsxRuntimeExports.jsx(Badge, { className: "self-start bg-primary text-primary-foreground text-xs mb-1.5 pointer-events-none", children: item.album }),
          /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-white text-xs font-semibold leading-snug line-clamp-2 text-left", children: item.judul })
        ] })
      ]
    }
  );
}
function GaleriLightbox({
  item,
  onClose
}) {
  if (!item) return null;
  const imgUrl = item.gambar ?? null;
  return /* @__PURE__ */ jsxRuntimeExports.jsx(Dialog, { open: !!item, onOpenChange: (v) => !v && onClose(), children: /* @__PURE__ */ jsxRuntimeExports.jsx(
    DialogContent,
    {
      className: "max-w-3xl p-0 overflow-hidden bg-card border-border",
      "data-ocid": "beranda.galeri_lightbox_dialog",
      children: /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "relative", children: [
        /* @__PURE__ */ jsxRuntimeExports.jsx(
          "button",
          {
            type: "button",
            onClick: onClose,
            className: "absolute top-3 right-3 z-10 bg-foreground/60 hover:bg-foreground/80 text-white rounded-full p-1.5 transition-colors",
            "aria-label": "Tutup",
            "data-ocid": "beranda.galeri_lightbox.close_button",
            children: /* @__PURE__ */ jsxRuntimeExports.jsx(X, { className: "w-4 h-4" })
          }
        ),
        imgUrl ? /* @__PURE__ */ jsxRuntimeExports.jsx(
          "img",
          {
            src: imgUrl,
            alt: item.judul,
            className: "w-full max-h-[70vh] object-contain bg-muted"
          }
        ) : /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "w-full h-64 flex items-center justify-center bg-muted", children: /* @__PURE__ */ jsxRuntimeExports.jsx(Image, { className: "w-16 h-16 text-muted-foreground/30" }) }),
        /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "p-5", children: [
          /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "flex items-start gap-3 mb-2", children: /* @__PURE__ */ jsxRuntimeExports.jsx(Badge, { className: "bg-primary text-primary-foreground shrink-0", children: item.album }) }),
          /* @__PURE__ */ jsxRuntimeExports.jsx("h2", { className: "font-display font-bold text-lg text-foreground leading-snug mb-1.5", children: item.judul }),
          item.deskripsi && /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-muted-foreground text-sm leading-relaxed", children: item.deskripsi })
        ] })
      ] })
    }
  ) });
}
function Beranda() {
  const [searchQuery, setSearchQuery] = reactExports.useState("");
  const navigate = useNavigate();
  const { data: statistik, isLoading: loadingStats } = useStatistik();
  const { data: dokumen, isLoading: loadingDokumen } = useDokumen({});
  const { data: berita, isLoading: loadingBerita } = useBerita();
  const { data: galeriItems, isLoading: loadingGaleri } = useGaleri();
  const [galeriLightbox, setGaleriLightbox] = reactExports.useState(null);
  const galeriPreview = (galeriItems ?? []).slice(0, 6);
  const handleSearch = (e) => {
    e.preventDefault();
    navigate({ to: "/katalog", search: { q: searchQuery } });
  };
  const STATS = [
    {
      key: "total-dokumen",
      label: "Total Dokumen",
      value: (statistik == null ? void 0 : statistik.totalDokumen) ?? 35,
      icon: FileText,
      color: "text-primary"
    },
    {
      key: "total-unduhan",
      label: "Total Unduhan",
      value: (statistik == null ? void 0 : statistik.totalUnduhan) ?? 1248,
      icon: Download,
      color: "text-accent"
    },
    {
      key: "kategori",
      label: "Kategori Produk Hukum",
      value: (statistik == null ? void 0 : statistik.totalKategori) ?? 6,
      icon: BookOpen,
      color: "text-primary"
    },
    {
      key: "artikel-berita",
      label: "Artikel Berita",
      value: (statistik == null ? void 0 : statistik.totalArtikel) ?? 12,
      icon: Newspaper,
      color: "text-accent"
    }
  ];
  return /* @__PURE__ */ jsxRuntimeExports.jsxs(Layout, { children: [
    /* @__PURE__ */ jsxRuntimeExports.jsxs(
      "section",
      {
        className: "relative overflow-hidden",
        style: {
          backgroundImage: "url('https://images.unsplash.com/photo-1568667256531-ffffff2a9eab?w=1920&q=80')",
          backgroundSize: "cover",
          backgroundPosition: "center",
          backgroundRepeat: "no-repeat"
        },
        "data-ocid": "beranda.hero_section",
        children: [
          /* @__PURE__ */ jsxRuntimeExports.jsx(
            "div",
            {
              className: "absolute inset-0",
              style: { background: "oklch(0.5 0.22 25 / 0.88)" }
            }
          ),
          /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "absolute inset-0 bg-gradient-to-b from-black/20 via-transparent to-black/30" }),
          /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "absolute inset-0 opacity-10", children: [
            /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "absolute top-0 right-0 w-96 h-96 bg-primary-foreground rounded-full blur-3xl -translate-y-1/2 translate-x-1/2" }),
            /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "absolute bottom-0 left-0 w-64 h-64 bg-primary-foreground rounded-full blur-3xl translate-y-1/2 -translate-x-1/2" })
          ] }),
          /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "relative container mx-auto px-4 py-16 md:py-24", children: /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "max-w-3xl mx-auto text-center", children: [
            /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "inline-flex items-center gap-2 bg-primary-foreground/10 border border-primary-foreground/20 text-primary-foreground text-xs font-semibold px-3 py-1.5 rounded-full mb-6", children: [
              /* @__PURE__ */ jsxRuntimeExports.jsx(Scale, { className: "w-3.5 h-3.5" }),
              "Portal Resmi JDIH UNTAG Banyuwangi"
            ] }),
            /* @__PURE__ */ jsxRuntimeExports.jsxs("h1", { className: "font-display font-bold text-3xl md:text-5xl text-primary-foreground leading-tight mb-4", children: [
              "Portal Dokumentasi ",
              /* @__PURE__ */ jsxRuntimeExports.jsx("br", { className: "hidden sm:block" }),
              /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "text-accent-foreground/90", children: "Hukum" }),
              " Kampus"
            ] }),
            /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-primary-foreground/75 text-base md:text-lg max-w-2xl mx-auto mb-10 leading-relaxed", children: "Akses mudah produk hukum Universitas 17 Agustus 1945 Banyuwangi — statuta, peraturan rektor, keputusan, surat edaran, dan dokumen hukum lainnya secara transparan dan terpadu." }),
            /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex flex-col sm:flex-row items-center justify-center gap-3 mb-10", children: [
              /* @__PURE__ */ jsxRuntimeExports.jsx(
                Button,
                {
                  asChild: true,
                  size: "lg",
                  className: "bg-accent hover:bg-accent/90 text-accent-foreground font-semibold px-8 h-12 shadow-lg gap-2 w-full sm:w-auto",
                  "data-ocid": "beranda.cari_dokumen_button",
                  children: /* @__PURE__ */ jsxRuntimeExports.jsxs(Link, { to: "/katalog", children: [
                    /* @__PURE__ */ jsxRuntimeExports.jsx(Search, { className: "w-4 h-4" }),
                    "Cari Dokumen"
                  ] })
                }
              ),
              /* @__PURE__ */ jsxRuntimeExports.jsx(
                Button,
                {
                  asChild: true,
                  size: "lg",
                  variant: "outline",
                  className: "border-primary-foreground/40 text-primary-foreground bg-transparent hover:bg-primary-foreground/10 font-semibold px-8 h-12 gap-2 w-full sm:w-auto",
                  "data-ocid": "beranda.tentang_jdih_button",
                  children: /* @__PURE__ */ jsxRuntimeExports.jsxs(Link, { to: "/tentang", children: [
                    /* @__PURE__ */ jsxRuntimeExports.jsx(BookOpen, { className: "w-4 h-4" }),
                    "Tentang JDIH"
                  ] })
                }
              )
            ] }),
            /* @__PURE__ */ jsxRuntimeExports.jsxs(
              "form",
              {
                onSubmit: handleSearch,
                className: "flex gap-2 max-w-2xl mx-auto",
                "data-ocid": "beranda.search_form",
                children: [
                  /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex-1 relative", children: [
                    /* @__PURE__ */ jsxRuntimeExports.jsx(Search, { className: "absolute left-3.5 top-1/2 -translate-y-1/2 w-4 h-4 text-muted-foreground pointer-events-none" }),
                    /* @__PURE__ */ jsxRuntimeExports.jsx(
                      Input,
                      {
                        type: "text",
                        placeholder: "Cari dokumen hukum... (misal: Statuta UNTAG)",
                        value: searchQuery,
                        onChange: (e) => setSearchQuery(e.target.value),
                        className: "pl-10 h-12 text-foreground bg-card border-0 shadow-md text-sm",
                        "data-ocid": "beranda.search_input"
                      }
                    )
                  ] }),
                  /* @__PURE__ */ jsxRuntimeExports.jsxs(
                    Button,
                    {
                      type: "submit",
                      className: "h-12 px-5 bg-accent hover:bg-accent/90 text-accent-foreground font-semibold gap-2 shrink-0 shadow-md",
                      "data-ocid": "beranda.search_submit_button",
                      children: [
                        /* @__PURE__ */ jsxRuntimeExports.jsx(Search, { className: "w-4 h-4" }),
                        /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "hidden sm:inline", children: "Cari" })
                      ]
                    }
                  )
                ]
              }
            )
          ] }) })
        ]
      }
    ),
    /* @__PURE__ */ jsxRuntimeExports.jsx(
      "section",
      {
        className: "bg-card border-b border-border shadow-subtle",
        "data-ocid": "beranda.stats_section",
        children: /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "container mx-auto px-4", children: /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "grid grid-cols-2 md:grid-cols-4 divide-x divide-border", children: STATS.map((stat) => {
          const Icon = stat.icon;
          return /* @__PURE__ */ jsxRuntimeExports.jsxs(
            "div",
            {
              className: "py-6 px-4 flex items-center gap-3",
              "data-ocid": `beranda.stat_card.${stat.key}`,
              children: [
                /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "w-10 h-10 bg-primary/8 rounded-lg flex items-center justify-center flex-shrink-0 hidden sm:flex", children: /* @__PURE__ */ jsxRuntimeExports.jsx(Icon, { className: `w-5 h-5 ${stat.color}` }) }),
                /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { children: [
                  loadingStats ? /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "skeleton h-6 w-12 mb-1" }) : /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "text-2xl font-display font-bold text-foreground leading-none mb-1", children: stat.value.toLocaleString("id-ID") }),
                  /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "text-xs text-muted-foreground font-medium leading-tight", children: stat.label })
                ] })
              ]
            },
            stat.key
          );
        }) }) })
      }
    ),
    /* @__PURE__ */ jsxRuntimeExports.jsx(
      "section",
      {
        className: "py-12 bg-secondary/30",
        "data-ocid": "beranda.kategori_section",
        children: /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "container mx-auto px-4", children: [
          /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex items-center justify-between mb-7", children: [
            /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { children: [
              /* @__PURE__ */ jsxRuntimeExports.jsx("h2", { className: "font-display font-bold text-xl text-foreground", children: "Kategori Produk Hukum" }),
              /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-muted-foreground text-sm mt-1", children: "Telusuri berdasarkan jenis dokumen" })
            ] }),
            /* @__PURE__ */ jsxRuntimeExports.jsxs(
              Link,
              {
                to: "/katalog",
                className: "text-accent text-sm font-semibold hover:underline flex items-center gap-1 transition-smooth",
                "data-ocid": "beranda.kategori_semua_link",
                children: [
                  "Lihat Semua ",
                  /* @__PURE__ */ jsxRuntimeExports.jsx(ChevronRight, { className: "w-4 h-4" })
                ]
              }
            )
          ] }),
          /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "grid grid-cols-2 sm:grid-cols-3 lg:grid-cols-6 gap-3", children: Object.entries(JENIS_LABEL).map(([key, label], idx) => {
            const Icon = KATEGORI_ICONS[idx] ?? FileText;
            return /* @__PURE__ */ jsxRuntimeExports.jsxs(
              Link,
              {
                to: "/katalog",
                className: "bg-card border border-border rounded-lg p-4 text-center hover:border-primary/40 hover:shadow-card transition-smooth group flex flex-col items-center gap-2",
                "data-ocid": `beranda.kategori_card.${idx + 1}`,
                children: [
                  /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "w-10 h-10 bg-primary/8 rounded-lg flex items-center justify-center group-hover:bg-primary/15 transition-smooth", children: /* @__PURE__ */ jsxRuntimeExports.jsx(Icon, { className: "w-5 h-5 text-primary" }) }),
                  /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "text-xs font-semibold text-foreground leading-snug group-hover:text-primary transition-smooth text-center", children: label })
                ]
              },
              key
            );
          }) })
        ] })
      }
    ),
    /* @__PURE__ */ jsxRuntimeExports.jsx(
      "section",
      {
        className: "py-12 bg-background",
        "data-ocid": "beranda.dokumen_terbaru_section",
        children: /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "container mx-auto px-4", children: [
          /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex items-center justify-between mb-7", children: [
            /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { children: [
              /* @__PURE__ */ jsxRuntimeExports.jsx("h2", { className: "font-display font-bold text-xl text-foreground", children: "Dokumen Terbaru" }),
              /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-muted-foreground text-sm mt-1", children: "Produk hukum yang baru diterbitkan" })
            ] }),
            /* @__PURE__ */ jsxRuntimeExports.jsxs(
              Link,
              {
                to: "/katalog",
                className: "text-accent text-sm font-semibold hover:underline flex items-center gap-1 transition-smooth",
                "data-ocid": "beranda.dokumen_semua_link",
                children: [
                  "Lihat Semua ",
                  /* @__PURE__ */ jsxRuntimeExports.jsx(ArrowRight, { className: "w-4 h-4" })
                ]
              }
            )
          ] }),
          /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-5", children: loadingDokumen ? SKELETON_KEYS.map((k) => /* @__PURE__ */ jsxRuntimeExports.jsx(SkeletonCard, {}, k)) : (dokumen ?? []).slice(0, 6).map((doc, idx) => /* @__PURE__ */ jsxRuntimeExports.jsxs(
            "div",
            {
              className: "bg-card border border-border rounded-lg p-5 hover:shadow-card hover:border-border/60 transition-smooth flex flex-col",
              "data-ocid": `beranda.dokumen_card.${idx + 1}`,
              children: [
                /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex items-start justify-between gap-2 mb-3", children: [
                  /* @__PURE__ */ jsxRuntimeExports.jsx(KategoriTag, { nama: doc.kategoriNama ?? "-" }),
                  /* @__PURE__ */ jsxRuntimeExports.jsx(StatusBadge, { status: doc.status })
                ] }),
                /* @__PURE__ */ jsxRuntimeExports.jsx("h3", { className: "font-semibold text-foreground text-sm leading-snug mb-2 flex-1 line-clamp-3", children: doc.judul }),
                /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "font-doc-number text-muted-foreground mb-3 truncate", children: doc.nomor }),
                /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "flex items-center justify-between text-xs text-muted-foreground mb-4 gap-2", children: /* @__PURE__ */ jsxRuntimeExports.jsxs("span", { className: "flex items-center gap-1", children: [
                  /* @__PURE__ */ jsxRuntimeExports.jsx(Calendar, { className: "w-3 h-3" }),
                  "Tahun ",
                  doc.tahun
                ] }) }),
                /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex gap-2 mt-auto", children: [
                  /* @__PURE__ */ jsxRuntimeExports.jsx(
                    Button,
                    {
                      asChild: true,
                      size: "sm",
                      className: "flex-1 bg-accent hover:bg-accent/90 text-accent-foreground gap-1.5 text-xs",
                      "data-ocid": `beranda.unduh_button.${idx + 1}`,
                      children: /* @__PURE__ */ jsxRuntimeExports.jsxs(Link, { to: "/katalog/$id", params: { id: String(doc.id) }, children: [
                        /* @__PURE__ */ jsxRuntimeExports.jsx(Download, { className: "w-3.5 h-3.5" }),
                        "Unduh PDF"
                      ] })
                    }
                  ),
                  /* @__PURE__ */ jsxRuntimeExports.jsx(
                    Button,
                    {
                      asChild: true,
                      variant: "outline",
                      size: "sm",
                      className: "border-border text-xs hover:border-primary/40 hover:text-primary transition-smooth",
                      "data-ocid": `beranda.detail_button.${idx + 1}`,
                      children: /* @__PURE__ */ jsxRuntimeExports.jsx(Link, { to: "/katalog/$id", params: { id: String(doc.id) }, children: "Detail" })
                    }
                  )
                ] })
              ]
            },
            doc.id
          )) }),
          /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "text-center mt-8", children: /* @__PURE__ */ jsxRuntimeExports.jsx(
            Button,
            {
              asChild: true,
              variant: "outline",
              className: "border-primary/30 text-primary hover:bg-primary/5 hover:border-primary/50 transition-smooth font-semibold gap-2",
              "data-ocid": "beranda.lihat_semua_dokumen_button",
              children: /* @__PURE__ */ jsxRuntimeExports.jsxs(Link, { to: "/katalog", children: [
                "Lihat Semua Dokumen",
                /* @__PURE__ */ jsxRuntimeExports.jsx(ArrowRight, { className: "w-4 h-4" })
              ] })
            }
          ) })
        ] })
      }
    ),
    /* @__PURE__ */ jsxRuntimeExports.jsx("section", { className: "py-12 bg-muted/40", "data-ocid": "beranda.berita_section", children: /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "container mx-auto px-4", children: [
      /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex items-center justify-between mb-7", children: [
        /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { children: [
          /* @__PURE__ */ jsxRuntimeExports.jsx("h2", { className: "font-display font-bold text-xl text-foreground", children: "Berita & Pengumuman" }),
          /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-muted-foreground text-sm mt-1", children: "Informasi terkini seputar JDIH UNTAG Banyuwangi" })
        ] }),
        /* @__PURE__ */ jsxRuntimeExports.jsxs(
          Link,
          {
            to: "/berita",
            className: "text-accent text-sm font-semibold hover:underline flex items-center gap-1 transition-smooth",
            "data-ocid": "beranda.berita_semua_link",
            children: [
              "Lihat Semua ",
              /* @__PURE__ */ jsxRuntimeExports.jsx(ArrowRight, { className: "w-4 h-4" })
            ]
          }
        )
      ] }),
      loadingBerita ? /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "grid grid-cols-1 md:grid-cols-3 gap-5", children: BERITA_SKELETON_KEYS.map((k) => /* @__PURE__ */ jsxRuntimeExports.jsxs(
        "div",
        {
          className: "bg-card border border-border rounded-lg p-5 animate-pulse space-y-3",
          children: [
            /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "skeleton h-3 w-24" }),
            /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "skeleton h-4 w-full" }),
            /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "skeleton h-4 w-4/5" }),
            /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "skeleton h-3 w-full" }),
            /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "skeleton h-3 w-3/4" })
          ]
        },
        k
      )) }) : /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "grid grid-cols-1 md:grid-cols-3 gap-5", children: (berita ?? []).slice(0, 3).map((artikel, idx) => /* @__PURE__ */ jsxRuntimeExports.jsxs(
        Link,
        {
          to: "/berita/$id",
          params: { id: String(artikel.id) },
          className: "bg-card border border-border rounded-lg overflow-hidden hover:shadow-card hover:border-border/60 transition-smooth flex flex-col group",
          "data-ocid": `beranda.berita_card.${idx + 1}`,
          children: [
            /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "w-full h-40 bg-secondary/50 flex items-center justify-center flex-shrink-0 border-b border-border relative overflow-hidden", children: [
              /* @__PURE__ */ jsxRuntimeExports.jsx(Newspaper, { className: "w-12 h-12 text-muted-foreground/25" }),
              /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "absolute inset-0 bg-gradient-to-br from-primary/5 to-transparent" })
            ] }),
            /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "p-5 flex-1 flex flex-col", children: [
              /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex items-center gap-2 mb-3", children: [
                /* @__PURE__ */ jsxRuntimeExports.jsx(Calendar, { className: "w-3.5 h-3.5 text-muted-foreground" }),
                /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-xs text-muted-foreground font-medium", children: artikel.tanggalPublikasi })
              ] }),
              /* @__PURE__ */ jsxRuntimeExports.jsx("h3", { className: "font-display font-semibold text-foreground text-sm leading-snug mb-2 group-hover:text-primary transition-smooth line-clamp-2 flex-1", children: artikel.judul }),
              /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-muted-foreground text-xs leading-relaxed line-clamp-3 mb-4", children: (artikel.ringkasan ?? "").length > 150 ? `${(artikel.ringkasan ?? "").substring(0, 150)}...` : artikel.ringkasan ?? "" }),
              /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex items-center gap-1 text-xs text-accent font-semibold group-hover:gap-2 transition-smooth mt-auto", children: [
                "Baca Selengkapnya",
                /* @__PURE__ */ jsxRuntimeExports.jsx(ArrowRight, { className: "w-3.5 h-3.5" })
              ] })
            ] })
          ]
        },
        artikel.id
      )) }),
      /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "text-center mt-8", children: /* @__PURE__ */ jsxRuntimeExports.jsx(
        Button,
        {
          asChild: true,
          variant: "outline",
          className: "border-primary/30 text-primary hover:bg-primary/5 hover:border-primary/50 transition-smooth font-semibold gap-2",
          "data-ocid": "beranda.lihat_semua_berita_button",
          children: /* @__PURE__ */ jsxRuntimeExports.jsxs(Link, { to: "/berita", children: [
            "Lihat Semua Berita",
            /* @__PURE__ */ jsxRuntimeExports.jsx(ArrowRight, { className: "w-4 h-4" })
          ] })
        }
      ) })
    ] }) }),
    /* @__PURE__ */ jsxRuntimeExports.jsx(
      "section",
      {
        className: "py-12 bg-primary mb-10",
        "data-ocid": "beranda.banner_section",
        children: /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "container mx-auto px-4 text-center", children: [
          /* @__PURE__ */ jsxRuntimeExports.jsx("h2", { className: "font-display font-bold text-xl md:text-2xl text-primary-foreground mb-3", children: "Butuh Bantuan?" }),
          /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-primary-foreground/70 text-sm max-w-md mx-auto mb-7 leading-relaxed", children: "Tim JDIH UNTAG Banyuwangi siap membantu Anda menemukan dokumen hukum yang dibutuhkan." }),
          /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex flex-col sm:flex-row items-center justify-center gap-3", children: [
            /* @__PURE__ */ jsxRuntimeExports.jsx(
              Button,
              {
                asChild: true,
                size: "lg",
                className: "bg-accent hover:bg-accent/90 text-accent-foreground font-semibold gap-2 w-full sm:w-auto",
                "data-ocid": "beranda.hubungi_kami_button",
                children: /* @__PURE__ */ jsxRuntimeExports.jsxs(Link, { to: "/kontak", children: [
                  "Hubungi Kami",
                  /* @__PURE__ */ jsxRuntimeExports.jsx(ArrowRight, { className: "w-4 h-4" })
                ] })
              }
            ),
            /* @__PURE__ */ jsxRuntimeExports.jsx(
              Button,
              {
                asChild: true,
                size: "lg",
                variant: "outline",
                className: "border-primary-foreground/40 text-primary-foreground bg-transparent hover:bg-primary-foreground/10 font-semibold gap-2 w-full sm:w-auto",
                "data-ocid": "beranda.katalog_button",
                children: /* @__PURE__ */ jsxRuntimeExports.jsx(Link, { to: "/katalog", children: "Lihat Katalog Dokumen" })
              }
            )
          ] })
        ] })
      }
    ),
    /* @__PURE__ */ jsxRuntimeExports.jsx(
      "section",
      {
        className: "py-14 bg-background border-t border-border",
        "data-ocid": "beranda.galeri_section",
        children: /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "container mx-auto px-4", children: [
          /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex items-center justify-between mb-7", children: [
            /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { children: [
              /* @__PURE__ */ jsxRuntimeExports.jsx("h2", { className: "font-display font-bold text-xl text-foreground", children: "Galeri" }),
              /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-muted-foreground text-sm mt-1", children: "Dokumentasi kegiatan dan fasilitas kampus UNTAG Banyuwangi" })
            ] }),
            /* @__PURE__ */ jsxRuntimeExports.jsxs(
              Link,
              {
                to: "/galeri",
                className: "text-primary text-sm font-semibold hover:underline flex items-center gap-1 transition-smooth",
                "data-ocid": "beranda.galeri_semua_link",
                children: [
                  "Lihat Semua ",
                  /* @__PURE__ */ jsxRuntimeExports.jsx(ChevronRight, { className: "w-4 h-4" })
                ]
              }
            )
          ] }),
          loadingGaleri ? /* @__PURE__ */ jsxRuntimeExports.jsx(
            "div",
            {
              className: "grid grid-cols-2 sm:grid-cols-3 lg:grid-cols-4 gap-3",
              "data-ocid": "beranda.galeri_loading_state",
              children: GALERI_SKELETON_KEYS.map((k) => /* @__PURE__ */ jsxRuntimeExports.jsx(Skeleton, { className: "aspect-square rounded-lg" }, k))
            }
          ) : galeriPreview.length === 0 ? /* @__PURE__ */ jsxRuntimeExports.jsxs(
            "div",
            {
              className: "flex flex-col items-center justify-center py-16 text-center",
              "data-ocid": "beranda.galeri_empty_state",
              children: [
                /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "w-16 h-16 rounded-full bg-muted flex items-center justify-center mb-4", children: /* @__PURE__ */ jsxRuntimeExports.jsx(Image, { className: "w-8 h-8 text-muted-foreground/40" }) }),
                /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-foreground font-semibold mb-1", children: "Belum ada foto tersedia" }),
                /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-muted-foreground text-sm", children: "Foto kegiatan akan ditampilkan di sini setelah diunggah oleh admin." })
              ]
            }
          ) : /* @__PURE__ */ jsxRuntimeExports.jsx(
            "div",
            {
              className: "grid grid-cols-2 sm:grid-cols-3 lg:grid-cols-4 gap-3",
              "data-ocid": "beranda.galeri_foto_list",
              children: galeriPreview.map((item, idx) => /* @__PURE__ */ jsxRuntimeExports.jsx(
                GaleriPreviewCard,
                {
                  item,
                  index: idx + 1,
                  onOpen: setGaleriLightbox
                },
                item.id
              ))
            }
          ),
          /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "text-center mt-8", children: /* @__PURE__ */ jsxRuntimeExports.jsx(
            Button,
            {
              asChild: true,
              className: "bg-primary hover:bg-primary/90 text-primary-foreground font-semibold gap-2 px-8",
              "data-ocid": "beranda.lihat_semua_galeri_button",
              children: /* @__PURE__ */ jsxRuntimeExports.jsxs(Link, { to: "/galeri", children: [
                "Lihat Semua Foto",
                /* @__PURE__ */ jsxRuntimeExports.jsx(ArrowRight, { className: "w-4 h-4" })
              ] })
            }
          ) })
        ] })
      }
    ),
    /* @__PURE__ */ jsxRuntimeExports.jsx(
      GaleriLightbox,
      {
        item: galeriLightbox,
        onClose: () => setGaleriLightbox(null)
      }
    )
  ] });
}
export {
  Beranda
};
