// @caffeine/template-frontend/dist/assets/Berita-DLtEGBHe.js
import { j as jsxRuntimeExports, L as Layout, b as Link, N as Newspaper } from "./index-B7-RSq-m.js";
import { E as EmptyState } from "./EmptyState-Bzc48I5U.js";
import { E as ErrorState } from "./ErrorState-o51Yfgm5.js";
import { P as PageHeader } from "./PageHeader-CmrF1ENQ.js";
import { b as useBerita } from "./useBackend-xFEMRG87.js";
import { C as Calendar } from "./calendar-mZT-2HLQ.js";
import { U as User } from "./user-CWnTlCz9.js";
import { A as ArrowRight } from "./arrow-right-pgjRVcXN.js";
import "./circle-alert-D-S0YTKb.js";
import "./BreadcrumbNav---FAxtpV.js";
import "./chevron-right-RI9IPXg2.js";
import "./useMutation-D--BDafu.js";
function formatTanggalIndonesia(dateStr) {
  if (!dateStr) return "-";
  const BULAN = [
    "Januari",
    "Februari",
    "Maret",
    "April",
    "Mei",
    "Juni",
    "Juli",
    "Agustus",
    "September",
    "Oktober",
    "November",
    "Desember"
  ];
  const date = new Date(dateStr);
  if (Number.isNaN(date.getTime())) return dateStr;
  return `${date.getDate()} ${BULAN[date.getMonth()]} ${date.getFullYear()}`;
}
const SKELETON_KEYS = ["a", "b", "c", "d"];
function SkeletonBeritaGrid() {
  return /* @__PURE__ */ jsxRuntimeExports.jsx(
    "div",
    {
      className: "grid grid-cols-1 md:grid-cols-2 gap-5",
      "data-ocid": "berita.loading_state",
      children: SKELETON_KEYS.map((key) => /* @__PURE__ */ jsxRuntimeExports.jsxs(
        "div",
        {
          className: "bg-card border border-border rounded-lg overflow-hidden animate-pulse",
          children: [
            /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "skeleton h-44 w-full" }),
            /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "p-5 space-y-3", children: [
              /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex gap-3", children: [
                /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "skeleton h-3 w-28 rounded" }),
                /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "skeleton h-3 w-20 rounded" })
              ] }),
              /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "skeleton h-5 w-4/5 rounded" }),
              /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "skeleton h-4 w-full rounded" }),
              /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "skeleton h-4 w-3/4 rounded" }),
              /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "skeleton h-4 w-5/6 rounded" }),
              /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "skeleton h-4 w-24 rounded mt-2" })
            ] })
          ]
        },
        key
      ))
    }
  );
}
function ArtikelCard({ artikel, index }) {
  const ringkasan = artikel.ringkasan.length > 200 ? `${artikel.ringkasan.substring(0, 200)}…` : artikel.ringkasan;
  const tanggal = formatTanggalIndonesia(artikel.tanggalPublikasi ?? artikel.tanggal);
  const coverUrl = artikel.gambar ?? null;
  return /* @__PURE__ */ jsxRuntimeExports.jsxs(
    Link,
    {
      to: "/berita/$id",
      params: { id: String(artikel.id) },
      className: "bg-card border border-border rounded-lg overflow-hidden hover:shadow-card transition-smooth group flex flex-col",
      "data-ocid": `berita.artikel_card.${index}`,
      children: [
        coverUrl ? /* @__PURE__ */ jsxRuntimeExports.jsx(
          "img",
          {
            src: coverUrl,
            alt: artikel.judul,
            className: "h-44 w-full object-cover border-b border-border",
            onError: (e) => {
              e.currentTarget.style.display = "none";
            }
          }
        ) : /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "h-44 bg-secondary flex items-center justify-center border-b border-border", children: /* @__PURE__ */ jsxRuntimeExports.jsx(Newspaper, { className: "w-12 h-12 text-primary/20" }) }),
        /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "p-5 flex flex-col flex-1", children: [
          /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex flex-wrap items-center gap-3 text-xs text-muted-foreground mb-3", children: [
            /* @__PURE__ */ jsxRuntimeExports.jsxs("span", { className: "flex items-center gap-1.5", children: [
              /* @__PURE__ */ jsxRuntimeExports.jsx(Calendar, { className: "w-3.5 h-3.5 flex-shrink-0" }),
              tanggal
            ] }),
            /* @__PURE__ */ jsxRuntimeExports.jsxs("span", { className: "flex items-center gap-1.5 min-w-0", children: [
              /* @__PURE__ */ jsxRuntimeExports.jsx(User, { className: "w-3.5 h-3.5 flex-shrink-0" }),
              /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "truncate", children: artikel.author ?? "" })
            ] })
          ] }),
          /* @__PURE__ */ jsxRuntimeExports.jsx("h2", { className: "font-display font-semibold text-foreground text-base leading-snug mb-2 group-hover:text-accent transition-smooth line-clamp-2", children: artikel.judul }),
          /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-muted-foreground text-sm leading-relaxed flex-1", children: ringkasan }),
          /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex items-center gap-1 text-accent text-sm font-semibold mt-4 group-hover:gap-2 transition-smooth", children: [
            "Baca Selengkapnya ",
            /* @__PURE__ */ jsxRuntimeExports.jsx(ArrowRight, { className: "w-4 h-4" })
          ] })
        ] })
      ]
    }
  );
}
function Berita() {
  const { data: artikelList, isLoading, isError, refetch } = useBerita();
  return /* @__PURE__ */ jsxRuntimeExports.jsxs(Layout, { children: [
    /* @__PURE__ */ jsxRuntimeExports.jsx(
      PageHeader,
      {
        title: "Berita & Pengumuman",
        description: "Informasi terkini seputar kegiatan dan regulasi JDIH UNTAG Banyuwangi",
        breadcrumbs: [{ label: "Berita & Pengumuman" }]
      }
    ),
    /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "container mx-auto px-4 py-8", "data-ocid": "berita.list", children: isLoading ? /* @__PURE__ */ jsxRuntimeExports.jsx(SkeletonBeritaGrid, {}) : isError ? /* @__PURE__ */ jsxRuntimeExports.jsx(ErrorState, { onRetry: () => refetch() }) : !artikelList || artikelList.length === 0 ? /* @__PURE__ */ jsxRuntimeExports.jsx(
      EmptyState,
      {
        title: "Belum ada berita",
        description: "Belum ada artikel berita yang dipublikasikan. Silakan kembali lagi nanti."
      }
    ) : /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "grid grid-cols-1 md:grid-cols-2 gap-5", children: artikelList.map((artikel, idx) => /* @__PURE__ */ jsxRuntimeExports.jsx(ArtikelCard, { artikel, index: idx + 1 }, artikel.id)) }) })
  ] });
}
export {
  Berita
};
