// @caffeine/template-frontend/dist/assets/BeritaDetail-CxdyAB_s.js
import { f as useParams, j as jsxRuntimeExports, L as Layout, g as LoadingSpinner, a as Button, b as Link, A as ArrowLeft } from "./index-B7-RSq-m.js";
import { B as Badge } from "./badge-KXcDp_20.js";
import { S as Separator } from "./separator-CrFHvepy.js";
import { E as ErrorState } from "./ErrorState-o51Yfgm5.js";
import { P as PageHeader } from "./PageHeader-CmrF1ENQ.js";
import { f as useBeritaDetail } from "./useBackend-xFEMRG87.js";
import { C as Calendar } from "./calendar-mZT-2HLQ.js";
import { U as User } from "./user-CWnTlCz9.js";
import { T as Tag } from "./tag-DZ8TZA2a.js";
import "./index-BcOIygc1.js";
import "./circle-alert-D-S0YTKb.js";
import "./BreadcrumbNav---FAxtpV.js";
import "./chevron-right-RI9IPXg2.js";
import "./useMutation-D--BDafu.js";
function formatTanggalIndonesia(dateStr) {
  if (!dateStr) return "-";
  const BULAN = [
    "Januari",
    "Februari",
    "Maret",
    "April",
    "Mei",
    "Juni",
    "Juli",
    "Agustus",
    "September",
    "Oktober",
    "November",
    "Desember"
  ];
  const date = new Date(dateStr);
  if (Number.isNaN(date.getTime())) return dateStr;
  return `${date.getDate()} ${BULAN[date.getMonth()]} ${date.getFullYear()}`;
}
function ArticleContent({ isi }) {
  const paragraphs = isi.split(/\n\n+/).filter((p) => p.trim().length > 0);
  if (paragraphs.length <= 1) {
    const lines = isi.split("\n");
    return /* @__PURE__ */ jsxRuntimeExports.jsx(
      "div",
      {
        className: "space-y-3 text-foreground leading-relaxed",
        "data-ocid": "berita_detail.konten",
        children: lines.map(
          (line, i) => line.trim() ? (
            // biome-ignore lint/suspicious/noArrayIndexKey: static content
            /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-base leading-7", children: line }, i)
          ) : (
            // biome-ignore lint/suspicious/noArrayIndexKey: static content
            /* @__PURE__ */ jsxRuntimeExports.jsx("br", {}, i)
          )
        )
      }
    );
  }
  return /* @__PURE__ */ jsxRuntimeExports.jsx(
    "div",
    {
      className: "space-y-4 text-foreground leading-relaxed",
      "data-ocid": "berita_detail.konten",
      children: paragraphs.map((para, i) => (
        // biome-ignore lint/suspicious/noArrayIndexKey: static content
        /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-base leading-7", children: para.trim() }, i)
      ))
    }
  );
}
function BeritaDetail() {
  var _a;
  const { id } = useParams({ from: "/berita/$id" });
  const { data: artikel, isLoading, isError, refetch } = useBeritaDetail(id);
  if (isLoading) {
    return /* @__PURE__ */ jsxRuntimeExports.jsx(Layout, { children: /* @__PURE__ */ jsxRuntimeExports.jsx(
      "div",
      {
        className: "min-h-[60vh] flex items-center justify-center",
        "data-ocid": "berita_detail.loading_state",
        children: /* @__PURE__ */ jsxRuntimeExports.jsx(LoadingSpinner, { size: "lg", text: "Memuat artikel..." })
      }
    ) });
  }
  if (isError || !artikel) {
    return /* @__PURE__ */ jsxRuntimeExports.jsx(Layout, { children: /* @__PURE__ */ jsxRuntimeExports.jsx("div", { "data-ocid": "berita_detail.error_state", children: /* @__PURE__ */ jsxRuntimeExports.jsx(
      ErrorState,
      {
        title: "Artikel tidak ditemukan",
        message: "Artikel yang Anda cari tidak tersedia atau telah dihapus.",
        onRetry: () => refetch(),
        className: "min-h-[60vh]"
      }
    ) }) });
  }
  const tanggal = formatTanggalIndonesia(artikel.tanggalPublikasi ?? artikel.tanggal);
  const coverUrl = artikel.gambar ?? null;
  const penulisNama = artikel.author ?? "-";
  const isiKonten = artikel.konten ?? "";
  return /* @__PURE__ */ jsxRuntimeExports.jsxs(Layout, { children: [
    /* @__PURE__ */ jsxRuntimeExports.jsx(
      PageHeader,
      {
        title: artikel.judul,
        breadcrumbs: [
          { label: "Berita & Pengumuman", to: "/berita" },
          { label: artikel.judul }
        ]
      }
    ),
    /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "container mx-auto px-4 py-8", children: /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "max-w-3xl mx-auto", children: [
      coverUrl && /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "mb-6 rounded-lg overflow-hidden border border-border", children: /* @__PURE__ */ jsxRuntimeExports.jsx(
        "img",
        {
          src: coverUrl,
          alt: artikel.judul,
          className: "w-full max-h-72 object-cover",
          onError: (e) => {
            var _a2;
            (_a2 = e.currentTarget.parentElement) == null ? void 0 : _a2.remove();
          }
        }
      ) }),
      /* @__PURE__ */ jsxRuntimeExports.jsxs(
        "div",
        {
          className: "flex flex-wrap items-center gap-4 text-sm text-muted-foreground mb-4",
          "data-ocid": "berita_detail.meta",
          children: [
            /* @__PURE__ */ jsxRuntimeExports.jsxs("span", { className: "flex items-center gap-1.5", children: [
              /* @__PURE__ */ jsxRuntimeExports.jsx(Calendar, { className: "w-4 h-4 flex-shrink-0" }),
              tanggal
            ] }),
            /* @__PURE__ */ jsxRuntimeExports.jsxs("span", { className: "flex items-center gap-1.5", children: [
              /* @__PURE__ */ jsxRuntimeExports.jsx(User, { className: "w-4 h-4 flex-shrink-0" }),
              penulisNama
            ] })
          ]
        }
      ),
      /* @__PURE__ */ jsxRuntimeExports.jsx(Separator, { className: "mb-6" }),
      /* @__PURE__ */ jsxRuntimeExports.jsx(ArticleContent, { isi: isiKonten }),
      (((_a = artikel.tags) == null ? void 0 : _a.length) ?? 0) > 0 && /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "mt-8 pt-6 border-t border-border", children: [
        /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex items-center gap-2 mb-3", children: [
          /* @__PURE__ */ jsxRuntimeExports.jsx(Tag, { className: "w-4 h-4 text-muted-foreground" }),
          /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "text-sm font-medium text-foreground", children: "Tag" })
        ] }),
        /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "flex flex-wrap gap-2", children: artikel.tags.map((tag) => /* @__PURE__ */ jsxRuntimeExports.jsx(
          Badge,
          {
            variant: "secondary",
            className: "text-xs",
            "data-ocid": "berita_detail.tag",
            children: tag
          },
          tag
        )) })
      ] }),
      /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "mt-8", children: /* @__PURE__ */ jsxRuntimeExports.jsx(
        Button,
        {
          asChild: true,
          variant: "outline",
          className: "gap-2 border-border",
          "data-ocid": "berita_detail.kembali_button",
          children: /* @__PURE__ */ jsxRuntimeExports.jsxs(Link, { to: "/berita", children: [
            /* @__PURE__ */ jsxRuntimeExports.jsx(ArrowLeft, { className: "w-4 h-4" }),
            "Kembali ke Berita"
          ] })
        }
      ) })
    ] }) })
  ] });
}
export {
  BeritaDetail
};
