// @caffeine/template-frontend/dist/assets/BreadcrumbNav---FAxtpV.js
import { c as createLucideIcon, j as jsxRuntimeExports, b as Link } from "./index-B7-RSq-m.js";
import { C as ChevronRight } from "./chevron-right-RI9IPXg2.js";
/**
 * @license lucide-react v0.511.0 - ISC
 *
 * This source code is licensed under the ISC license.
 * See the LICENSE file in the root directory of this source tree.
 */
const __iconNode = [
  ["path", { d: "M15 21v-8a1 1 0 0 0-1-1h-4a1 1 0 0 0-1 1v8", key: "5wwlr5" }],
  [
    "path",
    {
      d: "M3 10a2 2 0 0 1 .709-1.528l7-5.999a2 2 0 0 1 2.582 0l7 5.999A2 2 0 0 1 21 10v9a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2z",
      key: "1d0kgt"
    }
  ]
];
const House = createLucideIcon("house", __iconNode);
function BreadcrumbNav({ items, className = "" }) {
  return /* @__PURE__ */ jsxRuntimeExports.jsxs(
    "nav",
    {
      "aria-label": "Breadcrumb",
      className: `flex items-center gap-1 text-sm ${className}`,
      "data-ocid": "breadcrumb",
      children: [
        /* @__PURE__ */ jsxRuntimeExports.jsxs(
          Link,
          {
            to: "/",
            className: "text-muted-foreground hover:text-foreground transition-smooth flex items-center gap-1",
            children: [
              /* @__PURE__ */ jsxRuntimeExports.jsx(House, { className: "w-3.5 h-3.5" }),
              /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "sr-only", children: "Beranda" })
            ]
          }
        ),
        items.map((item, idx) => (
          // biome-ignore lint/suspicious/noArrayIndexKey: breadcrumb items are positional
          /* @__PURE__ */ jsxRuntimeExports.jsxs("span", { className: "flex items-center gap-1", children: [
            /* @__PURE__ */ jsxRuntimeExports.jsx(ChevronRight, { className: "w-3.5 h-3.5 text-muted-foreground/50 flex-shrink-0" }),
            item.to && idx < items.length - 1 ? /* @__PURE__ */ jsxRuntimeExports.jsx(
              Link,
              {
                to: item.to,
                className: "text-muted-foreground hover:text-foreground transition-smooth max-w-[180px] truncate",
                children: item.label
              }
            ) : /* @__PURE__ */ jsxRuntimeExports.jsx(
              "span",
              {
                className: "text-foreground font-medium max-w-[200px] truncate",
                "aria-current": idx === items.length - 1 ? "page" : void 0,
                children: item.label
              }
            )
          ] }, idx)
        ))
      ]
    }
  );
}
export {
  BreadcrumbNav as B
};
