// @caffeine/template-frontend/dist/assets/DokumenDetail-D2mLn9sP.js
import { c as createLucideIcon, f as useParams, j as jsxRuntimeExports, L as Layout, g as LoadingSpinner, a as Button, F as FileText, b as Link, A as ArrowLeft, r as reactExports, h as api } from "./index-B7-RSq-m.js";
import { S as Separator } from "./separator-CrFHvepy.js";
import { B as BreadcrumbNav } from "./BreadcrumbNav---FAxtpV.js";
import { E as ErrorState } from "./ErrorState-o51Yfgm5.js";
import { S as StatusBadge, a as STATUS_LABEL, K as KategoriTag } from "./StatusBadge-CanDaSra.js";
import { e as useDokumenDetail } from "./useBackend-xFEMRG87.js";
import { D as Download } from "./download-BnCFvid2.js";
import { C as Calendar } from "./calendar-mZT-2HLQ.js";
import { T as Tag } from "./tag-DZ8TZA2a.js";
import { C as CircleAlert } from "./circle-alert-D-S0YTKb.js";
import "./index-BcOIygc1.js";
import "./chevron-right-RI9IPXg2.js";
import "./badge-KXcDp_20.js";
import "./useMutation-D--BDafu.js";
/**
 * @license lucide-react v0.511.0 - ISC
 *
 * This source code is licensed under the ISC license.
 * See the LICENSE file in the root directory of this source tree.
 */
const __iconNode$1 = [
  [
    "path",
    {
      d: "M20 20a2 2 0 0 0 2-2V8a2 2 0 0 0-2-2h-7.9a2 2 0 0 1-1.69-.9L9.6 3.9A2 2 0 0 0 7.93 3H4a2 2 0 0 0-2 2v13a2 2 0 0 0 2 2Z",
      key: "1kt360"
    }
  ]
];
const Folder = createLucideIcon("folder", __iconNode$1);
/**
 * @license lucide-react v0.511.0 - ISC
 *
 * This source code is licensed under the ISC license.
 * See the LICENSE file in the root directory of this source tree.
 */
const __iconNode = [
  ["line", { x1: "4", x2: "20", y1: "9", y2: "9", key: "4lhtct" }],
  ["line", { x1: "4", x2: "20", y1: "15", y2: "15", key: "vyu0kd" }],
  ["line", { x1: "10", x2: "8", y1: "3", y2: "21", key: "1ggp8o" }],
  ["line", { x1: "16", x2: "14", y1: "3", y2: "21", key: "weycgp" }]
];
const Hash = createLucideIcon("hash", __iconNode);
function useDownloadDokumen() {
  const [isDownloading, setIsDownloading] = reactExports.useState(false);
  const download = reactExports.useCallback(
    async (id, filePdf, judul) => {
      setIsDownloading(true);
      try {
        await api.dokumen.download(id).catch(() => null);
        if (filePdf) {
          window.open(filePdf, "_blank", "noopener,noreferrer");
        } else {
          alert(`Dokumen "${judul}" belum tersedia untuk diunduh.`);
        }
      } finally {
        setIsDownloading(false);
      }
    },
    []
  );
  return { download, isDownloading };
}
function MetaRow({
  label,
  value,
  mono = false,
  icon
}) {
  return /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex flex-col gap-0.5", children: [
    /* @__PURE__ */ jsxRuntimeExports.jsx("dt", { className: "text-xs text-muted-foreground uppercase tracking-wider font-medium", children: label }),
    /* @__PURE__ */ jsxRuntimeExports.jsxs(
      "dd",
      {
        className: `text-sm text-foreground font-semibold flex items-center gap-1.5 ${mono ? "font-mono text-xs tracking-wide" : ""}`,
        children: [
          icon && /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "text-muted-foreground", children: icon }),
          value
        ]
      }
    )
  ] });
}
function DokumenDetail() {
  const { id } = useParams({ from: "/katalog/$id" });
  const { data: dokumen, isLoading, isError, refetch } = useDokumenDetail(id);
  const { download, isDownloading } = useDownloadDokumen();
  if (isLoading) {
    return /* @__PURE__ */ jsxRuntimeExports.jsx(Layout, { children: /* @__PURE__ */ jsxRuntimeExports.jsx(
      "div",
      {
        className: "min-h-[60vh] flex items-center justify-center",
        "data-ocid": "dokumen_detail.loading_state",
        children: /* @__PURE__ */ jsxRuntimeExports.jsx(LoadingSpinner, { size: "lg", text: "Memuat detail dokumen..." })
      }
    ) });
  }
  if (isError || !dokumen) {
    return /* @__PURE__ */ jsxRuntimeExports.jsx(Layout, { children: /* @__PURE__ */ jsxRuntimeExports.jsx("div", { "data-ocid": "dokumen_detail.error_state", children: /* @__PURE__ */ jsxRuntimeExports.jsx(
      ErrorState,
      {
        title: "Dokumen tidak ditemukan",
        message: "Dokumen yang Anda cari tidak tersedia atau telah dihapus.",
        onRetry: () => refetch(),
        className: "min-h-[60vh]"
      }
    ) }) });
  }
  const handleDownload = () => download(String(dokumen.id), dokumen.filePdf, dokumen.judul);
  const tags = dokumen.tag ?? [];
  return /* @__PURE__ */ jsxRuntimeExports.jsxs(Layout, { children: [
    /* @__PURE__ */ jsxRuntimeExports.jsx(
      "div",
      {
        className: "bg-card border-b border-border",
        "data-ocid": "dokumen_detail.page",
        children: /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "container mx-auto px-4 py-5", children: [
          /* @__PURE__ */ jsxRuntimeExports.jsx(
            BreadcrumbNav,
            {
              items: [
                { label: "Katalog Hukum", to: "/katalog" },
                { label: dokumen.judul }
              ]
            }
          ),
          /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "mt-4 flex flex-col sm:flex-row sm:items-start sm:justify-between gap-4", children: [
            /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "min-w-0 flex-1", children: [
              /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex items-center gap-2 mb-2", children: [
                /* @__PURE__ */ jsxRuntimeExports.jsx(Hash, { className: "w-3.5 h-3.5 text-muted-foreground flex-shrink-0" }),
                /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "font-mono text-xs tracking-wide text-muted-foreground", children: dokumen.nomor }),
                /* @__PURE__ */ jsxRuntimeExports.jsx(StatusBadge, { status: dokumen.status })
              ] }),
              /* @__PURE__ */ jsxRuntimeExports.jsx(
                "h1",
                {
                  className: "font-display font-bold text-xl sm:text-2xl text-foreground leading-tight break-words",
                  "data-ocid": "dokumen_detail.judul",
                  children: dokumen.judul
                }
              )
            ] }),
            /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "hidden sm:block flex-shrink-0", children: /* @__PURE__ */ jsxRuntimeExports.jsxs(
              Button,
              {
                onClick: handleDownload,
                disabled: isDownloading,
                className: "bg-accent hover:bg-accent/90 text-accent-foreground gap-2 font-semibold",
                "data-ocid": "dokumen_detail.unduh_button",
                children: [
                  /* @__PURE__ */ jsxRuntimeExports.jsx(Download, { className: "w-4 h-4" }),
                  isDownloading ? "Mengunduh..." : "Unduh Dokumen PDF"
                ]
              }
            ) })
          ] })
        ] })
      }
    ),
    /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "container mx-auto px-4 py-8", children: /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "grid grid-cols-1 lg:grid-cols-3 gap-6", children: [
      /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "lg:col-span-2 space-y-6", children: [
        /* @__PURE__ */ jsxRuntimeExports.jsxs(
          "div",
          {
            className: "bg-card border border-border rounded-lg p-6",
            "data-ocid": "dokumen_detail.metadata_card",
            children: [
              /* @__PURE__ */ jsxRuntimeExports.jsxs("h2", { className: "font-display font-semibold text-foreground text-sm uppercase tracking-wider mb-5 flex items-center gap-2", children: [
                /* @__PURE__ */ jsxRuntimeExports.jsx(FileText, { className: "w-4 h-4 text-accent" }),
                "Informasi Dokumen"
              ] }),
              /* @__PURE__ */ jsxRuntimeExports.jsxs("dl", { className: "grid grid-cols-1 sm:grid-cols-2 gap-5", children: [
                /* @__PURE__ */ jsxRuntimeExports.jsx(
                  MetaRow,
                  {
                    label: "Nomor Dokumen",
                    value: dokumen.nomor,
                    mono: true,
                    icon: /* @__PURE__ */ jsxRuntimeExports.jsx(Hash, { className: "w-3.5 h-3.5" })
                  }
                ),
                /* @__PURE__ */ jsxRuntimeExports.jsx(
                  MetaRow,
                  {
                    label: "Kategori Dokumen",
                    value: dokumen.kategoriNama ?? "-",
                    icon: /* @__PURE__ */ jsxRuntimeExports.jsx(Folder, { className: "w-3.5 h-3.5" })
                  }
                ),
                /* @__PURE__ */ jsxRuntimeExports.jsx(
                  MetaRow,
                  {
                    label: "Tahun Terbit",
                    value: String(dokumen.tahun),
                    icon: /* @__PURE__ */ jsxRuntimeExports.jsx(Calendar, { className: "w-3.5 h-3.5" })
                  }
                ),
                /* @__PURE__ */ jsxRuntimeExports.jsx(
                  MetaRow,
                  {
                    label: "Tanggal Penetapan",
                    value: dokumen.tanggalPenetapan ?? "-",
                    icon: /* @__PURE__ */ jsxRuntimeExports.jsx(Calendar, { className: "w-3.5 h-3.5" })
                  }
                ),
                dokumen.tanggalPengundangan && /* @__PURE__ */ jsxRuntimeExports.jsx(
                  MetaRow,
                  {
                    label: "Tanggal Pengundangan",
                    value: dokumen.tanggalPengundangan,
                    icon: /* @__PURE__ */ jsxRuntimeExports.jsx(Calendar, { className: "w-3.5 h-3.5" })
                  }
                ),
                /* @__PURE__ */ jsxRuntimeExports.jsx(MetaRow, { label: "Status", value: STATUS_LABEL[dokumen.status] ?? dokumen.status }),
                dokumen.relasiHukum && /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "sm:col-span-2", children: /* @__PURE__ */ jsxRuntimeExports.jsx(
                  MetaRow,
                  {
                    label: "Relasi Dengan Produk Hukum Lain",
                    value: dokumen.relasiHukum
                  }
                ) })
              ] })
            ]
          }
        ),
        /* @__PURE__ */ jsxRuntimeExports.jsxs(
          "div",
          {
            className: "bg-card border border-border rounded-lg p-6",
            "data-ocid": "dokumen_detail.abstrak_card",
            children: [
              /* @__PURE__ */ jsxRuntimeExports.jsxs("h2", { className: "font-display font-semibold text-foreground text-sm uppercase tracking-wider mb-3 flex items-center gap-2", children: [
                /* @__PURE__ */ jsxRuntimeExports.jsx(FileText, { className: "w-4 h-4 text-accent" }),
                "Abstrak / Ringkasan"
              ] }),
              /* @__PURE__ */ jsxRuntimeExports.jsx(Separator, { className: "mb-4" }),
              /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-foreground/80 text-sm leading-relaxed", children: dokumen.abstrak })
            ]
          }
        ),
        tags.length > 0 && /* @__PURE__ */ jsxRuntimeExports.jsxs(
          "div",
          {
            className: "bg-card border border-border rounded-lg p-6",
            "data-ocid": "dokumen_detail.tags_card",
            children: [
              /* @__PURE__ */ jsxRuntimeExports.jsxs("h2", { className: "font-display font-semibold text-foreground text-sm uppercase tracking-wider mb-4 flex items-center gap-2", children: [
                /* @__PURE__ */ jsxRuntimeExports.jsx(Tag, { className: "w-4 h-4 text-accent" }),
                "Tag / Subjek"
              ] }),
              /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "flex flex-wrap gap-2", children: tags.map((tag) => /* @__PURE__ */ jsxRuntimeExports.jsx(KategoriTag, { nama: tag }, tag)) })
            ]
          }
        ),
        /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "sm:hidden", children: /* @__PURE__ */ jsxRuntimeExports.jsxs(
          Button,
          {
            onClick: handleDownload,
            disabled: isDownloading,
            className: "w-full bg-accent hover:bg-accent/90 text-accent-foreground gap-2 font-semibold py-3 text-base",
            "data-ocid": "dokumen_detail.unduh_mobile_button",
            children: [
              /* @__PURE__ */ jsxRuntimeExports.jsx(Download, { className: "w-5 h-5" }),
              isDownloading ? "Mengunduh..." : "Unduh Dokumen PDF"
            ]
          }
        ) }),
        /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "flex items-center", children: /* @__PURE__ */ jsxRuntimeExports.jsx(
          Button,
          {
            asChild: true,
            variant: "outline",
            className: "gap-2 border-border",
            "data-ocid": "dokumen_detail.kembali_button",
            children: /* @__PURE__ */ jsxRuntimeExports.jsxs(Link, { to: "/katalog", children: [
              /* @__PURE__ */ jsxRuntimeExports.jsx(ArrowLeft, { className: "w-4 h-4" }),
              "Kembali ke Katalog"
            ] })
          }
        ) })
      ] }),
      /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "space-y-4", children: [
        /* @__PURE__ */ jsxRuntimeExports.jsxs(
          "div",
          {
            className: "bg-primary rounded-lg p-6 text-primary-foreground",
            "data-ocid": "dokumen_detail.download_card",
            children: [
              /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "w-12 h-12 bg-primary-foreground/10 rounded-lg flex items-center justify-center mb-4", children: /* @__PURE__ */ jsxRuntimeExports.jsx(FileText, { className: "w-6 h-6 text-primary-foreground" }) }),
              /* @__PURE__ */ jsxRuntimeExports.jsx("h3", { className: "font-semibold text-base mb-1", children: "Unduh Dokumen" }),
              /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-primary-foreground/70 text-xs mb-1", children: "Format PDF" }),
              /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-primary-foreground/50 text-xs mb-5", children: dokumen.downloadCount > 0 ? `${dokumen.downloadCount}× diunduh` : "Belum ada unduhan" }),
              /* @__PURE__ */ jsxRuntimeExports.jsxs(
                Button,
                {
                  onClick: handleDownload,
                  disabled: isDownloading,
                  className: "w-full bg-accent hover:bg-accent/90 text-accent-foreground gap-2 font-semibold",
                  "data-ocid": "dokumen_detail.unduh_pdf_button",
                  children: [
                    /* @__PURE__ */ jsxRuntimeExports.jsx(Download, { className: "w-4 h-4" }),
                    isDownloading ? "Mengunduh..." : "Unduh PDF"
                  ]
                }
              )
            ]
          }
        ),
        /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "bg-muted/60 border border-border rounded-lg p-4", children: /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex items-start gap-2.5", children: [
          /* @__PURE__ */ jsxRuntimeExports.jsx(CircleAlert, { className: "w-4 h-4 text-muted-foreground mt-0.5 flex-shrink-0" }),
          /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-muted-foreground text-xs leading-relaxed", children: "Pastikan Anda mengunduh dokumen dari sumber resmi JDIH UNTAG Banyuwangi untuk menjamin keasliannya." })
        ] }) }),
        /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "bg-card border border-border rounded-lg p-4 space-y-3", children: [
          /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex items-center justify-between", children: [
            /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "text-xs text-muted-foreground", children: "Status" }),
            /* @__PURE__ */ jsxRuntimeExports.jsx(StatusBadge, { status: dokumen.status })
          ] }),
          /* @__PURE__ */ jsxRuntimeExports.jsx(Separator, {}),
          /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex items-center justify-between", children: [
            /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "text-xs text-muted-foreground", children: "Kategori" }),
            /* @__PURE__ */ jsxRuntimeExports.jsx(KategoriTag, { nama: dokumen.kategoriNama ?? "-" })
          ] })
        ] }),
        /* @__PURE__ */ jsxRuntimeExports.jsx(
          Button,
          {
            asChild: true,
            variant: "outline",
            className: "w-full gap-2 border-border",
            "data-ocid": "dokumen_detail.kembali_sidebar_button",
            children: /* @__PURE__ */ jsxRuntimeExports.jsxs(Link, { to: "/katalog", children: [
              /* @__PURE__ */ jsxRuntimeExports.jsx(ArrowLeft, { className: "w-4 h-4" }),
              "Kembali ke Katalog"
            ] })
          }
        )
      ] })
    ] }) })
  ] });
}
export {
  DokumenDetail
};
