// @caffeine/template-frontend/dist/assets/Galeri-CMQh-0EG.js
import { r as reactExports, j as jsxRuntimeExports, L as Layout, a as Button, I as Image, X } from "./index-B7-RSq-m.js";
import { B as Badge } from "./badge-KXcDp_20.js";
import { D as Dialog, a as DialogContent } from "./dialog-CIgZYBSq.js";
import { S as Skeleton } from "./skeleton-Z-jxXZYe.js";
import { P as PageHeader } from "./PageHeader-CmrF1ENQ.js";
import { c as useGaleri } from "./useBackend-xFEMRG87.js";
import "./BreadcrumbNav---FAxtpV.js";
import "./chevron-right-RI9IPXg2.js";
import "./useMutation-D--BDafu.js";
const SKELETON_KEYS = ["a", "b", "c", "d", "e", "f", "g", "h"];
function GaleriSkeleton() {
  return /* @__PURE__ */ jsxRuntimeExports.jsx(
    "div",
    {
      className: "grid grid-cols-2 sm:grid-cols-3 lg:grid-cols-4 gap-3",
      "data-ocid": "galeri.loading_state",
      children: SKELETON_KEYS.map((k) => /* @__PURE__ */ jsxRuntimeExports.jsx(Skeleton, { className: "aspect-square rounded-lg" }, k))
    }
  );
}
function GaleriCard({
  item,
  index,
  onOpen
}) {
  const imgUrl = item.gambar ?? null;
  return /* @__PURE__ */ jsxRuntimeExports.jsxs(
    "button",
    {
      type: "button",
      onClick: () => onOpen(item),
      className: "group relative aspect-square overflow-hidden rounded-lg border border-border bg-muted focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-primary",
      "data-ocid": `galeri.foto_card.${index}`,
      children: [
        imgUrl ? /* @__PURE__ */ jsxRuntimeExports.jsx(
          "img",
          {
            src: imgUrl,
            alt: item.judul,
            className: "w-full h-full object-cover transition-transform duration-300 group-hover:scale-105",
            onError: (e) => {
              e.currentTarget.style.display = "none";
            }
          }
        ) : /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "w-full h-full flex items-center justify-center bg-muted", children: /* @__PURE__ */ jsxRuntimeExports.jsx(Image, { className: "w-10 h-10 text-muted-foreground/40" }) }),
        /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "absolute inset-0 bg-foreground/0 group-hover:bg-foreground/40 transition-all duration-300 flex flex-col justify-end p-3 opacity-0 group-hover:opacity-100", children: [
          /* @__PURE__ */ jsxRuntimeExports.jsx(Badge, { className: "self-start bg-primary text-primary-foreground text-xs mb-1.5 pointer-events-none", children: item.album }),
          /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-white text-sm font-semibold leading-snug line-clamp-2 text-left", children: item.judul })
        ] })
      ]
    }
  );
}
function Lightbox({
  item,
  onClose
}) {
  if (!item) return null;
  const imgUrl = item.gambar ?? null;
  return /* @__PURE__ */ jsxRuntimeExports.jsx(Dialog, { open: !!item, onOpenChange: (v) => !v && onClose(), children: /* @__PURE__ */ jsxRuntimeExports.jsx(
    DialogContent,
    {
      className: "max-w-3xl p-0 overflow-hidden bg-card border-border",
      "data-ocid": "galeri.lightbox_dialog",
      children: /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "relative", children: [
        /* @__PURE__ */ jsxRuntimeExports.jsx(
          "button",
          {
            type: "button",
            onClick: onClose,
            className: "absolute top-3 right-3 z-10 bg-foreground/60 hover:bg-foreground/80 text-white rounded-full p-1.5 transition-colors",
            "aria-label": "Tutup",
            "data-ocid": "galeri.lightbox.close_button",
            children: /* @__PURE__ */ jsxRuntimeExports.jsx(X, { className: "w-4 h-4" })
          }
        ),
        imgUrl ? /* @__PURE__ */ jsxRuntimeExports.jsx(
          "img",
          {
            src: imgUrl,
            alt: item.judul,
            className: "w-full max-h-[70vh] object-contain bg-muted"
          }
        ) : /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "w-full h-64 flex items-center justify-center bg-muted", children: /* @__PURE__ */ jsxRuntimeExports.jsx(Image, { className: "w-16 h-16 text-muted-foreground/30" }) }),
        /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "p-5", children: [
          /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "flex items-start gap-3 mb-2", children: /* @__PURE__ */ jsxRuntimeExports.jsx(Badge, { className: "bg-primary text-primary-foreground shrink-0", children: item.album }) }),
          /* @__PURE__ */ jsxRuntimeExports.jsx("h2", { className: "font-display font-bold text-lg text-foreground leading-snug mb-1.5", children: item.judul }),
          item.deskripsi && /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-muted-foreground text-sm leading-relaxed", children: item.deskripsi })
        ] })
      ] })
    }
  ) });
}
function Galeri() {
  const { data: galeriItems, isLoading, isError, refetch } = useGaleri();
  const [activeAlbum, setActiveAlbum] = reactExports.useState("Semua");
  const [lightboxItem, setLightboxItem] = reactExports.useState(null);
  const albums = galeriItems ? [
    "Semua",
    ...Array.from(new Set(galeriItems.map((i) => i.album).filter(Boolean)))
  ] : ["Semua"];
  const filtered = activeAlbum === "Semua" ? galeriItems ?? [] : (galeriItems ?? []).filter((i) => i.album === activeAlbum);
  return /* @__PURE__ */ jsxRuntimeExports.jsxs(Layout, { children: [
    /* @__PURE__ */ jsxRuntimeExports.jsx(
      PageHeader,
      {
        title: "Galeri",
        description: "Dokumentasi foto kegiatan dan lingkungan kampus UNTAG Banyuwangi",
        breadcrumbs: [{ label: "Galeri" }]
      }
    ),
    /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "container mx-auto px-4 py-8", "data-ocid": "galeri.page", children: [
      !isLoading && !isError && albums.length > 1 && /* @__PURE__ */ jsxRuntimeExports.jsx(
        "div",
        {
          className: "flex flex-wrap gap-2 mb-6",
          "data-ocid": "galeri.album_filter",
          children: albums.map((album) => /* @__PURE__ */ jsxRuntimeExports.jsx(
            Button,
            {
              variant: activeAlbum === album ? "default" : "outline",
              size: "sm",
              onClick: () => setActiveAlbum(album),
              className: activeAlbum === album ? "bg-primary text-primary-foreground hover:bg-primary/90" : "border-border text-foreground hover:bg-muted",
              "data-ocid": `galeri.album_tab.${album.toLowerCase().replace(/\s+/g, "_")}`,
              children: album
            },
            album
          ))
        }
      ),
      isLoading ? /* @__PURE__ */ jsxRuntimeExports.jsx(GaleriSkeleton, {}) : isError ? /* @__PURE__ */ jsxRuntimeExports.jsxs(
        "div",
        {
          className: "flex flex-col items-center justify-center py-20 text-center",
          "data-ocid": "galeri.error_state",
          children: [
            /* @__PURE__ */ jsxRuntimeExports.jsx(Image, { className: "w-14 h-14 text-muted-foreground/30 mb-4" }),
            /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-foreground font-semibold mb-2", children: "Gagal memuat galeri" }),
            /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-muted-foreground text-sm mb-5", children: "Terjadi kesalahan saat mengambil data. Silakan coba lagi." }),
            /* @__PURE__ */ jsxRuntimeExports.jsx(
              Button,
              {
                variant: "outline",
                size: "sm",
                onClick: () => refetch(),
                "data-ocid": "galeri.retry_button",
                children: "Coba Lagi"
              }
            )
          ]
        }
      ) : filtered.length === 0 ? /* @__PURE__ */ jsxRuntimeExports.jsxs(
        "div",
        {
          className: "flex flex-col items-center justify-center py-20 text-center",
          "data-ocid": "galeri.empty_state",
          children: [
            /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "w-20 h-20 rounded-full bg-muted flex items-center justify-center mb-5", children: /* @__PURE__ */ jsxRuntimeExports.jsx(Image, { className: "w-10 h-10 text-muted-foreground/40" }) }),
            /* @__PURE__ */ jsxRuntimeExports.jsx("h3", { className: "text-foreground font-semibold text-lg mb-2", children: "Belum ada foto tersedia" }),
            /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-muted-foreground text-sm max-w-xs", children: activeAlbum === "Semua" ? "Foto kegiatan akan ditampilkan di sini setelah diunggah oleh admin." : `Belum ada foto di album "${activeAlbum}".` })
          ]
        }
      ) : /* @__PURE__ */ jsxRuntimeExports.jsx(
        "div",
        {
          className: "grid grid-cols-2 sm:grid-cols-3 lg:grid-cols-4 gap-3",
          "data-ocid": "galeri.foto_list",
          children: filtered.map((item, idx) => /* @__PURE__ */ jsxRuntimeExports.jsx(
            GaleriCard,
            {
              item,
              index: idx + 1,
              onOpen: setLightboxItem
            },
            item.id
          ))
        }
      )
    ] }),
    /* @__PURE__ */ jsxRuntimeExports.jsx(Lightbox, { item: lightboxItem, onClose: () => setLightboxItem(null) })
  ] });
}
export {
  Galeri
};
