// @caffeine/template-frontend/dist/assets/Katalog-DRFl32B7.js
import { r as reactExports, j as jsxRuntimeExports, L as Layout, a as Button, C as ChevronDown, e as SkeletonList, F as FileText, b as Link } from "./index-B7-RSq-m.js";
import { I as Input } from "./input-B3nX7k9Z.js";
import { L as Label } from "./label-eF2rHvnY.js";
import { C as ChevronUp, S as Select, a as SelectTrigger, b as SelectValue, c as SelectContent, d as SelectItem } from "./select-37c5ocl_.js";
import { S as Separator } from "./separator-CrFHvepy.js";
import { E as EmptyState } from "./EmptyState-Bzc48I5U.js";
import { E as ErrorState } from "./ErrorState-o51Yfgm5.js";
import { J as JENIS_LABEL, T as TAHUN_OPTIONS, K as KategoriTag, S as StatusBadge } from "./StatusBadge-CanDaSra.js";
import { P as PageHeader } from "./PageHeader-CmrF1ENQ.js";
import { a as useDokumen, d as useKategori } from "./useBackend-xFEMRG87.js";
import { S as Search } from "./search-DeNGI9Ut.js";
import { R as RotateCcw } from "./rotate-ccw-DC-KaPvS.js";
import "./index-BcOIygc1.js";
import "./index-CUotUQbC.js";
import "./circle-alert-D-S0YTKb.js";
import "./badge-KXcDp_20.js";
import "./BreadcrumbNav---FAxtpV.js";
import "./chevron-right-RI9IPXg2.js";
import "./useMutation-D--BDafu.js";
const ITEM_KEYS = [
  "1",
  "2",
  "3",
  "4",
  "5",
  "6",
  "7",
  "8",
  "9",
  "10",
  "11",
  "12",
  "13",
  "14",
  "15",
  "16",
  "17",
  "18",
  "19",
  "20",
  "21",
  "22",
  "23",
  "24",
  "25",
  "26",
  "27",
  "28",
  "29",
  "30"
];
function FilterPanel({
  filter,
  onFilterChange,
  onReset,
  kategoriList
}) {
  var _a;
  return /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "space-y-5", children: [
    /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "space-y-1.5", children: [
      /* @__PURE__ */ jsxRuntimeExports.jsx(Label, { className: "text-xs font-semibold text-muted-foreground uppercase tracking-wide", children: "Jenis Dokumen" }),
      /* @__PURE__ */ jsxRuntimeExports.jsxs(
        Select,
        {
          value: filter.jenis || "semua",
          onValueChange: (v) => onFilterChange({ jenis: v === "semua" ? "" : v }),
          children: [
            /* @__PURE__ */ jsxRuntimeExports.jsx(
              SelectTrigger,
              {
                className: "w-full bg-background",
                "data-ocid": "katalog.filter_jenis.select",
                children: /* @__PURE__ */ jsxRuntimeExports.jsx(SelectValue, { placeholder: "Semua Jenis" })
              }
            ),
            /* @__PURE__ */ jsxRuntimeExports.jsxs(SelectContent, { children: [
              /* @__PURE__ */ jsxRuntimeExports.jsx(SelectItem, { value: "semua", children: "Semua Jenis" }),
              kategoriList.map((k) => /* @__PURE__ */ jsxRuntimeExports.jsx(SelectItem, { value: String(k.id), children: k.nama }, String(k.id)))
            ] })
          ]
        }
      )
    ] }),
    /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "space-y-1.5", children: [
      /* @__PURE__ */ jsxRuntimeExports.jsx(Label, { className: "text-xs font-semibold text-muted-foreground uppercase tracking-wide", children: "Tahun" }),
      /* @__PURE__ */ jsxRuntimeExports.jsxs(
        Select,
        {
          value: ((_a = filter.tahun) == null ? void 0 : _a.toString()) || "semua",
          onValueChange: (v) => onFilterChange({ tahun: v === "semua" ? "" : Number(v) }),
          children: [
            /* @__PURE__ */ jsxRuntimeExports.jsx(
              SelectTrigger,
              {
                className: "w-full bg-background",
                "data-ocid": "katalog.filter_tahun.select",
                children: /* @__PURE__ */ jsxRuntimeExports.jsx(SelectValue, { placeholder: "Semua Tahun" })
              }
            ),
            /* @__PURE__ */ jsxRuntimeExports.jsxs(SelectContent, { children: [
              /* @__PURE__ */ jsxRuntimeExports.jsx(SelectItem, { value: "semua", children: "Semua Tahun" }),
              TAHUN_OPTIONS.map((y) => /* @__PURE__ */ jsxRuntimeExports.jsx(SelectItem, { value: y.toString(), children: y }, y))
            ] })
          ]
        }
      )
    ] }),
    /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "space-y-1.5", children: [
      /* @__PURE__ */ jsxRuntimeExports.jsx(Label, { className: "text-xs font-semibold text-muted-foreground uppercase tracking-wide", children: "Status" }),
      /* @__PURE__ */ jsxRuntimeExports.jsxs(
        Select,
        {
          value: filter.status || "semua",
          onValueChange: (v) => onFilterChange({
            status: v === "semua" ? "" : v
          }),
          children: [
            /* @__PURE__ */ jsxRuntimeExports.jsx(
              SelectTrigger,
              {
                className: "w-full bg-background",
                "data-ocid": "katalog.filter_status.select",
                children: /* @__PURE__ */ jsxRuntimeExports.jsx(SelectValue, { placeholder: "Semua Status" })
              }
            ),
            /* @__PURE__ */ jsxRuntimeExports.jsxs(SelectContent, { children: [
              /* @__PURE__ */ jsxRuntimeExports.jsx(SelectItem, { value: "semua", children: "Semua Status" }),
              /* @__PURE__ */ jsxRuntimeExports.jsx(SelectItem, { value: "Berlaku", children: "Berlaku" }),
              /* @__PURE__ */ jsxRuntimeExports.jsx(SelectItem, { value: "TidakBerlaku", children: "Tidak Berlaku" }),
              /* @__PURE__ */ jsxRuntimeExports.jsx(SelectItem, { value: "Dicabut", children: "Dicabut" })
            ] })
          ]
        }
      )
    ] }),
    /* @__PURE__ */ jsxRuntimeExports.jsx(Separator, {}),
    /* @__PURE__ */ jsxRuntimeExports.jsxs(
      Button,
      {
        variant: "outline",
        size: "sm",
        className: "w-full gap-2 text-muted-foreground hover:text-foreground",
        onClick: onReset,
        "data-ocid": "katalog.reset_filter.button",
        children: [
          /* @__PURE__ */ jsxRuntimeExports.jsx(RotateCcw, { className: "w-3.5 h-3.5" }),
          "Reset Filter"
        ]
      }
    )
  ] });
}
function DokumenRow({
  id,
  judul,
  nomor,
  kategoriNama,
  tahun,
  status,
  keyIndex
}) {
  return /* @__PURE__ */ jsxRuntimeExports.jsxs(
    "tr",
    {
      className: "border-b border-border last:border-0 hover:bg-muted/30 transition-colors",
      "data-ocid": `katalog.item.${keyIndex}`,
      children: [
        /* @__PURE__ */ jsxRuntimeExports.jsx("td", { className: "px-4 py-3.5", children: /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "min-w-0", children: [
          /* @__PURE__ */ jsxRuntimeExports.jsx(
            Link,
            {
              to: "/katalog/$id",
              params: { id: String(id) },
              className: "font-semibold text-sm text-foreground hover:text-accent transition-colors line-clamp-2 leading-snug",
              "data-ocid": `katalog.judul_link.${keyIndex}`,
              children: judul
            }
          ),
          /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "font-doc-number text-muted-foreground text-xs mt-0.5 truncate", children: nomor })
        ] }) }),
        /* @__PURE__ */ jsxRuntimeExports.jsx("td", { className: "px-4 py-3.5 hidden md:table-cell", children: /* @__PURE__ */ jsxRuntimeExports.jsx(KategoriTag, { nama: kategoriNama ?? "-" }) }),
        /* @__PURE__ */ jsxRuntimeExports.jsx("td", { className: "px-4 py-3.5 hidden sm:table-cell text-sm text-center text-muted-foreground font-medium", children: tahun }),
        /* @__PURE__ */ jsxRuntimeExports.jsx("td", { className: "px-4 py-3.5", children: /* @__PURE__ */ jsxRuntimeExports.jsx(StatusBadge, { status }) }),
        /* @__PURE__ */ jsxRuntimeExports.jsx("td", { className: "px-4 py-3.5 text-right", children: /* @__PURE__ */ jsxRuntimeExports.jsx(
          Button,
          {
            asChild: true,
            size: "sm",
            className: "bg-accent hover:bg-accent/90 text-accent-foreground text-xs h-8 px-3",
            "data-ocid": `katalog.detail_button.${keyIndex}`,
            children: /* @__PURE__ */ jsxRuntimeExports.jsx(Link, { to: "/katalog/$id", params: { id: String(id) }, children: "Detail" })
          }
        ) })
      ]
    }
  );
}
function DokumenCard({
  id,
  judul,
  nomor,
  kategoriNama,
  tahun,
  status,
  keyIndex
}) {
  return /* @__PURE__ */ jsxRuntimeExports.jsxs(
    "div",
    {
      className: "bg-card border border-border rounded-lg p-4 shadow-card",
      "data-ocid": `katalog.item.${keyIndex}`,
      children: [
        /* @__PURE__ */ jsxRuntimeExports.jsx(
          Link,
          {
            to: "/katalog/$id",
            params: { id: String(id) },
            className: "block font-semibold text-sm text-foreground hover:text-accent transition-colors leading-snug mb-1",
            "data-ocid": `katalog.judul_link.${keyIndex}`,
            children: judul
          }
        ),
        /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "font-doc-number text-muted-foreground text-xs mb-3 truncate", children: nomor }),
        /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex flex-wrap items-center gap-2 mb-3", children: [
          /* @__PURE__ */ jsxRuntimeExports.jsx(KategoriTag, { nama: kategoriNama ?? "-" }),
          /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "text-xs text-muted-foreground", children: tahun }),
          /* @__PURE__ */ jsxRuntimeExports.jsx(StatusBadge, { status })
        ] }),
        /* @__PURE__ */ jsxRuntimeExports.jsx(
          Button,
          {
            asChild: true,
            size: "sm",
            className: "w-full bg-accent hover:bg-accent/90 text-accent-foreground text-xs h-8",
            "data-ocid": `katalog.detail_button.${keyIndex}`,
            children: /* @__PURE__ */ jsxRuntimeExports.jsx(Link, { to: "/katalog/$id", params: { id: String(id) }, children: "Lihat Detail" })
          }
        )
      ]
    }
  );
}
const JENIS_FILTER_ITEMS = Object.entries(JENIS_LABEL).map(([key, label]) => ({
  id: key,
  nama: label
}));
const DEFAULT_FILTER = {};
function Katalog() {
  const [filter, setFilter] = reactExports.useState(DEFAULT_FILTER);
  const [inputQuery, setInputQuery] = reactExports.useState("");
  const [mobileFilterOpen, setMobileFilterOpen] = reactExports.useState(false);
  const { data: dokumenList, isLoading, isError, refetch } = useDokumen(filter);
  const { data: kategoriList } = useKategori();
  const filterKategoriItems = kategoriList && kategoriList.length > 0 ? kategoriList.map((k) => ({ id: k.id, nama: k.nama })) : JENIS_FILTER_ITEMS;
  const handleSearch = () => {
    setFilter((prev) => ({ ...prev, query: inputQuery }));
  };
  const handleKeyDown = (e) => {
    if (e.key === "Enter") handleSearch();
  };
  const handleFilterChange = (partial) => {
    setFilter((prev) => ({ ...prev, ...partial }));
  };
  const handleReset = () => {
    setFilter(DEFAULT_FILTER);
    setInputQuery("");
  };
  const results = dokumenList ?? [];
  const totalHasil = results.length;
  const hasActiveFilter = !!(filter.query || filter.jenis || filter.tahun || filter.status);
  return /* @__PURE__ */ jsxRuntimeExports.jsxs(Layout, { children: [
    /* @__PURE__ */ jsxRuntimeExports.jsx(
      PageHeader,
      {
        title: "Katalog Produk Hukum",
        description: "Telusuri dan unduh dokumen peraturan, keputusan, surat edaran, dan produk hukum lainnya dari UNTAG Banyuwangi.",
        breadcrumbs: [{ label: "Katalog Produk Hukum" }]
      }
    ),
    /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "container mx-auto px-4 py-6", children: [
      /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "bg-card border border-border rounded-lg p-4 mb-6 shadow-subtle", children: /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex gap-2", children: [
        /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "relative flex-1", children: [
          /* @__PURE__ */ jsxRuntimeExports.jsx(Search, { className: "absolute left-3 top-1/2 -translate-y-1/2 w-4 h-4 text-muted-foreground pointer-events-none" }),
          /* @__PURE__ */ jsxRuntimeExports.jsx(
            Input,
            {
              value: inputQuery,
              onChange: (e) => setInputQuery(e.target.value),
              onKeyDown: handleKeyDown,
              placeholder: "Cari berdasarkan judul, nomor, atau kata kunci...",
              className: "pl-9 bg-background",
              "data-ocid": "katalog.search_input"
            }
          )
        ] }),
        /* @__PURE__ */ jsxRuntimeExports.jsxs(
          Button,
          {
            onClick: handleSearch,
            className: "bg-accent hover:bg-accent/90 text-accent-foreground gap-2 px-5 flex-shrink-0",
            "data-ocid": "katalog.search_button",
            children: [
              /* @__PURE__ */ jsxRuntimeExports.jsx(Search, { className: "w-4 h-4" }),
              /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "hidden sm:inline", children: "Cari" })
            ]
          }
        )
      ] }) }),
      /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex gap-6", children: [
        /* @__PURE__ */ jsxRuntimeExports.jsx("aside", { className: "hidden lg:block w-64 flex-shrink-0", children: /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "bg-card border border-border rounded-lg p-5 shadow-subtle sticky top-4", children: [
          /* @__PURE__ */ jsxRuntimeExports.jsx("h2", { className: "font-display font-semibold text-sm text-foreground mb-4 uppercase tracking-wide", children: "Filter Dokumen" }),
          /* @__PURE__ */ jsxRuntimeExports.jsx(
            FilterPanel,
            {
              filter,
              onFilterChange: handleFilterChange,
              onReset: handleReset,
              kategoriList: filterKategoriItems
            }
          )
        ] }) }),
        /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex-1 min-w-0", children: [
          /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "lg:hidden mb-4", children: [
            /* @__PURE__ */ jsxRuntimeExports.jsxs(
              Button,
              {
                variant: "outline",
                size: "sm",
                className: "gap-2 w-full justify-between",
                onClick: () => setMobileFilterOpen((v) => !v),
                "data-ocid": "katalog.mobile_filter.toggle",
                children: [
                  /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "font-medium", children: "Filter Dokumen" }),
                  mobileFilterOpen ? /* @__PURE__ */ jsxRuntimeExports.jsx(ChevronUp, { className: "w-4 h-4" }) : /* @__PURE__ */ jsxRuntimeExports.jsx(ChevronDown, { className: "w-4 h-4" })
                ]
              }
            ),
            mobileFilterOpen && /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "mt-2 bg-card border border-border rounded-lg p-4 shadow-subtle", children: /* @__PURE__ */ jsxRuntimeExports.jsx(
              FilterPanel,
              {
                filter,
                onFilterChange: handleFilterChange,
                onReset: handleReset,
                kategoriList: filterKategoriItems
              }
            ) })
          ] }),
          !isLoading && !isError && /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex items-center justify-between mb-4", children: [
            /* @__PURE__ */ jsxRuntimeExports.jsxs(
              "p",
              {
                className: "text-sm text-muted-foreground",
                "data-ocid": "katalog.result_count",
                children: [
                  "Ditemukan",
                  " ",
                  /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "font-semibold text-foreground", children: totalHasil }),
                  " ",
                  "dokumen"
                ]
              }
            ),
            hasActiveFilter && /* @__PURE__ */ jsxRuntimeExports.jsxs(
              "button",
              {
                type: "button",
                onClick: handleReset,
                className: "text-xs text-accent hover:underline flex items-center gap-1",
                "data-ocid": "katalog.reset_filter_inline.button",
                children: [
                  /* @__PURE__ */ jsxRuntimeExports.jsx(RotateCcw, { className: "w-3 h-3" }),
                  "Reset Filter"
                ]
              }
            )
          ] }),
          isLoading && /* @__PURE__ */ jsxRuntimeExports.jsx(SkeletonList, { count: 6 }),
          isError && !isLoading && /* @__PURE__ */ jsxRuntimeExports.jsx(ErrorState, { onRetry: () => refetch() }),
          !isLoading && !isError && results.length === 0 && /* @__PURE__ */ jsxRuntimeExports.jsx(
            "div",
            {
              className: "bg-card border border-border rounded-lg",
              "data-ocid": "katalog.empty_state",
              children: /* @__PURE__ */ jsxRuntimeExports.jsx(
                EmptyState,
                {
                  icon: /* @__PURE__ */ jsxRuntimeExports.jsx(FileText, { className: "w-8 h-8 text-muted-foreground" }),
                  title: "Dokumen tidak ditemukan",
                  description: "Coba ubah kata kunci pencarian atau reset filter untuk melihat semua dokumen.",
                  actionLabel: "Reset Filter",
                  onAction: handleReset
                }
              )
            }
          ),
          !isLoading && !isError && results.length > 0 && /* @__PURE__ */ jsxRuntimeExports.jsxs(jsxRuntimeExports.Fragment, { children: [
            /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "hidden sm:block bg-card border border-border rounded-lg overflow-hidden shadow-subtle", children: /* @__PURE__ */ jsxRuntimeExports.jsxs("table", { className: "w-full text-sm", children: [
              /* @__PURE__ */ jsxRuntimeExports.jsx("thead", { children: /* @__PURE__ */ jsxRuntimeExports.jsxs("tr", { className: "bg-secondary border-b border-border", children: [
                /* @__PURE__ */ jsxRuntimeExports.jsx("th", { className: "px-4 py-3 text-left font-semibold text-secondary-foreground text-xs uppercase tracking-wide", children: "Judul / Nomor" }),
                /* @__PURE__ */ jsxRuntimeExports.jsx("th", { className: "px-4 py-3 text-left font-semibold text-secondary-foreground text-xs uppercase tracking-wide hidden md:table-cell", children: "Jenis" }),
                /* @__PURE__ */ jsxRuntimeExports.jsx("th", { className: "px-4 py-3 text-center font-semibold text-secondary-foreground text-xs uppercase tracking-wide hidden sm:table-cell w-20", children: "Tahun" }),
                /* @__PURE__ */ jsxRuntimeExports.jsx("th", { className: "px-4 py-3 text-left font-semibold text-secondary-foreground text-xs uppercase tracking-wide w-32", children: "Status" }),
                /* @__PURE__ */ jsxRuntimeExports.jsx("th", { className: "px-4 py-3 text-right font-semibold text-secondary-foreground text-xs uppercase tracking-wide w-24", children: "Aksi" })
              ] }) }),
              /* @__PURE__ */ jsxRuntimeExports.jsx("tbody", { children: results.map((doc, idx) => /* @__PURE__ */ jsxRuntimeExports.jsx(
                DokumenRow,
                {
                  id: String(doc.id),
                  judul: doc.judul,
                  nomor: doc.nomor,
                  kategoriNama: doc.kategoriNama ?? "-",
                  tahun: doc.tahun,
                  status: doc.status,
                  keyIndex: ITEM_KEYS[idx] ?? String(idx + 1)
                },
                doc.id
              )) })
            ] }) }),
            /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "sm:hidden space-y-3", children: results.map((doc, idx) => /* @__PURE__ */ jsxRuntimeExports.jsx(
              DokumenCard,
              {
                id: String(doc.id),
                judul: doc.judul,
                nomor: doc.nomor,
                kategoriNama: doc.kategoriNama ?? "-",
                tahun: doc.tahun,
                status: doc.status,
                keyIndex: ITEM_KEYS[idx] ?? String(idx + 1)
              },
              doc.id
            )) })
          ] })
        ] })
      ] })
    ] })
  ] });
}
export {
  Katalog
};
