// @caffeine/template-frontend/dist/assets/Kontak-MdAiaWrL.js
import { j as jsxRuntimeExports, L as Layout, M as MapPin, P as Phone, n as Mail } from "./index-B7-RSq-m.js";
import { P as PageHeader } from "./PageHeader-CmrF1ENQ.js";
import { C as Clock } from "./clock-CMHdisaV.js";
import "./BreadcrumbNav---FAxtpV.js";
import "./chevron-right-RI9IPXg2.js";
const CONTACT_INFO = [
  {
    icon: MapPin,
    label: "Alamat",
    value: "Jl. Adi Sucipto No. 26, Banyuwangi, Jawa Timur 68416",
    description: "Kampus Universitas 17 Agustus 1945 Banyuwangi"
  },
  {
    icon: Phone,
    label: "Telepon",
    value: "+62 333 415032",
    description: "Jam kerja Senin–Jumat"
  },
  {
    icon: Mail,
    label: "Email",
    value: "jdih@untag-banyuwangi.ac.id",
    description: "Direspons dalam 1–2 hari kerja"
  },
  {
    icon: Clock,
    label: "Jam Operasional",
    value: "Senin – Jumat, 08.00 – 16.00 WIB",
    description: "Sabtu & Minggu libur"
  }
];
function Kontak() {
  return /* @__PURE__ */ jsxRuntimeExports.jsxs(Layout, { children: [
    /* @__PURE__ */ jsxRuntimeExports.jsx(
      PageHeader,
      {
        title: "Kontak Kami",
        description: "Hubungi JDIH UNTAG Banyuwangi untuk pertanyaan atau bantuan seputar dokumen hukum kampus",
        breadcrumbs: [{ label: "Kontak" }]
      }
    ),
    /* @__PURE__ */ jsxRuntimeExports.jsx("section", { className: "py-12 bg-background", "data-ocid": "kontak.page", children: /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "container mx-auto px-4", children: /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "max-w-4xl mx-auto", children: [
      /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "text-center mb-10", children: [
        /* @__PURE__ */ jsxRuntimeExports.jsx("h2", { className: "font-display font-bold text-lg text-foreground mb-2", children: "Informasi Kontak" }),
        /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-muted-foreground text-sm max-w-xl mx-auto", children: "Kami siap membantu Anda menemukan informasi produk hukum kampus. Silakan hubungi kami melalui saluran berikut." })
      ] }),
      /* @__PURE__ */ jsxRuntimeExports.jsx(
        "div",
        {
          className: "grid grid-cols-1 sm:grid-cols-2 gap-5 mb-10",
          "data-ocid": "kontak.info_section",
          children: CONTACT_INFO.map((item) => /* @__PURE__ */ jsxRuntimeExports.jsxs(
            "div",
            {
              className: "bg-card border border-border rounded-xl p-6 flex items-start gap-4 shadow-card hover:border-primary/30 transition-smooth",
              children: [
                /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "w-11 h-11 bg-primary/10 rounded-xl flex items-center justify-center flex-shrink-0", children: /* @__PURE__ */ jsxRuntimeExports.jsx(item.icon, { className: "w-5 h-5 text-primary" }) }),
                /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "min-w-0", children: [
                  /* @__PURE__ */ jsxRuntimeExports.jsx("dt", { className: "text-xs font-semibold uppercase tracking-widest text-muted-foreground mb-1", children: item.label }),
                  /* @__PURE__ */ jsxRuntimeExports.jsx("dd", { className: "text-sm font-semibold text-foreground break-words", children: item.value }),
                  /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-xs text-muted-foreground mt-0.5", children: item.description })
                ] })
              ]
            },
            item.label
          ))
        }
      ),
      /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "bg-primary rounded-2xl p-6 sm:p-8 flex flex-col sm:flex-row gap-6 items-start sm:items-center shadow-card", children: [
        /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex-1", children: [
          /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-xs font-bold uppercase tracking-widest text-primary-foreground/60 mb-2", children: "Unit Pengelola" }),
          /* @__PURE__ */ jsxRuntimeExports.jsx("h3", { className: "font-display font-bold text-lg text-primary-foreground mb-1", children: "JDIH UNTAG Banyuwangi" }),
          /* @__PURE__ */ jsxRuntimeExports.jsxs("p", { className: "text-sm text-primary-foreground/80 leading-relaxed", children: [
            "Biro Hukum & Kerjasama",
            /* @__PURE__ */ jsxRuntimeExports.jsx("br", {}),
            "Universitas 17 Agustus 1945 Banyuwangi",
            /* @__PURE__ */ jsxRuntimeExports.jsx("br", {}),
            "Jl. Adi Sucipto No. 26, Banyuwangi, Jawa Timur"
          ] })
        ] }),
        /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "flex-shrink-0 hidden sm:block", children: /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "w-16 h-16 rounded-2xl bg-primary-foreground/10 border border-primary-foreground/20 flex items-center justify-center", children: /* @__PURE__ */ jsxRuntimeExports.jsx(MapPin, { className: "w-8 h-8 text-primary-foreground/70" }) }) })
      ] }),
      /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "mt-8 bg-secondary/40 border border-border rounded-xl p-5 flex items-start gap-3", children: [
        /* @__PURE__ */ jsxRuntimeExports.jsx(Mail, { className: "w-5 h-5 text-primary flex-shrink-0 mt-0.5" }),
        /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { children: [
          /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-sm font-semibold text-foreground mb-0.5", children: "Pengajuan Permohonan Informasi" }),
          /* @__PURE__ */ jsxRuntimeExports.jsxs("p", { className: "text-xs text-muted-foreground leading-relaxed", children: [
            "Untuk permohonan salinan dokumen atau informasi hukum secara formal, silakan kirimkan surat permohonan melalui email",
            " ",
            /* @__PURE__ */ jsxRuntimeExports.jsx("strong", { className: "text-foreground", children: "jdih@untag-banyuwangi.ac.id" }),
            " ",
            "atau datang langsung ke kantor Biro Hukum & Kerjasama UNTAG Banyuwangi pada jam operasional."
          ] })
        ] })
      ] })
    ] }) }) })
  ] });
}
export {
  Kontak
};
