// @caffeine/template-frontend/dist/assets/StatusBadge-CanDaSra.js
import { j as jsxRuntimeExports } from "./index-B7-RSq-m.js";
import { B as Badge } from "./badge-KXcDp_20.js";
function KategoriTag({
  nama,
  className = "",
  onClick
}) {
  return /* @__PURE__ */ jsxRuntimeExports.jsx(
    Badge,
    {
      variant: "secondary",
      onClick,
      className: `text-xs font-medium bg-secondary text-secondary-foreground border border-border cursor-${onClick ? "pointer" : "default"} hover:bg-secondary/80 transition-smooth ${className}`,
      children: nama
    }
  );
}
const JENIS_LABEL = {
  Statuta: "Statuta",
  PeraturanRektor: "Peraturan Rektor",
  KeputusanRektor: "Keputusan Rektor",
  SuratEdaran: "Surat Edaran",
  PeraturanSenat: "Peraturan Senat",
  SOP: "SOP",
  MoU: "Perjanjian / MoU"
};
const STATUS_LABEL = {
  Berlaku: "Berlaku",
  TidakBerlaku: "Tidak Berlaku",
  Dicabut: "Dicabut"
};
const WORKFLOW_STATUS_LABEL = {
  Draft: "Draf",
  PendingReview: "Menunggu Review",
  Published: "Terbit"
};
const TAHUN_OPTIONS = Array.from({ length: 10 }, (_, i) => 2024 - i);
const STATUS_STYLES = {
  Berlaku: "bg-foreground/5 text-foreground border border-border hover:bg-foreground/5",
  TidakBerlaku: "bg-muted text-muted-foreground border border-border hover:bg-muted",
  Dicabut: "bg-primary/10 text-primary border border-primary/30 hover:bg-primary/10"
};
function StatusBadge({ status, className = "" }) {
  return /* @__PURE__ */ jsxRuntimeExports.jsx(
    Badge,
    {
      variant: "outline",
      className: `${STATUS_STYLES[status]} text-xs font-medium px-2 py-0.5 ${className}`,
      children: STATUS_LABEL[status]
    }
  );
}
const WORKFLOW_STYLES = {
  Draft: "bg-muted text-muted-foreground border border-border hover:bg-muted",
  PendingReview: "bg-yellow-50 text-yellow-800 border border-yellow-300 hover:bg-yellow-50",
  Published: "bg-green-50 text-green-800 border border-green-300 hover:bg-green-50"
};
function WorkflowStatusBadge({
  status,
  className = ""
}) {
  return /* @__PURE__ */ jsxRuntimeExports.jsx(
    Badge,
    {
      variant: "outline",
      className: `${WORKFLOW_STYLES[status]} text-xs font-medium px-2 py-0.5 ${className}`,
      children: WORKFLOW_STATUS_LABEL[status]
    }
  );
}
export {
  JENIS_LABEL as J,
  KategoriTag as K,
  StatusBadge as S,
  TAHUN_OPTIONS as T,
  WORKFLOW_STATUS_LABEL as W,
  STATUS_LABEL as a,
  WorkflowStatusBadge as b
};
