// @caffeine/template-frontend/dist/assets/TentangVisiMisi-BkkfRQEx.js
import { j as jsxRuntimeExports, L as Layout, B as BookOpen } from "./index-B7-RSq-m.js";
import { P as PageHeader } from "./PageHeader-CmrF1ENQ.js";
import { S as Skeleton } from "./skeleton-Z-jxXZYe.js";
import { g as useTentangPage } from "./useBackend-xFEMRG87.js";
import { C as CircleCheck } from "./circle-check-H4K-d60y.js";
import "./BreadcrumbNav---FAxtpV.js";
import "./chevron-right-RI9IPXg2.js";
import "./useMutation-D--BDafu.js";
const DEFAULT_VISI = "Menjadi pusat dokumentasi dan informasi hukum kampus yang terpercaya, mudah diakses, dan bermanfaat bagi civitas akademika Universitas 17 Agustus 1945 Banyuwangi.";
const DEFAULT_MISI = [
  "Membangun sistem dokumentasi hukum kampus yang terstruktur, lengkap, dan mudah diakses",
  "Melakukan pengelolaan dan pemutakhiran data produk hukum secara berkala dan konsisten",
  "Menyebarluaskan informasi hukum kepada seluruh civitas akademika UNTAG Banyuwangi",
  "Mendukung penegakan hukum dan tata kelola universitas yang transparan dan akuntabel",
  "Menjalin koordinasi dan sinkronisasi dengan JDIHN pusat dan jaringan JDIH perguruan tinggi"
];
function TentangVisiMisi() {
  var _a, _b, _c, _d, _e, _f;
  const { data: page, isLoading } = useTentangPage("visiMisi");
  const visi = ((_c = (_b = (_a = page == null ? void 0 : page.konten) == null ? void 0 : _a.blocks) == null ? void 0 : _b.find((b) => b.__kind__ === "paragraf")) == null ? void 0 : _c.paragraf) ?? DEFAULT_VISI;
  const misi = ((_f = (_e = (_d = page == null ? void 0 : page.konten) == null ? void 0 : _d.blocks) == null ? void 0 : _e.find((b) => b.__kind__ === "daftarItem")) == null ? void 0 : _f.daftarItem) ?? DEFAULT_MISI;
  return /* @__PURE__ */ jsxRuntimeExports.jsxs(Layout, { children: [
    /* @__PURE__ */ jsxRuntimeExports.jsx(
      PageHeader,
      {
        title: "Visi & Misi",
        description: "Visi dan misi JDIH Universitas 17 Agustus 1945 Banyuwangi dalam pengelolaan dokumentasi hukum",
        breadcrumbs: [
          { label: "Tentang JDIH", to: "/tentang/sejarah" },
          { label: "Visi & Misi" }
        ]
      }
    ),
    /* @__PURE__ */ jsxRuntimeExports.jsx(
      "section",
      {
        className: "py-12 bg-muted/30",
        "data-ocid": "tentang_visi_misi.section",
        children: /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "container mx-auto px-4", children: /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "max-w-3xl mx-auto", children: [
          /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex items-center gap-3 mb-6", children: [
            /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "w-10 h-10 bg-primary/10 rounded-lg flex items-center justify-center flex-shrink-0", children: /* @__PURE__ */ jsxRuntimeExports.jsx(BookOpen, { className: "w-5 h-5 text-primary" }) }),
            /* @__PURE__ */ jsxRuntimeExports.jsx("h2", { className: "font-display font-bold text-xl text-foreground", children: "Visi & Misi" })
          ] }),
          /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "bg-primary rounded-xl p-6 mb-6 shadow-card", children: [
            /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "flex items-center gap-2 mb-3", children: /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "text-xs font-bold uppercase tracking-widest text-primary-foreground/70", children: "Visi" }) }),
            isLoading ? /* @__PURE__ */ jsxRuntimeExports.jsx(Skeleton, { className: "h-12 w-full bg-primary-foreground/20" }) : /* @__PURE__ */ jsxRuntimeExports.jsxs("p", { className: "text-primary-foreground font-display font-semibold text-base sm:text-lg leading-relaxed", children: [
              '"',
              visi,
              '"'
            ] })
          ] }),
          /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "bg-card border border-border rounded-xl p-6 shadow-card", children: [
            /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-xs font-bold uppercase tracking-widest text-muted-foreground mb-4", children: "Misi" }),
            isLoading ? /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "space-y-3", children: [
              /* @__PURE__ */ jsxRuntimeExports.jsx(Skeleton, { className: "h-5 w-full" }),
              /* @__PURE__ */ jsxRuntimeExports.jsx(Skeleton, { className: "h-5 w-5/6" }),
              /* @__PURE__ */ jsxRuntimeExports.jsx(Skeleton, { className: "h-5 w-full" })
            ] }) : /* @__PURE__ */ jsxRuntimeExports.jsx("ul", { className: "space-y-3", children: misi.map((item, idx) => (
              // biome-ignore lint/suspicious/noArrayIndexKey: ordered list
              /* @__PURE__ */ jsxRuntimeExports.jsxs("li", { className: "flex items-start gap-3", children: [
                /* @__PURE__ */ jsxRuntimeExports.jsx(CircleCheck, { className: "w-4 h-4 text-accent flex-shrink-0 mt-0.5" }),
                /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "text-sm text-foreground leading-relaxed", children: item })
              ] }, idx)
            )) })
          ] })
        ] }) })
      }
    )
  ] });
}
export {
  TentangVisiMisi
};
