// @caffeine/template-frontend/dist/assets/TentangDasarHukum-j8XENSPe.js
import { j as jsxRuntimeExports, L as Layout, F as FileText } from "./index-B7-RSq-m.js";
import { P as PageHeader } from "./PageHeader-CmrF1ENQ.js";
import { S as Skeleton } from "./skeleton-Z-jxXZYe.js";
import { g as useTentangPage } from "./useBackend-xFEMRG87.js";
import "./BreadcrumbNav---FAxtpV.js";
import "./chevron-right-RI9IPXg2.js";
import "./useMutation-D--BDafu.js";
const DEFAULT_DASAR_HUKUM = [
  "Peraturan Presiden Nomor 33 Tahun 2012 tentang Jaringan Dokumentasi dan Informasi Hukum Nasional",
  "Peraturan Menteri Hukum dan HAM Nomor 8 Tahun 2019 tentang Standar Pengelolaan Dokumen dan Informasi Hukum",
  "Undang-Undang Nomor 12 Tahun 2011 tentang Pembentukan Peraturan Perundang-undangan",
  "Statuta Universitas 17 Agustus 1945 Banyuwangi",
  "Keputusan Rektor UNTAG Banyuwangi tentang Pembentukan JDIH Kampus"
];
function TentangDasarHukum() {
  var _a, _b, _c;
  const { data: page, isLoading } = useTentangPage("dasarHukum");
  const items = ((_c = (_b = (_a = page == null ? void 0 : page.konten) == null ? void 0 : _a.blocks) == null ? void 0 : _b.find((b) => b.__kind__ === "daftarItem")) == null ? void 0 : _c.daftarItem) ?? DEFAULT_DASAR_HUKUM;
  return /* @__PURE__ */ jsxRuntimeExports.jsxs(Layout, { children: [
    /* @__PURE__ */ jsxRuntimeExports.jsx(
      PageHeader,
      {
        title: "Dasar Hukum",
        description: "Landasan hukum pembentukan dan penyelenggaraan JDIH Universitas 17 Agustus 1945 Banyuwangi",
        breadcrumbs: [
          { label: "Tentang JDIH", to: "/tentang/sejarah" },
          { label: "Dasar Hukum" }
        ]
      }
    ),
    /* @__PURE__ */ jsxRuntimeExports.jsx(
      "section",
      {
        className: "py-12 bg-background",
        "data-ocid": "tentang_dasar_hukum.section",
        children: /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "container mx-auto px-4", children: /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "max-w-3xl mx-auto", children: [
          /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex items-center gap-3 mb-6", children: [
            /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "w-10 h-10 bg-primary/10 rounded-lg flex items-center justify-center flex-shrink-0", children: /* @__PURE__ */ jsxRuntimeExports.jsx(FileText, { className: "w-5 h-5 text-primary" }) }),
            /* @__PURE__ */ jsxRuntimeExports.jsx("h2", { className: "font-display font-bold text-xl text-foreground", children: "Dasar Hukum" })
          ] }),
          isLoading ? /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "bg-card border border-border rounded-xl p-4 space-y-3 shadow-card", children: [1, 2, 3, 4, 5].map((n) => /* @__PURE__ */ jsxRuntimeExports.jsx(Skeleton, { className: "h-10 w-full" }, n)) }) : /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "bg-card border border-border rounded-xl divide-y divide-border shadow-card", children: items.map((item, idx) => /* @__PURE__ */ jsxRuntimeExports.jsxs(
            "div",
            {
              className: "flex items-start gap-4 p-4",
              "data-ocid": `tentang_dasar_hukum.item.${idx + 1}`,
              children: [
                /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "w-7 h-7 bg-primary text-primary-foreground text-xs font-bold rounded-md flex items-center justify-center flex-shrink-0 mt-0.5", children: idx + 1 }),
                /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-sm text-foreground leading-relaxed", children: item })
              ]
            },
            idx
          )) })
        ] }) })
      }
    )
  ] });
}
export {
  TentangDasarHukum
};
