// @caffeine/template-frontend/dist/assets/TentangFungsi-BOapJkZp.js
import { j as jsxRuntimeExports, L as Layout, l as Info } from "./index-B7-RSq-m.js";
import { P as PageHeader } from "./PageHeader-CmrF1ENQ.js";
import { S as Skeleton } from "./skeleton-Z-jxXZYe.js";
import { g as useTentangPage } from "./useBackend-xFEMRG87.js";
import "./BreadcrumbNav---FAxtpV.js";
import "./chevron-right-RI9IPXg2.js";
import "./useMutation-D--BDafu.js";
const DEFAULT_FUNGSI = [
  "Mendokumentasikan seluruh produk hukum yang diterbitkan oleh Universitas 17 Agustus 1945 Banyuwangi",
  "Menyediakan layanan informasi hukum yang mudah diakses oleh civitas akademika dan masyarakat umum",
  "Menjaga ketertiban, kelengkapan, dan kemutakhiran data produk hukum kampus",
  "Mendukung integrasi dan sinkronisasi data dengan Jaringan Dokumentasi dan Informasi Hukum Nasional (JDIHN)",
  "Meningkatkan transparansi dan akuntabilitas tata kelola hukum universitas",
  "Menyediakan sarana publikasi dan sosialisasi regulasi internal kampus"
];
function TentangFungsi() {
  var _a, _b, _c;
  const { data: page, isLoading } = useTentangPage("fungsi");
  const items = ((_c = (_b = (_a = page == null ? void 0 : page.konten) == null ? void 0 : _a.blocks) == null ? void 0 : _b.find((b) => b.__kind__ === "daftarItem")) == null ? void 0 : _c.daftarItem) ?? DEFAULT_FUNGSI;
  return /* @__PURE__ */ jsxRuntimeExports.jsxs(Layout, { children: [
    /* @__PURE__ */ jsxRuntimeExports.jsx(
      PageHeader,
      {
        title: "Fungsi JDIH",
        description: "Peran dan fungsi JDIH dalam mendukung tata kelola hukum Universitas 17 Agustus 1945 Banyuwangi",
        breadcrumbs: [
          { label: "Tentang JDIH", to: "/tentang/sejarah" },
          { label: "Fungsi JDIH" }
        ]
      }
    ),
    /* @__PURE__ */ jsxRuntimeExports.jsx("section", { className: "py-12 bg-muted/30", "data-ocid": "tentang_fungsi.section", children: /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "container mx-auto px-4", children: /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "max-w-3xl mx-auto", children: [
      /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex items-center gap-3 mb-6", children: [
        /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "w-10 h-10 bg-primary/10 rounded-lg flex items-center justify-center flex-shrink-0", children: /* @__PURE__ */ jsxRuntimeExports.jsx(Info, { className: "w-5 h-5 text-primary" }) }),
        /* @__PURE__ */ jsxRuntimeExports.jsx("h2", { className: "font-display font-bold text-xl text-foreground", children: "Fungsi JDIH" })
      ] }),
      isLoading ? /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "grid grid-cols-1 sm:grid-cols-2 gap-4", children: [1, 2, 3, 4].map((n) => /* @__PURE__ */ jsxRuntimeExports.jsx(Skeleton, { className: "h-24 w-full rounded-xl" }, n)) }) : /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "grid grid-cols-1 sm:grid-cols-2 gap-4", children: items.map((fungsi, idx) => /* @__PURE__ */ jsxRuntimeExports.jsxs(
        "div",
        {
          className: "bg-card border border-border rounded-xl p-5 flex items-start gap-3 shadow-card",
          "data-ocid": `tentang_fungsi.item.${idx + 1}`,
          children: [
            /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "w-6 h-6 bg-accent/10 text-accent text-xs font-bold rounded flex items-center justify-center flex-shrink-0 mt-0.5", children: idx + 1 }),
            /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-sm text-foreground leading-relaxed", children: fungsi })
          ]
        },
        fungsi
      )) })
    ] }) }) })
  ] });
}
export {
  TentangFungsi
};
