// @caffeine/template-frontend/dist/assets/TentangSejarah-sYLzDSMJ.js
import { j as jsxRuntimeExports, L as Layout, S as Scale } from "./index-B7-RSq-m.js";
import { P as PageHeader } from "./PageHeader-CmrF1ENQ.js";
import { S as Skeleton } from "./skeleton-Z-jxXZYe.js";
import { g as useTentangPage } from "./useBackend-xFEMRG87.js";
import "./BreadcrumbNav---FAxtpV.js";
import "./chevron-right-RI9IPXg2.js";
import "./useMutation-D--BDafu.js";
const DEFAULT_PARAGRAPHS = [
  "JDIH (Jaringan Dokumentasi dan Informasi Hukum) Kampus Universitas 17 Agustus 1945 Banyuwangi dibentuk sebagai bagian dari upaya universitas dalam mendukung sistem dokumentasi hukum nasional yang diatur melalui Peraturan Presiden Nomor 33 Tahun 2012 tentang Jaringan Dokumentasi dan Informasi Hukum Nasional.",
  "Pembentukan JDIH UNTAG Banyuwangi juga dilandasi oleh Peraturan Menteri Hukum dan HAM Nomor 8 Tahun 2019 tentang Standar Pengelolaan Dokumen dan Informasi Hukum, yang mewajibkan setiap institusi, termasuk perguruan tinggi, untuk mengelola produk hukum secara tertib, terstruktur, dan mudah diakses.",
  "Portal ini hadir sebagai wujud komitmen UNTAG Banyuwangi terhadap transparansi tata kelola hukum dan kemudahan akses informasi regulasi bagi seluruh civitas akademika, serta sebagai bentuk integrasi dengan Jaringan Dokumentasi dan Informasi Hukum Nasional (JDIHN) yang dikoordinasikan oleh Kementerian Hukum dan HAM Republik Indonesia.",
  "Sejak didirikan, JDIH UNTAG Banyuwangi berkomitmen untuk terus memperbarui dan memperluas koleksi dokumen hukum agar seluruh civitas akademika, tenaga pendidik, tenaga kependidikan, dan masyarakat umum dapat mengakses regulasi kampus secara mudah, cepat, dan gratis."
];
function TentangSejarah() {
  var _a, _b;
  const { data: page, isLoading } = useTentangPage("sejarah");
  const paragraphs = ((_b = (_a = page == null ? void 0 : page.konten) == null ? void 0 : _a.blocks) == null ? void 0 : _b.filter((b) => b.__kind__ === "paragraf").map((b) => b.paragraf)) ?? DEFAULT_PARAGRAPHS;
  return /* @__PURE__ */ jsxRuntimeExports.jsxs(Layout, { children: [
    /* @__PURE__ */ jsxRuntimeExports.jsx(
      PageHeader,
      {
        title: "Sejarah & Latar Belakang",
        description: "Sejarah pembentukan dan latar belakang JDIH Universitas 17 Agustus 1945 Banyuwangi",
        breadcrumbs: [
          { label: "Tentang JDIH", to: "/tentang/sejarah" },
          { label: "Sejarah" }
        ]
      }
    ),
    /* @__PURE__ */ jsxRuntimeExports.jsx(
      "section",
      {
        className: "py-12 bg-background",
        "data-ocid": "tentang_sejarah.section",
        children: /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "container mx-auto px-4", children: /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "max-w-3xl mx-auto", children: [
          /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex items-center gap-3 mb-5", children: [
            /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "w-10 h-10 bg-primary/10 rounded-lg flex items-center justify-center flex-shrink-0", children: /* @__PURE__ */ jsxRuntimeExports.jsx(Scale, { className: "w-5 h-5 text-primary" }) }),
            /* @__PURE__ */ jsxRuntimeExports.jsx("h2", { className: "font-display font-bold text-xl text-foreground", children: "Sejarah & Latar Belakang" })
          ] }),
          /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "bg-card border border-border rounded-lg p-6 shadow-card space-y-4", children: isLoading ? /* @__PURE__ */ jsxRuntimeExports.jsxs(jsxRuntimeExports.Fragment, { children: [
            /* @__PURE__ */ jsxRuntimeExports.jsx(Skeleton, { className: "h-16 w-full" }),
            /* @__PURE__ */ jsxRuntimeExports.jsx(Skeleton, { className: "h-16 w-full" }),
            /* @__PURE__ */ jsxRuntimeExports.jsx(Skeleton, { className: "h-12 w-3/4" })
          ] }) : paragraphs.map((p, idx) => /* @__PURE__ */ jsxRuntimeExports.jsx(
            "p",
            {
              className: "text-muted-foreground text-sm leading-relaxed",
              children: p
            },
            `para-${idx}-${p.slice(0, 12)}`
          )) })
        ] }) })
      }
    )
  ] });
}
export {
  TentangSejarah
};
