// @caffeine/template-frontend/dist/assets/TentangStrukturOrganisasi-T12nitEe.js
import { j as jsxRuntimeExports, L as Layout, U as Users, m as Building2 } from "./index-B7-RSq-m.js";
import { P as PageHeader } from "./PageHeader-CmrF1ENQ.js";
import { S as Skeleton } from "./skeleton-Z-jxXZYe.js";
import { g as useTentangPage } from "./useBackend-xFEMRG87.js";
import "./BreadcrumbNav---FAxtpV.js";
import "./chevron-right-RI9IPXg2.js";
import "./useMutation-D--BDafu.js";
const DEFAULT_STRUKTUR = [
  {
    jabatan: "Penanggung Jawab",
    nama: "Prof. Dr. H. Teguh Sulistyo, S.H., M.H.",
    unit: "Rektor UNTAG Banyuwangi"
  },
  {
    jabatan: "Ketua JDIH",
    nama: "Dr. Hj. Sri Wahyuni, S.H., M.H.",
    unit: "Wakil Rektor Bidang Akademik"
  },
  {
    jabatan: "Koordinator Teknis",
    nama: "Agus Prasetyo, S.H., M.H.",
    unit: "Kepala Biro Hukum & Kerjasama"
  },
  {
    jabatan: "Pengelola Dokumen",
    nama: "Dewi Rahmawati, S.H.",
    unit: "Staf Biro Hukum"
  },
  {
    jabatan: "Anggota",
    nama: "Rian Hidayat, S.Kom.",
    unit: "UPT Teknologi Informasi"
  },
  {
    jabatan: "Anggota",
    nama: "Siti Nurhayati, A.Md.",
    unit: "Staf Biro Akademik"
  }
];
function parseStrukturItem(raw) {
  const parts = {};
  for (const seg of raw.split(" | ")) {
    const colonIdx = seg.indexOf(": ");
    if (colonIdx !== -1) {
      parts[seg.slice(0, colonIdx).trim()] = seg.slice(colonIdx + 2).trim();
    }
  }
  return {
    jabatan: parts.Jabatan ?? "",
    nama: parts.Nama ?? "",
    unit: parts.Unit ?? ""
  };
}
function TentangStrukturOrganisasi() {
  var _a, _b, _c;
  const { data: page, isLoading } = useTentangPage("struktur");
  const rawItems = (_c = (_b = (_a = page == null ? void 0 : page.konten) == null ? void 0 : _a.blocks) == null ? void 0 : _b.find((b) => b.__kind__ === "daftarItem")) == null ? void 0 : _c.daftarItem;
  const struktur = rawItems ? rawItems.map(parseStrukturItem).filter((m) => m.jabatan || m.nama) : DEFAULT_STRUKTUR;
  return /* @__PURE__ */ jsxRuntimeExports.jsxs(Layout, { children: [
    /* @__PURE__ */ jsxRuntimeExports.jsx(
      PageHeader,
      {
        title: "Struktur Organisasi",
        description: "Susunan pengurus dan pengelola JDIH Universitas 17 Agustus 1945 Banyuwangi",
        breadcrumbs: [
          { label: "Tentang JDIH", to: "/tentang/sejarah" },
          { label: "Struktur Organisasi" }
        ]
      }
    ),
    /* @__PURE__ */ jsxRuntimeExports.jsx(
      "section",
      {
        className: "py-12 bg-background",
        "data-ocid": "tentang_struktur.section",
        children: /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "container mx-auto px-4", children: /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "max-w-3xl mx-auto", children: [
          /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex items-center gap-3 mb-6", children: [
            /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "w-10 h-10 bg-primary/10 rounded-lg flex items-center justify-center flex-shrink-0", children: /* @__PURE__ */ jsxRuntimeExports.jsx(Users, { className: "w-5 h-5 text-primary" }) }),
            /* @__PURE__ */ jsxRuntimeExports.jsx("h2", { className: "font-display font-bold text-xl text-foreground", children: "Struktur Organisasi" })
          ] }),
          isLoading ? /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "space-y-2", children: [1, 2, 3, 4, 5, 6].map((n) => /* @__PURE__ */ jsxRuntimeExports.jsx(Skeleton, { className: "h-12 w-full" }, n)) }) : /* @__PURE__ */ jsxRuntimeExports.jsxs(jsxRuntimeExports.Fragment, { children: [
            /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "hidden sm:block bg-card border border-border rounded-xl overflow-hidden shadow-card", children: /* @__PURE__ */ jsxRuntimeExports.jsxs("table", { className: "w-full text-sm", children: [
              /* @__PURE__ */ jsxRuntimeExports.jsx("thead", { children: /* @__PURE__ */ jsxRuntimeExports.jsxs("tr", { className: "bg-primary/5 border-b border-border", children: [
                /* @__PURE__ */ jsxRuntimeExports.jsx("th", { className: "text-left py-3 px-4 font-semibold text-foreground w-8", children: "No." }),
                /* @__PURE__ */ jsxRuntimeExports.jsx("th", { className: "text-left py-3 px-4 font-semibold text-foreground", children: "Jabatan" }),
                /* @__PURE__ */ jsxRuntimeExports.jsx("th", { className: "text-left py-3 px-4 font-semibold text-foreground", children: "Nama" }),
                /* @__PURE__ */ jsxRuntimeExports.jsx("th", { className: "text-left py-3 px-4 font-semibold text-foreground", children: "Unit" })
              ] }) }),
              /* @__PURE__ */ jsxRuntimeExports.jsx("tbody", { className: "divide-y divide-border", children: struktur.map((item, idx) => /* @__PURE__ */ jsxRuntimeExports.jsxs(
                "tr",
                {
                  className: "hover:bg-muted/20 transition-colors",
                  "data-ocid": `tentang_struktur.row.${idx + 1}`,
                  children: [
                    /* @__PURE__ */ jsxRuntimeExports.jsx("td", { className: "py-3 px-4 text-muted-foreground", children: idx + 1 }),
                    /* @__PURE__ */ jsxRuntimeExports.jsx("td", { className: "py-3 px-4 font-medium text-foreground", children: item.jabatan }),
                    /* @__PURE__ */ jsxRuntimeExports.jsx("td", { className: "py-3 px-4 text-foreground", children: item.nama }),
                    /* @__PURE__ */ jsxRuntimeExports.jsx("td", { className: "py-3 px-4 text-muted-foreground text-xs", children: item.unit })
                  ]
                },
                `${item.jabatan}-${item.nama}`
              )) })
            ] }) }),
            /* @__PURE__ */ jsxRuntimeExports.jsx("div", { className: "sm:hidden space-y-3", children: struktur.map((item, idx) => /* @__PURE__ */ jsxRuntimeExports.jsxs(
              "div",
              {
                className: "bg-card border border-border rounded-xl p-4 shadow-card",
                "data-ocid": `tentang_struktur.card.${idx + 1}`,
                children: [
                  /* @__PURE__ */ jsxRuntimeExports.jsxs("div", { className: "flex items-center gap-2 mb-2", children: [
                    /* @__PURE__ */ jsxRuntimeExports.jsx(Building2, { className: "w-4 h-4 text-primary" }),
                    /* @__PURE__ */ jsxRuntimeExports.jsx("span", { className: "text-xs font-semibold text-accent uppercase tracking-wide", children: item.jabatan })
                  ] }),
                  /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-sm font-semibold text-foreground", children: item.nama }),
                  /* @__PURE__ */ jsxRuntimeExports.jsx("p", { className: "text-xs text-muted-foreground mt-0.5", children: item.unit })
                ]
              },
              item.jabatan
            )) })
          ] })
        ] }) })
      }
    )
  ] });
}
export {
  TentangStrukturOrganisasi
};
