// @caffeine/template-frontend/dist/assets/useBackend-xFEMRG87.js
import { s as useQuery, t as useQueryClient, h as api } from "./index-B7-RSq-m.js";
import { u as useMutation } from "./useMutation-D--BDafu.js";
function useInvalidateAll() {
  const qc = useQueryClient();
  return () => {
    qc.invalidateQueries({ queryKey: ["dokumen"] });
    qc.invalidateQueries({ queryKey: ["dokumenAdmin"] });
    qc.invalidateQueries({ queryKey: ["kategori"] });
    qc.invalidateQueries({ queryKey: ["berita"] });
    qc.invalidateQueries({ queryKey: ["statistik"] });
    qc.invalidateQueries({ queryKey: ["activityLog"] });
    qc.invalidateQueries({ queryKey: ["pendingReviewCount"] });
    qc.invalidateQueries({ queryKey: ["galeri"] });
    qc.invalidateQueries({ queryKey: ["users"] });
  };
}
function useDokumen(filter) {
  return useQuery({
    queryKey: ["dokumen", filter],
    queryFn: () => api.dokumen.list(filter),
    staleTime: 3e4,
    select: (d) => d ?? []
  });
}
function useDokumenAdmin(filter) {
  return useQuery({
    queryKey: ["dokumenAdmin", filter],
    queryFn: () => api.dokumen.listAdmin(filter),
    staleTime: 3e4,
    select: (d) => d ?? []
  });
}
function useDokumenDetail(id) {
  return useQuery({
    queryKey: ["dokumen", id],
    queryFn: () => api.dokumen.detail(id),
    enabled: !!id,
    staleTime: 3e4
  });
}
function useDeleteDokumen() {
  const inv = useInvalidateAll();
  return useMutation({
    mutationFn: (id) => api.dokumen.delete(id),
    onSuccess: inv
  });
}
function useSubmitForReview() {
  const inv = useInvalidateAll();
  return useMutation({
    mutationFn: async (id) => {
      const r = await api.dokumen.submitForReview(id);
      if (r.err) throw new Error(r.err);
      return r;
    },
    onSuccess: inv
  });
}
function usePublishDokumen() {
  const inv = useInvalidateAll();
  return useMutation({
    mutationFn: async (id) => {
      const r = await api.dokumen.publish(id);
      if (r.err) throw new Error(r.err);
      return r;
    },
    onSuccess: inv
  });
}
function useReturnToDraft() {
  const inv = useInvalidateAll();
  return useMutation({
    mutationFn: async ({ id, catatan }) => {
      const r = await api.dokumen.returnToDraft(id, catatan);
      if (r.err) throw new Error(r.err);
      return r;
    },
    onSuccess: inv
  });
}
function useKategori() {
  return useQuery({
    queryKey: ["kategori"],
    queryFn: () => api.kategori.list(),
    staleTime: 6e4,
    select: (d) => d ?? []
  });
}
function useDeleteKategori() {
  const inv = useInvalidateAll();
  return useMutation({
    mutationFn: (id) => api.kategori.delete(id),
    onSuccess: inv
  });
}
function useBerita() {
  return useQuery({
    queryKey: ["berita"],
    queryFn: () => api.berita.list(),
    staleTime: 3e4,
    select: (d) => d ?? []
  });
}
function useBeritaDetail(id) {
  return useQuery({
    queryKey: ["berita", id],
    queryFn: () => api.berita.detail(id),
    enabled: !!id,
    staleTime: 3e4
  });
}
function useDeleteBerita() {
  const inv = useInvalidateAll();
  return useMutation({
    mutationFn: (id) => api.berita.delete(id),
    onSuccess: inv
  });
}
function useStatistik() {
  return useQuery({
    queryKey: ["statistik"],
    queryFn: () => api.statistik.get(),
    staleTime: 6e4
  });
}
function useGaleri() {
  return useQuery({
    queryKey: ["galeri"],
    queryFn: () => api.galeri.list(),
    staleTime: 3e4,
    select: (d) => d ?? []
  });
}
function useDeleteGaleriItem() {
  const qc = useQueryClient();
  return useMutation({
    mutationFn: (id) => api.galeri.delete(id),
    onSuccess: () => qc.invalidateQueries({ queryKey: ["galeri"] })
  });
}
function useTentangPages() {
  return useQuery({
    queryKey: ["tentangPages"],
    queryFn: () => api.tentang.list(),
    staleTime: 6e4,
    select: (d) => d ?? []
  });
}
function useTentangPage(slug) {
  return useQuery({
    queryKey: ["tentangPage", slug],
    queryFn: () => api.tentang.get(slug),
    enabled: !!slug,
    staleTime: 6e4
  });
}
function useUpdateTentangPage() {
  const qc = useQueryClient();
  return useMutation({
    mutationFn: ({ slug, input }) => api.tentang.update(slug, input),
    onSuccess: () => {
      qc.invalidateQueries({ queryKey: ["tentangPages"] });
      qc.invalidateQueries({ queryKey: ["tentangPage"] });
    }
  });
}
function useActivityLog() {
  return useQuery({
    queryKey: ["activityLog"],
    queryFn: () => api.users.activityLog(),
    staleTime: 6e4,
    select: (d) => d ?? []
  });
}
export {
  useDokumen as a,
  useBerita as b,
  useGaleri as c,
  useKategori as d,
  useDokumenDetail as e,
  useBeritaDetail as f,
  useTentangPage as g,
  useActivityLog as h,
  useDokumenAdmin as i,
  useDeleteDokumen as j,
  useSubmitForReview as k,
  usePublishDokumen as l,
  useReturnToDraft as m,
  useDeleteKategori as n,
  useDeleteBerita as o,
  useTentangPages as p,
  useUpdateTentangPage as q,
  useDeleteGaleriItem as r,
  useStatistik as u
};
