// @caffeine/template-frontend/src/mocks/backend.ts
import type { backendInterface } from "../backend";
import { StatusDokumen, UserRole } from "../backend";
import { ExternalBlob } from "../backend";

const sampleBlob = ExternalBlob.fromURL("https://example.com/sample.pdf");

export const mockBackend: backendInterface = {
  addBerita: async (input) => ({
    id: BigInt(1),
    tanggal: BigInt(Date.now()),
    judul: input.judul,
    author: input.author,
    konten: input.konten,
  }),
  addDokumen: async (input) => ({
    id: BigInt(1),
    tag: input.tag,
    kategoriId: input.kategoriId,
    status: input.status,
    tahun: input.tahun,
    judul: input.judul,
    abstrak: input.abstrak,
    tanggalPenetapan: input.tanggalPenetapan,
    nomor: input.nomor,
    downloadCount: BigInt(0),
    filePdf: input.filePdf,
  }),
  addKategori: async (nama) => ({ id: BigInt(1), nama }),
  assignCallerUserRole: async () => undefined,
  deleteBerita: async () => true,
  deleteDokumen: async () => true,
  deleteKategori: async () => true,
  downloadDokumen: async () => true,
  getBeritaDetail: async (id) => ({
    id,
    tanggal: BigInt(1704067200000),
    judul: "Sosialisasi Peraturan Akademik Tahun 2024",
    author: "Admin JDIH",
    konten:
      "JDIH UNTAG Banyuwangi menyelenggarakan sosialisasi peraturan akademik terbaru kepada seluruh civitas akademika. Kegiatan ini bertujuan untuk memastikan semua pihak memahami regulasi yang berlaku di lingkungan kampus.",
  }),
  getCallerUserRole: async () => UserRole.guest,
  getDokumenDetail: async (id) => ({
    id,
    tag: ["akademik", "peraturan"],
    kategoriId: BigInt(1),
    status: StatusDokumen.Berlaku,
    tahun: BigInt(2024),
    judul: "Peraturan Rektor No. 01 Tahun 2024 tentang Tata Tertib Akademik",
    abstrak:
      "Peraturan ini mengatur tata tertib akademik mahasiswa dan dosen di lingkungan UNTAG Banyuwangi, mencakup hak, kewajiban, dan sanksi pelanggaran.",
    tanggalPenetapan: "2024-01-15",
    nomor: "001/PR/UNTAG-BWI/2024",
    downloadCount: BigInt(127),
    filePdf: sampleBlob,
  }),
  getStatistik: async () => ({
    totalArtikel: BigInt(2),
    totalDokumen: BigInt(5),
    perKategori: [
      [BigInt(1), "Peraturan Rektor", BigInt(2)],
      [BigInt(2), "Keputusan Rektor", BigInt(1)],
      [BigInt(3), "Surat Edaran", BigInt(1)],
      [BigInt(4), "Statuta", BigInt(1)],
    ],
    totalUnduhan: BigInt(342),
  }),
  isCallerAdmin: async () => false,
  listBerita: async () => [
    {
      id: BigInt(1),
      tanggal: BigInt(1704067200000),
      judul: "Sosialisasi Peraturan Akademik Tahun 2024",
      author: "Admin JDIH",
      konten:
        "JDIH UNTAG Banyuwangi menyelenggarakan sosialisasi peraturan akademik terbaru kepada seluruh civitas akademika.",
    },
    {
      id: BigInt(2),
      tanggal: BigInt(1701388800000),
      judul: "Pembaruan Dokumen MoU Kerjasama Institusi 2023",
      author: "Bagian Hukum",
      konten:
        "Dalam rangka memperkuat kerjasama antar institusi, UNTAG Banyuwangi telah memperbarui sejumlah dokumen MoU dengan mitra strategis.",
    },
  ],
  listDokumen: async () => [
    {
      id: BigInt(1),
      tag: ["akademik"],
      kategoriId: BigInt(1),
      status: StatusDokumen.Berlaku,
      tahun: BigInt(2024),
      judul: "Peraturan Rektor No. 01 Tahun 2024 tentang Tata Tertib Akademik",
      abstrak:
        "Peraturan ini mengatur tata tertib akademik mahasiswa dan dosen di lingkungan UNTAG Banyuwangi.",
      tanggalPenetapan: "2024-01-15",
      nomor: "001/PR/UNTAG-BWI/2024",
      downloadCount: BigInt(127),
      filePdf: sampleBlob,
    },
    {
      id: BigInt(2),
      tag: ["keuangan"],
      kategoriId: BigInt(2),
      status: StatusDokumen.Berlaku,
      tahun: BigInt(2023),
      judul: "Keputusan Rektor No. 05 Tahun 2023 tentang Standar Biaya Pendidikan",
      abstrak:
        "Keputusan ini menetapkan standar biaya pendidikan yang berlaku di UNTAG Banyuwangi untuk tahun akademik 2023/2024.",
      tanggalPenetapan: "2023-07-01",
      nomor: "005/KR/UNTAG-BWI/2023",
      downloadCount: BigInt(89),
      filePdf: sampleBlob,
    },
    {
      id: BigInt(3),
      tag: ["kepegawaian"],
      kategoriId: BigInt(3),
      status: StatusDokumen.Berlaku,
      tahun: BigInt(2023),
      judul: "Surat Edaran Rektor tentang Disiplin Kepegawaian",
      abstrak:
        "Surat edaran ini mengingatkan seluruh pegawai untuk mematuhi peraturan disiplin yang berlaku.",
      tanggalPenetapan: "2023-09-01",
      nomor: "SE-12/UNTAG-BWI/2023",
      downloadCount: BigInt(45),
      filePdf: sampleBlob,
    },
    {
      id: BigInt(4),
      tag: ["kelembagaan"],
      kategoriId: BigInt(4),
      status: StatusDokumen.Berlaku,
      tahun: BigInt(2022),
      judul: "Statuta UNTAG Banyuwangi Tahun 2022",
      abstrak:
        "Statuta ini merupakan peraturan dasar yang mengatur penyelenggaraan UNTAG Banyuwangi secara keseluruhan.",
      tanggalPenetapan: "2022-03-20",
      nomor: "STATUTA/UNTAG-BWI/2022",
      downloadCount: BigInt(203),
      filePdf: sampleBlob,
    },
    {
      id: BigInt(5),
      tag: ["akademik", "lama"],
      kategoriId: BigInt(1),
      status: StatusDokumen.TidakBerlaku,
      tahun: BigInt(2021),
      judul: "Peraturan Rektor No. 03 Tahun 2021 tentang Kurikulum Merdeka Belajar",
      abstrak:
        "Peraturan ini mengatur implementasi Kurikulum Merdeka Belajar di lingkungan UNTAG Banyuwangi.",
      tanggalPenetapan: "2021-08-10",
      nomor: "003/PR/UNTAG-BWI/2021",
      downloadCount: BigInt(67),
      filePdf: sampleBlob,
    },
  ],
  listKategori: async () => [
    { id: BigInt(1), nama: "Peraturan Rektor" },
    { id: BigInt(2), nama: "Keputusan Rektor" },
    { id: BigInt(3), nama: "Surat Edaran" },
    { id: BigInt(4), nama: "Statuta" },
    { id: BigInt(5), nama: "Peraturan Senat" },
    { id: BigInt(6), nama: "SOP" },
    { id: BigInt(7), nama: "MoU / Perjanjian" },
  ],
  updateBerita: async (id, input) => ({
    id,
    tanggal: BigInt(Date.now()),
    judul: input.judul,
    author: input.author,
    konten: input.konten,
  }),
  updateDokumen: async (id, input) => ({
    id,
    tag: input.tag,
    kategoriId: input.kategoriId,
    status: input.status,
    tahun: input.tahun,
    judul: input.judul,
    abstrak: input.abstrak,
    tanggalPenetapan: input.tanggalPenetapan,
    nomor: input.nomor,
    downloadCount: BigInt(0),
    filePdf: input.filePdf,
  }),
  updateKategori: async (id, nama) => ({ id, nama }),
  _immutableObjectStorageBlobsAreLive: async () => [],
  _immutableObjectStorageBlobsToDelete: async () => [],
  _immutableObjectStorageConfirmBlobDeletion: async () => undefined,
  _immutableObjectStorageCreateCertificate: async () => ({ certificate: new Uint8Array(), key: "" }) as never,
  _immutableObjectStorageRefillCashier: async () => ({ ok: null }) as never,
  _immutableObjectStorageUpdateGatewayPrincipals: async () => undefined,
} as unknown as backendInterface;
